
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="rlx input variables">
      
      
        <meta name="author" content="The Abinit group">
      
      
        <link rel="canonical" href="https://gmatteo.github.io/abinit/variables/rlx/">
      
      
        <link rel="prev" href="../dfpt/">
      
      
        <link rel="next" href="../rttddft/">
      
      
      <link rel="icon" href="../../images/favicon.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.43">
    
    
      
        <title>Relaxation - abinit</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0253249f.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
    
      
    

    <!--
    IMPORTANT: Don't use boostrap in particular boostrap.css since it comes into conflict with mkdocs-material theme.
    Prefer low-level tools based on Jquery or JS library such as Tippyjs to get the work done.
    -->

    <!-- Jquery from Google CDN -->
    <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>

    <!-- Load Jquery UI -->
    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.min.js"></script>
    <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">

    <!-- Handle jQuery plugin naming conflict between jQuery UI and Bootstrap
    https://stackoverflow.com/questions/13731400/jqueryui-tooltips-are-competing-with-twitter-bootstrap
    <script>
      $.widget.bridge('uibutton', $.ui.button);
      $.widget.bridge('uitooltip', $.ui.tooltip);
    </script>
    -->

    <!-- jQuery Modal -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-modal/0.9.1/jquery.modal.min.js"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-modal/0.9.1/jquery.modal.min.css" />

    <!-- Tippyjs for popovers https://atomiks.github.io/tippyjs/ -->
    <script src="https://unpkg.com/@popperjs/core@2"></script>
    <script src="https://unpkg.com/tippy.js@6"></script>

    <!-- font-awesome -->
    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">

    <!--Pace support. http://github.hubspot.com/pace/ -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/pace.min.js"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/white/pace-theme-big-counter.min.css" />

    <script>
        paceOptions = {
            ajax: true,      // disabled
            document: true,  // activated
            eventLag: false  // disabled
            //target: 'article'
            //elements: {selectors: ['.my-page']}
        };
    </script>

    <!-- MathJax support
    <script type="text/javascript" async
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
    -->

    <!-- This version is compatible with plotly. -->
    <script type="text/javascript" async
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_SVG">
    </script>

    <!-- Configure MathJax to produce automatic equation numbers http://docs.mathjax.org/en/latest/tex.html -->
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        //jax: ["input/TeX","output/HTML-CSS"],
        //tex2jax: {inlineMath: [['$','$']]},
        //displayAlign: "center",
        TeX: {
            equationNumbers: { autoNumber: "AMS" },
            extensions: ["AMSmath.js", "boldsymbol.js"],
            Macros: {
                "*": "{\\dagger}",
                "ra": "{\\rangle}",
                "la": "{\\langle}",
                // Workaround for bm package. See https://github.com/mathjax/MathJax/issues/1219
                bm: ["{\\boldsymbol #1}",1],
                // Mathcal symbols
                mcL: "{\\mathcal{L}}",
                mcG: "{\\mathcal{G}}",
                mcR: "{\\mathcal{R}}",
                mcS: "{\\mathcal{S}}",
                mcC: "{\\mathcal{C}}",
                mcO: "{\\mathcal{O}}",
                // Acronyms
                KS: "{\\text{KS}}",
                eph: "{\\text{e-ph}}",
                ee: "{\\text{e-e}}",
                phph: "{\\text{ph-ph}}",
                FM: "{\\text{FM}}",
                DW: "{\\text{DW}}",
                QP: "{\\text{QP}}",
                BZ: "{\\text{BZ}}",
                sign: "{\\text{sign}}",
                xc: "{\\text{xc}}",
                tchi: "{\\tilde\\chi}",
                // Symbols commonly used (k-points, eigenvalues ...)
                rr: "{\\bf r}",
                RR: "{\\bf R}",
                tt: "{\\bf t}",
                GG: "{\\bf G}",
                bg: "{\\bf g}",
                kk: "{\\bf k}",
                qq: "{\\bf q}",
                kq: "{\\kk + \\qq}",
                qG: "{\\bf q + G}",
                kG: "{\\bf k + G}",
                kmq: "{\\kk - \\qq}",
                kpq: "{\\kk + \\qq}",
                kpG: "{\\kk + \\GG}",
                qpG: "{\\qq + \\GG}",
                nk: "{n \\kk}",
                mk: "{, \\kk}",
                nks: "{n \\kk\\sigma}",
                mks: "{m \\kk\\sigma}",
                enk: "{\\varepsilon_\\nk}",
                emkq: "{\\varepsilon_{m\\kk+\\qq}}",
                qnu: "{\\qq \\nu}",
                wqnu: "{\\omega_{\\qnu}}",
                vnk: "{\\mathbf{v}_{n\\kk}}",
                vmkq: "{\\mathbf{v}_{m\\kk+\\qq}}",
                vnka: "{\\mathrm{v}_{n\\kk,\\alpha}}",
                vnkb: "{\\mathrm{v}_{n\\kk,\\beta}}",
                ef: "{\\varepsilon_F}",
                // EPH quantities
                gkkp: "{g_{mn\\nu}(\\kk,\\qq)}",
                gkq: "{g_{mn\\nu}(\\kk,\\qq)}",
                //\newcommand{\gkkpL}{g_{mn\nu}^\Lc(\kb,\qb)}
                // Many-body theory
                Go: "{G_0}",
                Wo: "{W_0}",
                Ueff: "{U_{\\text{eff}}}",
                Gove: "{\\overline{G}}",
                Gund: "{\\underline{G}}",
                // Greek symbols + tilde version
                ee: "{\\epsilon}",
                tee: "{\\tilde{\\epsilon}}",
                ww: "{\\omega}",
                tww: "{\\tilde\\omega}",
                dd: "{\\,\\text{d}}",
                vxc: "{v_\\xc}",
                // symrel
                Ri: "{\\mcR^{-1}}",
                Rit: "{\\mcR^{-1\\tau}}",
                // symrec
                Si: "{\\mcS^{-1}}}",
                Sit: "{\\Si^\\tau}}",
                // partial and functional derivative.
                PDER: ["{\\dfrac{\\partial #1}{\\partial #2}}", 2],
                FDER: ["{\\dfrac{\\delta #1}{\\delta #2}}", 2],
                // PAW
                tPsi: "{\\tilde\\Psi}",
                tpsi: "{\\tilde\\psi}",
                tPhi: "{\\tilde\\Phi}",
                tphi: "{\\tilde\\phi}",
                tprj: "{\\tilde p}"
            }
          }
      });
    </script>

    

    

    <!-- Load fonts used for input variables -->
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">

    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/extra.css">
    
      <link rel="stylesheet" href="../../css/gallery.css">
    
      <link rel="stylesheet" href="../../css/my_codehilite.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="deep-orange" data-md-color-accent="deep-orange">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#rlx-input-variables" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="abinit" class="md-header__button md-logo" aria-label="abinit" data-md-component="logo">
      
  <img src="../../images/logo-abinit-2015.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            abinit
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Relaxation
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/abinit/abinit/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    github-abinit
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
    
      
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../installation/" class="md-tabs__link">
          
  
  User Guides

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../topics/features/" class="md-tabs__link">
          
  
  Topics

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../" class="md-tabs__link">
          
  
  Variables

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tutorial/" class="md-tabs__link">
          
  
  Tutorials

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../faq/configuration_and_compilation/" class="md-tabs__link">
          
  
  FAQ

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../theory/geometry/" class="md-tabs__link">
          
  
  Theory

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../developers/git_and_gitlab/" class="md-tabs__link">
          
  
  Developers

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../about/release-notes/" class="md-tabs__link">
          
  
  About

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="abinit" class="md-nav__button md-logo" aria-label="abinit" data-md-component="logo">
      
  <img src="../../images/logo-abinit-2015.png" alt="logo">

    </a>
    abinit
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/abinit/abinit/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    github-abinit
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    User Guides
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            User Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_1" >
        
          
          <label class="md-nav__link" for="__nav_1_1" id="__nav_1_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Installation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_1">
            <span class="md-nav__icon md-icon"></span>
            Installation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    HowTo
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../INSTALL_MacOS/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MacOS
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../INSTALL_Ubuntu/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ubuntu
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../INSTALL_CentOS/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CentOS
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../INSTALL_EasyBuild/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    EasyBuild
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../INSTALL_gpu/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GPUs
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NewUser
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/abinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Abinit
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/abitk/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Abitk
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/aim/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Aim
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/anaddb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Anaddb
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/atdep/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    aTDEP
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/cut3d/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Cut3d
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/fold2bloch/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Fold2Bloch
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/mrgddb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Mrgddb
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/mrgscr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Mrgscr
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/multibinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multibinit
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/optic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Optic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/respfn/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Respfn
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guide/spacegroup/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spacegroup
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Topics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Topics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/features/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Features
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Topic List
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Abipy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Abipy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/APPA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPA
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Artificial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Artificial
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/aTDEP/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    aTDEP
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/AtomCentered/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    AtomCentered
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/AtomManipulator/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    AtomManipulator
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/AtomTypes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    AtomTypes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Bader/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bader
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Band2eps/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Band2eps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Berry/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Berry
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/BandOcc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    BandOcc
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/BSE/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    BSE
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/BoundingProcess/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    BoundingProcess
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ConstrainedDFT/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ConstrainedDFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ConstrainedPol/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ConstrainedPol
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Control
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Coulomb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coulomb
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/CrossingBarriers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CrossingBarriers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/CalcUJ/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CalcUJ
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/crystal/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crystal
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/DFT%2BU/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFT+U
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/DeltaSCF/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DeltaSCF
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/DensityPotential/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DensityPotential
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Dev/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Dev
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/DFPT/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFPT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/DMFT/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DMFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/DynamicsMultibinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DynamicsMultibinit
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/EffectiveMass/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    EffectiveMass
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/EFG/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    EFG
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Elastic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Elastic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ElPhonInt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ElPhonInt
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ElPhonTransport/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ElPhonTransport
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ElecDOS/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ElecDOS
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ElecBandStructure/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ElecBandStructure
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ExtFPMD/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ExtFPMD
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/FileFormats/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FileFormats
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/FitProcess/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FitProcess
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/ForcesStresses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ForcesStresses
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/FrequencyMeshMBPT/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FrequencyMeshMBPT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/GeoConstraints/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GeoConstraints
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/GeoOpt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GeoOpt
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Git/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Git
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/GSintroduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GSintroduction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/GW/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/GWR/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GWR
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/GWls/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GWls
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Hybrids/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Hybrids
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Install/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Install
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/k-points/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    K points
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/LatticeModel/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    LatticeModel
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/LatticeWannier/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    LatticeWannier
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/LWFModel/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    LWFModel
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/LDAminushalf/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    LDAminushalf
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/LOTF/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    LOTF
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/longwave/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Longwave
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Macroave/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Macroave
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/MagField/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MagField
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/MagMom/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MagMom
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/MolecularDynamics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MolecularDynamics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/multidtset/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multidtset
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/NMR/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMR
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/nonlinear/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Nonlinear
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Optic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Optic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Output/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Output
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/parallelism/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Parallelism
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/PAW/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PAW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/PIMD/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PIMD
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Planewaves/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Planewaves
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Phonons/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phonons
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/PhononBands/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PhononBands
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/PhononWidth/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PhononWidth
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Polaron/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Polaron
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/PortabilityNonRegression/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PortabilityNonRegression
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/positron/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Positron
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/printing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Printing
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/PseudosPAW/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PseudosPAW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/q-points/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Q points
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/RandStopPow/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RandStopPow
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Recursion/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Recursion
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/RPACorrEn/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RPACorrEn
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/RTTDDFT/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RTTDDFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/SCFControl/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SCFControl
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/SCFAlgorithms/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SCFAlgorithms
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/SelfEnergy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SelfEnergy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/SmartSymm/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SmartSymm
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/spinpolarisation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spinpolarisation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/SpinDynamicsMultibinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SpinDynamicsMultibinit
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/STM/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    STM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Susceptibility/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Susceptibility
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/TDDFT/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TDDFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/TDepES/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TDepES
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Temperature/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Temperature
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/TransPath/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TransPath
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/TuningSpeedMem/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TuningSpeedMem
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Unfolding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Unfolding
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/UnitCell/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    UnitCell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/vdw/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Vdw
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Verification/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Verification
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Wannier/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Wannier
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/Wavelets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Wavelets
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/xc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Xc
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../topics/external_resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Resources
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Variables
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Variables
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    All Variables
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" checked>
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Abinit
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Abinit
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../basic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bse/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bethe-Salpeter
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../dev/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Developers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../dmft/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DMFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eph/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Electron-Phonon
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ffield/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Varff
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../files/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    File-handling
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../geo/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Geometry
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gstate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ground-State
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gw/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../internal/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Internal
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../paral/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Parallelism
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../paw/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PAW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../dfpt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFPT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Relaxation
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Relaxation
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adpimd" class="md-nav__link">
    <span class="md-ellipsis">
      adpimd
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#adpimd_gamma" class="md-nav__link">
    <span class="md-ellipsis">
      adpimd_gamma
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#amu" class="md-nav__link">
    <span class="md-ellipsis">
      amu
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bmass" class="md-nav__link">
    <span class="md-ellipsis">
      bmass
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#chkdilatmx" class="md-nav__link">
    <span class="md-ellipsis">
      chkdilatmx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cineb_start" class="md-nav__link">
    <span class="md-ellipsis">
      cineb_start
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#delayperm" class="md-nav__link">
    <span class="md-ellipsis">
      delayperm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diismemory" class="md-nav__link">
    <span class="md-ellipsis">
      diismemory
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dilatmx" class="md-nav__link">
    <span class="md-ellipsis">
      dilatmx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dtion" class="md-nav__link">
    <span class="md-ellipsis">
      dtion
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dynimage" class="md-nav__link">
    <span class="md-ellipsis">
      dynimage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ecutsm" class="md-nav__link">
    <span class="md-ellipsis">
      ecutsm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#friction" class="md-nav__link">
    <span class="md-ellipsis">
      friction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frictionbar" class="md-nav__link">
    <span class="md-ellipsis">
      frictionbar
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fxcartfactor" class="md-nav__link">
    <span class="md-ellipsis">
      fxcartfactor
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_algor" class="md-nav__link">
    <span class="md-ellipsis">
      ga_algor
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_fitness" class="md-nav__link">
    <span class="md-ellipsis">
      ga_fitness
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_n_rules" class="md-nav__link">
    <span class="md-ellipsis">
      ga_n_rules
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_opt_percent" class="md-nav__link">
    <span class="md-ellipsis">
      ga_opt_percent
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_rules" class="md-nav__link">
    <span class="md-ellipsis">
      ga_rules
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#geoopt" class="md-nav__link">
    <span class="md-ellipsis">
      geoopt
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getcell" class="md-nav__link">
    <span class="md-ellipsis">
      getcell
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getvel" class="md-nav__link">
    <span class="md-ellipsis">
      getvel
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxcart" class="md-nav__link">
    <span class="md-ellipsis">
      getxcart
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxred" class="md-nav__link">
    <span class="md-ellipsis">
      getxred
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecon" class="md-nav__link">
    <span class="md-ellipsis">
      goprecon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecprm" class="md-nav__link">
    <span class="md-ellipsis">
      goprecprm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hmcsst" class="md-nav__link">
    <span class="md-ellipsis">
      hmcsst
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hmctt" class="md-nav__link">
    <span class="md-ellipsis">
      hmctt
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatcon" class="md-nav__link">
    <span class="md-ellipsis">
      iatcon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfix" class="md-nav__link">
    <span class="md-ellipsis">
      iatfix
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixx" class="md-nav__link">
    <span class="md-ellipsis">
      iatfixx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixy" class="md-nav__link">
    <span class="md-ellipsis">
      iatfixy
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixz" class="md-nav__link">
    <span class="md-ellipsis">
      iatfixz
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#imgmov" class="md-nav__link">
    <span class="md-ellipsis">
      imgmov
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#imgwfstor" class="md-nav__link">
    <span class="md-ellipsis">
      imgwfstor
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#istatimg" class="md-nav__link">
    <span class="md-ellipsis">
      istatimg
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdtemp" class="md-nav__link">
    <span class="md-ellipsis">
      mdtemp
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdwall" class="md-nav__link">
    <span class="md-ellipsis">
      mdwall
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_mxstep" class="md-nav__link">
    <span class="md-ellipsis">
      mep_mxstep
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_solver" class="md-nav__link">
    <span class="md-ellipsis">
      mep_solver
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mixesimgf" class="md-nav__link">
    <span class="md-ellipsis">
      mixesimgf
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#moldyn" class="md-nav__link">
    <span class="md-ellipsis">
      moldyn
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natcon" class="md-nav__link">
    <span class="md-ellipsis">
      natcon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfix" class="md-nav__link">
    <span class="md-ellipsis">
      natfix
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixx" class="md-nav__link">
    <span class="md-ellipsis">
      natfixx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixy" class="md-nav__link">
    <span class="md-ellipsis">
      natfixy
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixz" class="md-nav__link">
    <span class="md-ellipsis">
      natfixz
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nconeq" class="md-nav__link">
    <span class="md-ellipsis">
      nconeq
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_algo" class="md-nav__link">
    <span class="md-ellipsis">
      neb_algo
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_cell_algo" class="md-nav__link">
    <span class="md-ellipsis">
      neb_cell_algo
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_spring" class="md-nav__link">
    <span class="md-ellipsis">
      neb_spring
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nimage" class="md-nav__link">
    <span class="md-ellipsis">
      nimage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nnos" class="md-nav__link">
    <span class="md-ellipsis">
      nnos
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#noseinert" class="md-nav__link">
    <span class="md-ellipsis">
      noseinert
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntime" class="md-nav__link">
    <span class="md-ellipsis">
      ntime
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntimimage" class="md-nav__link">
    <span class="md-ellipsis">
      ntimimage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optcell" class="md-nav__link">
    <span class="md-ellipsis">
      optcell
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimass" class="md-nav__link">
    <span class="md-ellipsis">
      pimass
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimd_constraint" class="md-nav__link">
    <span class="md-ellipsis">
      pimd_constraint
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pitransform" class="md-nav__link">
    <span class="md-ellipsis">
      pitransform
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prtatlist" class="md-nav__link">
    <span class="md-ellipsis">
      prtatlist
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qmass" class="md-nav__link">
    <span class="md-ellipsis">
      qmass
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#random_atpos" class="md-nav__link">
    <span class="md-ellipsis">
      random_atpos
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#restartxf" class="md-nav__link">
    <span class="md-ellipsis">
      restartxf
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#signperm" class="md-nav__link">
    <span class="md-ellipsis">
      signperm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strfact" class="md-nav__link">
    <span class="md-ellipsis">
      strfact
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#string_algo" class="md-nav__link">
    <span class="md-ellipsis">
      string_algo
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strprecon" class="md-nav__link">
    <span class="md-ellipsis">
      strprecon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strtarget" class="md-nav__link">
    <span class="md-ellipsis">
      strtarget
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolimg" class="md-nav__link">
    <span class="md-ellipsis">
      tolimg
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxde" class="md-nav__link">
    <span class="md-ellipsis">
      tolmxde
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxf" class="md-nav__link">
    <span class="md-ellipsis">
      tolmxf
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel" class="md-nav__link">
    <span class="md-ellipsis">
      vel
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel_cell" class="md-nav__link">
    <span class="md-ellipsis">
      vel_cell
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vis" class="md-nav__link">
    <span class="md-ellipsis">
      vis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#wtatcon" class="md-nav__link">
    <span class="md-ellipsis">
      wtatcon
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../rttddft/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RTTDDFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vdw/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    VdW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../w90/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Wannier90
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../anaddb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Anaddb
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../atdep/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    aTDEP
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../multibinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multibinit
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../optic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Optic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../aim/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Aim
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../external_parameters/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../varset_stats/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Statistics
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Tutorials
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_2" >
        
          
          <label class="md-nav__link" for="__nav_4_2" id="__nav_4_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Basic tutorials
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_2">
            <span class="md-nav__icon md-icon"></span>
            Basic tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/base1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    basic1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/base2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    basic2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/base3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    basic3
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/base4/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    basic4
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/abinit_build/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ABINIT_build
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paw1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PAW1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/spin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spin
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/polarization/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Polarization & finite field
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/rf1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFPT1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/rf2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFPT2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/basepar/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Parallelism
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_10" >
        
          
          <label class="md-nav__link" for="__nav_4_10" id="__nav_4_10_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    ABINIT postprocessors
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_10_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_10">
            <span class="md-nav__icon md-icon"></span>
            ABINIT postprocessors
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/atdep1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    aTDEP1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/cut3d/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    cut3D
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/fold2bloch/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    fold2bloch
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/optic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    optic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/wannier90/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    wannier90
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/z2pack/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Z2pack
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_11" >
        
          
          <label class="md-nav__link" for="__nav_4_11" id="__nav_4_11_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    DFT beyond basics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_11_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_11">
            <span class="md-nav__icon md-icon"></span>
            DFT beyond basics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/nuc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    properties at the nucleus
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/dftu/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFT+U
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/ucalc_crpa/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    cRPA U(J)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/lruj/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Linear Response U(J)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paw2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PAW2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paw3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PAW3
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/positron/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    positron
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_12" >
        
          
          <label class="md-nav__link" for="__nav_4_12" id="__nav_4_12_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Many-Body Theory & excitations
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_12_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_12">
            <span class="md-nav__icon md-icon"></span>
            Many-Body Theory & excitations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/gw1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GW1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/gw2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GW2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/gwr_intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GWR_intro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/gwr1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GWR1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/bse/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bethe-Salpeter
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/dmft/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DMFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/tddft/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TDDFT (Casida)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/rttddft/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RT-TDDFT
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_13" >
        
          
          <label class="md-nav__link" for="__nav_4_13" id="__nav_4_13_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    DFPT & electron-phonon
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_13_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_13">
            <span class="md-nav__icon md-icon"></span>
            DFPT & electron-phonon
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/elastic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    elastic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/nlo/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    nonlinear
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/lattice_wannier/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    lattice_wannier
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/lw_quad/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    quadrupoles
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/eph_intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eph_intro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/eph4mob/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eph for mobilities
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/eph4zpr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eph for zero-point renormalization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/eph4vpq/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eph for variational polaron equations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/eph_legacy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eph_legacy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/eph_tdep_legacy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eph_tdep_legacy
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_14" >
        
          
          <label class="md-nav__link" for="__nav_4_14" id="__nav_4_14_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Multibinit
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_14_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_14">
            <span class="md-nav__icon md-icon"></span>
            Multibinit
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/multibinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    introduction to multibinit
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/lattice_model/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    lattice_model
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/spin_model/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    spin_model
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/lwf_model/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    lwf_model
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_15" >
        
          
          <label class="md-nav__link" for="__nav_4_15" id="__nav_4_15_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Advanced parallelism
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_15_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_15">
            <span class="md-nav__icon md-icon"></span>
            Advanced parallelism
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paral_bandpw/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    paral_band_PW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paral_gswvl/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    paral_wavelets
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paral_moldyn/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    paral_MD
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paral_images/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    paral_images
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paral_dfpt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    paral_DFPT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/paral_mbt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    paral_MBPT
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    FAQ
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            FAQ
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/configuration_and_compilation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Configuration and compilation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/general_questions/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    General questions
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/pre_and_post_processing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pre/Post processing
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/ground_state/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ground state
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/dfpt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DFPT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/Troubleshooting/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Troubleshooting (old-dokuwiki)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../faq/tuning_abinit/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tuning (old-dokuwiki)
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Theory
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Theory
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/geometry/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Geometry
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/spherical_harmonics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spherical Harmonics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/wavefunctions/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Wavefunctions
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/noncollinear/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Noncollinear
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/pseudopotentials/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pseudopotentials
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/mbt/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Many-Body Perturbation Theory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/bse/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bethe-Salpeter
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/documents/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Documents
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/acknowledgments/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Acknowledgments
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/bibliography/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bibliography
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../theory/acronyms/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Acronyms
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Developers
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Developers
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/git_and_gitlab/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Git and Gitlab
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/developers_howto/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    HowTo for developers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/abimkdocs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Abimkdocs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/markdown/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Markdown
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_5" >
        
          
          <label class="md-nav__link" for="__nav_7_5" id="__nav_7_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Test Suite
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_5">
            <span class="md-nav__icon md-icon"></span>
            Test Suite
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/testsuite/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Test list
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/testsuite_howto/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Test Suite Howto
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/new_testsuite/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    YAML-based test suite Howto
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/yaml_tools_api/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    YAML-based test suite Python tools and API
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/XgType/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    XgType
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_7" >
        
          
          <label class="md-nav__link" for="__nav_7_7" id="__nav_7_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Pseudos
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_7">
            <span class="md-nav__icon md-icon"></span>
            Pseudos
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/psp1_info/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    psp1
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/psp3_info/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    psp3
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/psp45_info/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    psp45
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/psp5spinorbit_info/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    psp5
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/psp6_info/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    psp6
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/psp8_info/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    psp8
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/codestats/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code stats
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/autoconf_examples/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Autoconf_examples
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../developers/testlink/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Test Page for linkcheck
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    About
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            About
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/release-notes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Release Notes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/contributing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Contributing
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/history/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    History
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/license/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    License
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adpimd" class="md-nav__link">
    <span class="md-ellipsis">
      adpimd
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#adpimd_gamma" class="md-nav__link">
    <span class="md-ellipsis">
      adpimd_gamma
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#amu" class="md-nav__link">
    <span class="md-ellipsis">
      amu
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bmass" class="md-nav__link">
    <span class="md-ellipsis">
      bmass
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#chkdilatmx" class="md-nav__link">
    <span class="md-ellipsis">
      chkdilatmx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cineb_start" class="md-nav__link">
    <span class="md-ellipsis">
      cineb_start
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#delayperm" class="md-nav__link">
    <span class="md-ellipsis">
      delayperm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diismemory" class="md-nav__link">
    <span class="md-ellipsis">
      diismemory
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dilatmx" class="md-nav__link">
    <span class="md-ellipsis">
      dilatmx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dtion" class="md-nav__link">
    <span class="md-ellipsis">
      dtion
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dynimage" class="md-nav__link">
    <span class="md-ellipsis">
      dynimage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ecutsm" class="md-nav__link">
    <span class="md-ellipsis">
      ecutsm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#friction" class="md-nav__link">
    <span class="md-ellipsis">
      friction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frictionbar" class="md-nav__link">
    <span class="md-ellipsis">
      frictionbar
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fxcartfactor" class="md-nav__link">
    <span class="md-ellipsis">
      fxcartfactor
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_algor" class="md-nav__link">
    <span class="md-ellipsis">
      ga_algor
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_fitness" class="md-nav__link">
    <span class="md-ellipsis">
      ga_fitness
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_n_rules" class="md-nav__link">
    <span class="md-ellipsis">
      ga_n_rules
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_opt_percent" class="md-nav__link">
    <span class="md-ellipsis">
      ga_opt_percent
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_rules" class="md-nav__link">
    <span class="md-ellipsis">
      ga_rules
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#geoopt" class="md-nav__link">
    <span class="md-ellipsis">
      geoopt
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getcell" class="md-nav__link">
    <span class="md-ellipsis">
      getcell
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getvel" class="md-nav__link">
    <span class="md-ellipsis">
      getvel
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxcart" class="md-nav__link">
    <span class="md-ellipsis">
      getxcart
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxred" class="md-nav__link">
    <span class="md-ellipsis">
      getxred
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecon" class="md-nav__link">
    <span class="md-ellipsis">
      goprecon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecprm" class="md-nav__link">
    <span class="md-ellipsis">
      goprecprm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hmcsst" class="md-nav__link">
    <span class="md-ellipsis">
      hmcsst
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hmctt" class="md-nav__link">
    <span class="md-ellipsis">
      hmctt
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatcon" class="md-nav__link">
    <span class="md-ellipsis">
      iatcon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfix" class="md-nav__link">
    <span class="md-ellipsis">
      iatfix
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixx" class="md-nav__link">
    <span class="md-ellipsis">
      iatfixx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixy" class="md-nav__link">
    <span class="md-ellipsis">
      iatfixy
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixz" class="md-nav__link">
    <span class="md-ellipsis">
      iatfixz
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#imgmov" class="md-nav__link">
    <span class="md-ellipsis">
      imgmov
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#imgwfstor" class="md-nav__link">
    <span class="md-ellipsis">
      imgwfstor
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#istatimg" class="md-nav__link">
    <span class="md-ellipsis">
      istatimg
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdtemp" class="md-nav__link">
    <span class="md-ellipsis">
      mdtemp
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdwall" class="md-nav__link">
    <span class="md-ellipsis">
      mdwall
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_mxstep" class="md-nav__link">
    <span class="md-ellipsis">
      mep_mxstep
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_solver" class="md-nav__link">
    <span class="md-ellipsis">
      mep_solver
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mixesimgf" class="md-nav__link">
    <span class="md-ellipsis">
      mixesimgf
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#moldyn" class="md-nav__link">
    <span class="md-ellipsis">
      moldyn
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natcon" class="md-nav__link">
    <span class="md-ellipsis">
      natcon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfix" class="md-nav__link">
    <span class="md-ellipsis">
      natfix
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixx" class="md-nav__link">
    <span class="md-ellipsis">
      natfixx
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixy" class="md-nav__link">
    <span class="md-ellipsis">
      natfixy
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixz" class="md-nav__link">
    <span class="md-ellipsis">
      natfixz
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nconeq" class="md-nav__link">
    <span class="md-ellipsis">
      nconeq
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_algo" class="md-nav__link">
    <span class="md-ellipsis">
      neb_algo
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_cell_algo" class="md-nav__link">
    <span class="md-ellipsis">
      neb_cell_algo
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_spring" class="md-nav__link">
    <span class="md-ellipsis">
      neb_spring
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nimage" class="md-nav__link">
    <span class="md-ellipsis">
      nimage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nnos" class="md-nav__link">
    <span class="md-ellipsis">
      nnos
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#noseinert" class="md-nav__link">
    <span class="md-ellipsis">
      noseinert
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntime" class="md-nav__link">
    <span class="md-ellipsis">
      ntime
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntimimage" class="md-nav__link">
    <span class="md-ellipsis">
      ntimimage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optcell" class="md-nav__link">
    <span class="md-ellipsis">
      optcell
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimass" class="md-nav__link">
    <span class="md-ellipsis">
      pimass
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimd_constraint" class="md-nav__link">
    <span class="md-ellipsis">
      pimd_constraint
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pitransform" class="md-nav__link">
    <span class="md-ellipsis">
      pitransform
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prtatlist" class="md-nav__link">
    <span class="md-ellipsis">
      prtatlist
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qmass" class="md-nav__link">
    <span class="md-ellipsis">
      qmass
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#random_atpos" class="md-nav__link">
    <span class="md-ellipsis">
      random_atpos
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#restartxf" class="md-nav__link">
    <span class="md-ellipsis">
      restartxf
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#signperm" class="md-nav__link">
    <span class="md-ellipsis">
      signperm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strfact" class="md-nav__link">
    <span class="md-ellipsis">
      strfact
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#string_algo" class="md-nav__link">
    <span class="md-ellipsis">
      string_algo
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strprecon" class="md-nav__link">
    <span class="md-ellipsis">
      strprecon
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strtarget" class="md-nav__link">
    <span class="md-ellipsis">
      strtarget
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolimg" class="md-nav__link">
    <span class="md-ellipsis">
      tolimg
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxde" class="md-nav__link">
    <span class="md-ellipsis">
      tolmxde
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxf" class="md-nav__link">
    <span class="md-ellipsis">
      tolmxf
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel" class="md-nav__link">
    <span class="md-ellipsis">
      vel
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel_cell" class="md-nav__link">
    <span class="md-ellipsis">
      vel_cell
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vis" class="md-nav__link">
    <span class="md-ellipsis">
      vis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#wtatcon" class="md-nav__link">
    <span class="md-ellipsis">
      wtatcon
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                

                  

  
  


<!--
This file is automatically generated by mksite.py. All changes will be lost.
Change the input yaml files or the python code
-->
<h1 id="rlx-input-variables">rlx input variables<a class="headerlink" href="#rlx-input-variables" title="Permanent link">&para;</a></h1>
<p>This document lists and provides the description of the name (keywords) of the
rlx input variables to be used in the input file for the abinit executable.</p>
<h2 id="adpimd"><strong>adpimd</strong><a class="headerlink" href="#adpimd" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> ADiabatic Path-Integral Molecular Dynamics<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 9 or <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 13<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
</ul>
</details>
<p>Controls whether adiabatic Path-Integral Molecular Dynamics is performed or not.
The corresponding adiabaticity parameter is given by <a class="wikilink codevar-wikilink" href="/variables/rlx#adpimd_gamma">adpimd_gamma</a>.</p>
<p>If equal to 0, no adiabatic Path-Integral Molecular Dynamics (standard PIMD) is performed.
If equal to 1, adiabatic Path-Integral Molecular Dynamics is activated.
Only relevant with <a class="wikilink codevar-wikilink" href="/variables/rlx#pitransform">pitransform</a> = 1 (normal mode transformation). In that case,</p>
<ul>
<li>the mass associated with the zero-frequency mode is the true mass <a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>,</li>
<li>the mass associated to the other higher frequency modes of the polymer
chains is equal to the normal mode mass divided by <a class="wikilink codevar-wikilink" href="/variables/rlx#adpimd_gamma">adpimd_gamma</a> (adiabaticity parameter),</li>
<li>the equation of motion on the zero-frequency mode is not thermostated.</li>
</ul>
<p>NOT YET USABLE</p>
<h2 id="adpimd_gamma"><strong>adpimd_gamma</strong><a class="headerlink" href="#adpimd_gamma" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> ADiabatic Path-Integral Molecular Dynamics: GAMMA factor<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#adpimd">adpimd</a> == 1 and <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> in [9,13]<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
</ul>
</details>
<p>Adiabaticity parameter to be used in adiabatic Path-Integral Molecular Dynamics.
NOT YET USABLE</p>
<h2 id="amu"><strong>amu</strong><a class="headerlink" href="#amu" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Atomic Mass Units<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to compute phonon frequencies and modes, IR and Raman spectra, Born effective charges, IR reflectivity ..." data-toggle="popover" href="/topics/Phonons">topic_Phonons</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to specify the types of atoms that form the system" data-toggle="popover" href="/topics/AtomTypes">topic_AtomTypes</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform some artificial modifications of the physics" data-toggle="popover" href="/topics/Artificial">topic_Artificial</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/basic#ntypat">ntypat</a>)<br />
<em>Default value:</em> None<br />
<em>Comment:</em> Defaults provided by a database of atomic masses.<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [85/1314] in all abinit tests, [1/178] in abinit tutorials</summary>
<ul>
<li>bigdft:  <a class="wikilink abifile-wikilink" data-tippy-content="H in isolated boundary conditions (plane waves, PAW).
Tests free boundary conditions Poisson solver for PAW." data-toggle="popover" href="/tests/bigdft/Input/t23.abi" target="_blank">t23.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H in a box (PAW). PAW projectors are fitted to Gaussians." data-toggle="popover" href="/tests/bigdft/Input/t31.abi" target="_blank">t31.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H in a box (PAW). PAW projectors are fitted to Gaussians." data-toggle="popover" href="/tests/bigdft/Input/t32.abi" target="_blank">t32.abi</a> &hellip;</li>
<li>bigdft_paral:  <a class="wikilink abifile-wikilink" data-tippy-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0)
Test: DIIS, potential/density mixing and parallelism" data-toggle="popover" href="/tests/bigdft_paral/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0)
Test: DIIS, potential/density mixing and parallelism" data-toggle="popover" href="/tests/bigdft_paral/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>built-in:  <a class="wikilink abifile-wikilink" data-tippy-content="GaAs crystal with few bands." data-toggle="popover" href="/tests/built-in/Input/testin_wannier90.abi" target="_blank">testin_wannier90.abi</a> &hellip;</li>
<li>etsf_io:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si2 molecule, static, spin-polarized.
Same system as test 49 of v2, except that nctime is non-zero.
Test ionmov=6 (Verlet) as well as ionmov=7." data-toggle="popover" href="/tests/etsf_io/Input/t21.abi" target="_blank">t21.abi</a> &hellip;</li>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="
21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-toggle="popover" href="/tests/fast/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
move with natfix.
Output also GEO files" data-toggle="popover" href="/tests/fast/Input/t29.abi" target="_blank">t29.abi</a> &hellip;</li>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
GaAs phonons calculation - norm-conserving psp
Test the automatic detection of the GPU and the automatic setting of
gpu_option keyword according to the requested calculation.
Ground-states calculations should be run with gpu_option activated
(except if set to 0 in input file) while response fonction calculations
should be run with gpu_option deactivated (at present : ABINIT v9.10)." data-toggle="popover" href="/tests/gpu/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Tests series for the mGGA implementation (from 31 to 33)
Silicon bulk. Two datasets are run. One (classic) LDA calculation with ixc 7.
One with a fake native mGGA (fake1, i.e. ixc 31) and using effmass_free=1.01.
Both calculation should give the same total energy (change in kinetic energy due to
effmass_free is compensated by the fake mGGA using kinetic energy density: taur).
(Two other datasets are also run for spin polarized case)" data-toggle="popover" href="/tests/gpu_omp/Input/t36.abi" target="_blank">t36.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="GaAs linear response with NC. Test the parallelisation on perturbations" data-toggle="popover" href="/tests/mpiio/Input/t69.abi" target="_blank">t69.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="GaAs linear response with NC. Test the parallelisation on perturbations" data-toggle="popover" href="/tests/mpiio/Input/t69.abi" target="_blank">t69.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Gold, 8 atom supercell, with PAW.
Test lotf_nitex, lotf_classic, lotf_version, lotf_nneigx.
These variables, concerning LOTF method, are taken into account only if the
enable_lotf=&quot;yes&quot; is used in configuration." data-toggle="popover" href="/tests/paral/Input/t41.abi" target="_blank">t41.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="FCC Al metallic; 10 phonon at 1/4 1/8 1/8" data-toggle="popover" href="/tests/paral/Input/t53.abi" target="_blank">t53.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="FCC Al metallic; 10 phonon at 1/4 1/8 1/8" data-toggle="popover" href="/tests/paral/Input/t53.abi" target="_blank">t53.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
GaAs, zinc-blende structure.
Compute Berry phase along 0 0 1, using a 4x4x10 sampling." data-toggle="popover" href="/tests/seq/Input/tsv2_82.abi" target="_blank">tsv2_82.abi</a> &hellip;</li>
<li>tutoplugs:  <a class="wikilink abifile-wikilink" data-tippy-content="
Gallium Arsenide, zinc-blende. 8 bands. prtwant=2
Should still be completed by the call to wannier90, and subsequent fldiff use...
example for post-processing mode of ABINIT-Wannier90 with wfk_task" data-toggle="popover" href="/tests/tutoplugs/Input/tw90_7.abi" target="_blank">tw90_7.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Linear chain of Ge2 molecules (2 atoms per unit cell),
using a local (Starkloff-Joannopoulos) pseudopotential,
and no exchange-correlation (ixc=0). Uses 4 k-points.
Computation of the second derivative of the total energy
with respect to a atomic displacement along the chain,
with q(0 0 0) wavevector.
(see test.ge.xc0 of RESPFN)" data-toggle="popover" href="/tests/v2/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Linear chain of Ge2 molecules (2 atoms per unit cell),
using a local (Starkloff-Joannopoulos) pseudopotential,
with exchange-correlation (ixc=3). Uses 2 k-points.
Computation of the second derivative of the total energy
with respect to a atomic displacement along the chain,
with q(0 0 0) wavevector.
Same as test 1, except ixc and the k-points, and
except that it doesn&#x27;t uses multi-data mode, so GS and RF
calculations are separated.
(see test.ge.xc5 of RESPFN)" data-toggle="popover" href="/tests/v2/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Linear chain of Si2 molecules (2 atoms per unit cell),
using a separable pseudopotential, with ixc=3.
Computation of the second derivative of the total energy
with respect to a atomic displacement perpendicular to
the chain, with q(0 0 1/2) wavevector.
(see test.si.chain of RESPFN)" data-toggle="popover" href="/tests/v2/Input/t04.abi" target="_blank">t04.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
Linear chain of Ge2 molecules (2 atoms per unit cell),
using a local (Starkloff-Joannopoulos) pseudopotential,
and no exchange-correlation (ixc=0). Uses 4 k-points.
Computation of the second derivative of the total energy
with respect to a atomic displacement along the chain,
with q(0 0 0) wavevector. Similar to test v2 #1, except
that it uses dilatmx=1.1 ." data-toggle="popover" href="/tests/v3/Input/t06.abi" target="_blank">t06.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule, using a local pseudopotential, and GGA.
Similar to test 33 of v2.
Use the x-only PBE functional.
Compute the interatomic force constant for the
displacement of the second atom along x.
RF calculation of the 2DTE gives 189.10691 Ha
while finite differences gives :
with delta(xred)=0.0002, 189.10790 Ha
with delta(xred)=0.0001, 189.10703 Ha .
Combining the results gives 189.10675 Ha. The
agreement is quite good, and might likely be improved
by increasing tolvrs." data-toggle="popover" href="/tests/v3/Input/t07.abi" target="_blank">t07.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule, using a local pseudopotential, and GGA.
Similar to test 33 of v2, and preceeding test,
but use the full XC PBE functional.
Also, tolvrs is better.
Compute the interatomic force constant for the
displacement of the second atom along x.
RF calculation of the 2DTE gives 188.81162533 Ha
while finite differences gives :
with delta(xred)=0.0002, 188.81213804 Ha
with delta(xred)=0.0001, 188.81175352 Ha .
Combining the results gives 188.81162525 Ha." data-toggle="popover" href="/tests/v3/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98).
Compute responses at Gamma:
1) Ground state calculation, with a k-point sampling in the IBZ only;
2) Computation of the ddk response;
3) Computation of the dynamical matrix at Gamma, as well as the
Born effective charges, and the dielectric tensor ;
Perform five datasets:
* First dataset is the ground state calculation,
* Second dataset compute ddk perturbation
with the use of Legendre Polynomials for the nl operator,
* Third dataset compute phonon and elec. field perturbation
with the use of Legendre Polynomials for the nl operator,
* Fourth dataset compute ddk perturbation
with the use of spherical harmonics for the nl operator,
* Fifth dataset compute phonon and elec. field perturbation
with the use of spherical harmonics for the nl operator.
Datasets 2 and 3 must give the same results as 4 and 5." data-toggle="popover" href="/tests/v4/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
GaAs, zinc-blende structure. Generate the data for optic (linear and non-linear coefficients).
to be analysed in the test 57." data-toggle="popover" href="/tests/v4/Input/t56.abi" target="_blank">t56.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test to prepare a conductivity calculation (test v5#t10)
Ground state calculation for 4 Al atoms
The number of bands is unrealistic. This test produces files that are inputs for test 10.
It tests the following parts of the code:
optic_paw, optic_paw_core
Note that, in addition to the psp file, the core wave functions file is needed." data-toggle="popover" href="/tests/v5/Input/t09.abi" target="_blank">t09.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule, using a local pseudopotential, and Wu-Cohen GGA.
Similar to test v3#08
Compute the interatomic force constant for the
displacement of the second atom along x.
RF calculation of the 2DTE gives 189.49410411 Ha
while finite differences gives :
with delta(xred)=0.0002, 189.4945677  Ha
with delta(xred)=0.0001, 189.49422003  Ha .
Combining the results gives 189.4941041 Ha. Perfect !" data-toggle="popover" href="/tests/v5/Input/t29.abi" target="_blank">t29.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule, using a local pseudopotential,
and the Wu-Cohen GGA.
Similar to Test v3#48
Compute the total energy and force as a
displacement of the second atom along x.
Direct calculation of the gradient with respect
to reduced coordinates gives -3.469435369425Ha
while finite differences gives :
with delta(xred)=0.0002, -3.4694479488 Ha
with delta(xred)=0.0001, -3.4694385143 Ha .
Combining the results gives -3.4694353695 Ha.
Excellent." data-toggle="popover" href="/tests/v5/Input/t48.abi" target="_blank">t48.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Tests series for the mGGA implementation (from 31 to 33)
Silicon bulk. Two datasets are run. One (classic) LDA calculation with ixc 7.
One with a fake native mGGA (fake1, i.e. ixc 31) and using effmass_free=1.01.
Both calculation should give the same total energy (change in kinetic energy due to
effmass_free is compensated by the fake mGGA using kinetic energy density: taur).
(Two other datasets are also run for spin polarized case)" data-toggle="popover" href="/tests/v6/Input/t31.abi" target="_blank">t31.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the mGGA implementation. Silicon bulk. Two datasets are run.
One with a fake native mGGA (fake2, i.e. ixc 32) involving laplacian of the density.
One with a fake native mGGA (fake3, i.e. ixc 33) involving gradient of the density.
Both calculation should give the same total and exchange-correlation energies.
(Two other datasets are also run for spin polarized case)" data-toggle="popover" href="/tests/v6/Input/t32.abi" target="_blank">t32.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Boron, 32 atoms - Thomas-Fermi method (without orbitals)
Test the temperature-dependent XC functional (ixc=50)
Ichimaru et al Phys. Rep. 149, 91-205 (1987)
Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-toggle="popover" href="/tests/v7/Input/t11.abi" target="_blank">t11.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test to prepare a SOC conductivity calculation (test v9#t101)
Ground state calculation for 4 Al atoms
The number of bands is unrealistic. This test produces files that are inputs for test 101.
It tests the following parts of the code:
optic_paw, optic_paw_core, optic_paw_soc
Note that, in addition to the psp file, the spinor core wave functions file is needed." data-toggle="popover" href="/tests/v9/Input/t100.abi" target="_blank">t100.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test to prepare a SOC conductivity calculation (test v9#t102)
Ground state calculation for 4 Al atoms
The number of bands is unrealistic. This test produces files that are inputs for test 103.
It tests the following parts of the code:
optic_paw_core
Note that, in addition to the psp file, the spinor core wave functions file is needed." data-toggle="popover" href="/tests/v9/Input/t102.abi" target="_blank">t102.abi</a> &hellip;</li>
<li>wannier90:  <a class="wikilink abifile-wikilink" data-tippy-content="GaAs crystal with few bands. Very quick built-in test, to check that Wannier90 is installed." data-toggle="popover" href="/tests/wannier90/Input/t00.abi" target="_blank">t00.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Gallium Arsenide, zinc-blende. 8 bands. prtwant=2
Should still be completed by the call to wannier90, and subsequent fldiff use..." data-toggle="popover" href="/tests/wannier90/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="The same as in 01 but using PAW." data-toggle="popover" href="/tests/wannier90/Input/t02.abi" target="_blank">t02.abi</a> &hellip;</li>
</ul>
</details>
<p>Gives the masses in atomic mass units for each kind of atom in the input cell. These
masses are used in performing molecular dynamical atomic motion if
<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;viscous&rdquo; or &ldquo;quenched&rdquo; (corresponding to <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 1 or 7) or if <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> /= &ldquo;none&rdquo;.
They are also used in phonon calculations during the diagonalization of the dynamical matrix.
Note that one may set all masses to 1 for certain cases in which merely structural relaxation is desired and not
actual molecular dynamics.</p>
<p>Using the recommended values of <a class="wikilink citation-wikilink" data-tippy-content="Atomic Weights of the Elements" data-toggle="popover" href="/theory/bibliography#martin1987">[Martin1987]</a>, 1 atomic mass unit = 1.6605402e-27 kg. In this
unit the mass of Carbon 12 is exactly 12.</p>
<p>A database of atomic masses is provided which provides the default values. Note that the
default database uses mixed isotope masses (for Carbon the natural occurrence
of Carbon 13 is taken into account). The values are those recommended by the
commission on Atomic Weights and Isotopic Abundances, Inorganic Chemistry
Division, IUPAC <a class="wikilink citation-wikilink" data-tippy-content="Atomic Weights of the Elements" data-toggle="popover" href="/theory/bibliography#martin1987">[Martin1987]</a>. For Tc, Pm, Po to
Ac, Pa and beyond U, none of the isotopes have a half-life greater than 3.0d10
years, and the values provided in the database do not come from that source.</p>
<p>For alchemical pseudoatoms, the masses of the constituents atoms are mixed,
according to the alchemical mixing coefficients <a class="wikilink codevar-wikilink" href="/variables/gstate#mixalch">mixalch</a></p>
<p>In most cases, the use of <em>amu</em> will be as a static (non-evolving) variable.
However, the possibility to have different values of <em>amu</em> for different
images has been coded. A population of cells with different atomic
characteristics can thus be considered, and can be made to evolve, e.g. with a
genetic algorithm (not coded in v7.0.0 though).</p>
<h2 id="bmass"><strong>bmass</strong><a class="headerlink" href="#bmass" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Barostat MASS<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 10<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a></li>
</ul>
</details>
<p>bmass is the mass of the barostat when <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;npt_martyna&rdquo;, &ldquo;nst_martyna&rdquo; or &ldquo;npt_langevin&rdquo; (corresponding to <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 13 and 16).</p>
<h2 id="chkdilatmx"><strong>chkdilatmx</strong><a class="headerlink" href="#chkdilatmx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> CHecK DILATMX<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [9/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Test dilatmx. Increase it, the number of planewaves increases, but
the total energy stays practically the same. The small fluctuations
(-8.7967199088907,
-8.7967198524822,
-8.7967198520954 ) come from varying the FFT grid size, with
effect on the xc functional. It has been checked that using exactly the
same FFT grid for the three cases gives the same energy.
It has been checked also that larger fluctuations arise with intxc=0." data-toggle="popover" href="/tests/v1/Input/t77.abi" target="_blank">t77.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Generate first-order responses for GaAs (zinc-blende).
Compute responses at Gamma, in the presence of a non-zero ecutsm and dilatmx.
1) ground state calculation, with a k-point sampling in the irreducible
Brillouin zone only;
2) computation of the ddk response, in preparation to the electric field
response ;
3) computation of the dynamical matrix at Gamma, as well as the
Born effective charges, and the dielectric tensor ;" data-toggle="popover" href="/tests/v2/Input/t98.abi" target="_blank">t98.abi</a></li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
Magnesium HCP. Optimize the cell volume and shape using different constraints.
Checks whether the stresses are symmetric, and whether the optimisation algorithm does
not break the symmetry.
Also checks the possibility to disable the stop after violation of the condition related to dilatmx.
This is done in dataset 6. The optimization delivers acell values slightly different from those of dataset 2." data-toggle="popover" href="/tests/v3/Input/t42.abi" target="_blank">t42.abi</a></li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98).
Compute responses at Gamma:
1) Ground state calculation, with a k-point sampling in the IBZ only;
2) Computation of the ddk response;
3) Computation of the dynamical matrix at Gamma, as well as the
Born effective charges, and the dielectric tensor ;
Perform five datasets:
* First dataset is the ground state calculation,
* Second dataset compute ddk perturbation
with the use of Legendre Polynomials for the nl operator,
* Third dataset compute phonon and elec. field perturbation
with the use of Legendre Polynomials for the nl operator,
* Fourth dataset compute ddk perturbation
with the use of spherical harmonics for the nl operator,
* Fifth dataset compute phonon and elec. field perturbation
with the use of spherical harmonics for the nl operator.
Datasets 2 and 3 must give the same results as 4 and 5." data-toggle="popover" href="/tests/v4/Input/t02.abi" target="_blank">t02.abi</a></li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrure of Lithium-Boron, in a face-centered monoclinic cell.
Test the recognition of the space group (C2/c (# 15)). Was lacking a case before Dec 2011." data-toggle="popover" href="/tests/v6/Input/t30.abi" target="_blank">t30.abi</a></li>
</ul>
</details>
<p>If 0, the code will not stop execution if the threshold of <a class="wikilink codevar-wikilink" href="/variables/rlx#dilatmx">dilatmx</a> is exceeded,
it will simply issue a warning. There will be no rescaling. If 1, after tentative
rescaling as described in <a class="wikilink codevar-wikilink" href="/variables/rlx#dilatmx">dilatmx</a>, the code will stop execution.
Also, the use of  <strong>chkdilatmx</strong>  = 0 allows one to set <a class="wikilink codevar-wikilink" href="/variables/rlx#dilatmx">dilatmx</a> to a larger value than 1.15,
otherwise forbidden as being a waste of CPU and memory.</p>
<p>So, when using  <strong>chkdilatmx</strong> =0, the relaxed lattice parameters might not be accurate, but will simply better
than the starting ones.</p>
<p><strong>chkdilatmx</strong> =0 is useful when the starting geometry is likely very inaccurate. However, if the user is in search
of an accurate geometry estimation, then a first determination of the (better but inaccurate) geometry with  <strong>chkdilatmx</strong> =0
should be followed by a more accurate second run from the better geometry with  <strong>chkdilatmx</strong> =1 and <a class="wikilink codevar-wikilink" href="/variables/rlx#dilatmx">dilatmx</a> slightly larger than 1,
(possibly 1.05).</p>
<h2 id="cineb_start"><strong>cineb_start</strong><a class="headerlink" href="#cineb_start" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Climbing-Image Nudged Elastic Band: STARTing iteration<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 7<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 5 and <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_algo">neb_algo</a> == 2<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
</ul>
</details>
<p>Gives the index of the first CI-NEB iteration.
The CI-NEB method constitutes a small modification to the NEB method allowing
a rigorous convergence to the saddle point. As the image with the highest
energy has to be identified, the calculation begins with several iterations of
the standard NEB algorithm. The effective CI-NEB begins at the  <strong>cineb_start</strong> 
iteration. See <a class="wikilink citation-wikilink" data-tippy-content="A climbing image nudged elastic band method for finding saddle points and minimum energy paths" data-toggle="popover" href="/theory/bibliography#henkelman2000a">[Henkelman2000a]</a> for additional details of this method.</p>
<h2 id="delayperm"><strong>delayperm</strong><a class="headerlink" href="#delayperm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> DELAY between trials to PERMUTE atoms<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a></li>
</ul>
</details>
<p>Delay (number of time steps) between trials to permute two atoms, in view of
accelerated search of minima. Still in development.</p>
<p>See the routine moldyn.F90 and <a class="wikilink codevar-wikilink" href="/variables/rlx#signperm">signperm</a> for additional information.</p>
<p>When  <strong>delayperm</strong>  is zero, there are no permutation trials.</p>
<h2 id="diismemory"><strong>diismemory</strong><a class="headerlink" href="#diismemory" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Direct Inversion in the Iterative Subspace MEMORY<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 8<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [0/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
</details>
<p>Gives the maximum number of &ldquo;time&rdquo; steps for which the forces and stresses are
stored, and taken into account in the DIIS algorithm (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 20) to find
zero-force and stress configurations.</p>
<h2 id="dilatmx"><strong>dilatmx</strong><a class="headerlink" href="#dilatmx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> lattice DILATation: MaXimal value<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1.0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [76/1314] in all abinit tests, [15/178] in abinit tutorials</summary>
<ul>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu/Input/t04.abi" target="_blank">t04.abi</a> &hellip;</li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu_omp/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/mpiio/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="Compare cprj_in_memory=1 with cprj_in_memory=0 (lobpcg2 and chebfi2). PAW,istwfk&gt;1,dilatmx&gt;1,npband=4" data-toggle="popover" href="/tests/paral/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Compare cprj_in_memory=1 with cprj_in_memory=0 (lobpcg2 and chebfi2). PAW,istwfk&gt;1,dilatmx&gt;1,npband=4" data-toggle="popover" href="/tests/paral/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium,
to compare with the PSML results of t07 and t11" data-toggle="popover" href="/tests/psml/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t07.abi" target="_blank">t07.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=14 (finite reduced electric field calculation,relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_121.abi" target="_blank">tsv6_121.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=16 (finite reduced electric displacement field calculation,
relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_122.abi" target="_blank">tsv6_122.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=4 (finite electric field calculation, relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_123.abi" target="_blank">tsv6_123.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Finite difference calculation of the Born effective charges of AlP" data-toggle="popover" href="/tests/tutorespfn/Input/tpolarization_1.abi" target="_blank">tpolarization_1.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline silicon : computation of the optimal lattice parameter
Convergence with respect to the number of k points." data-toggle="popover" href="/tests/tutorial/Input/tbase3_4.abi" target="_blank">tbase3_4.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline aluminum : optimization of the lattice parameter
at fixed number of k points and broadening." data-toggle="popover" href="/tests/tutorial/Input/tbase4_1.abi" target="_blank">tbase4_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline aluminum : optimization of the lattice parameter

Convergence with respect to k points." data-toggle="popover" href="/tests/tutorial/Input/tbase4_2.abi" target="_blank">tbase4_2.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Test dilatmx. Increase it, the number of planewaves increases, but
the total energy stays practically the same. The small fluctuations
(-8.7967199088907,
-8.7967198524822,
-8.7967198520954 ) come from varying the FFT grid size, with
effect on the xc functional. It has been checked that using exactly the
same FFT grid for the three cases gives the same energy.
It has been checked also that larger fluctuations arise with intxc=0." data-toggle="popover" href="/tests/v1/Input/t77.abi" target="_blank">t77.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Optimization of unit cell volume (optcell=1).
Use 2 datasets, and the variable getcell." data-toggle="popover" href="/tests/v1/Input/t78.abi" target="_blank">t78.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a cell of variable size along the H2 chain =&gt; H polymer.
Optimization of cell size and atomic distances (optcell=4)." data-toggle="popover" href="/tests/v1/Input/t79.abi" target="_blank">t79.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Compare cprj_in_memory=1 with cprj_in_memory=0 (lobpcg2 and chebfi2). PAW,istwfk&gt;1,dilatmx&gt;1" data-toggle="popover" href="/tests/v10/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Compare cprj_in_memory=1 with cprj_in_memory=0 (lobpcg2 and chebfi2). PAW,istwfk&gt;1,dilatmx&gt;1" data-toggle="popover" href="/tests/v10/Input/t25.abi" target="_blank">t25.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Arsenic in rhombohedral structure.
Optimization of unit cell size and shape.
With 4x4x4 k point grid, ecut 3.0, occopt 4 and tsmear 0.06d0,
one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277
These input parameters are too small, but still give a realistic
geometry, since with the much better parameters
12x12x12 k point grid, ecut 12.0, occopt 4 and tsmear 0.04d0,
one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296, while
the experimental values are
acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-toggle="popover" href="/tests/v2/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Li, simple cubic, to test accurately RF in the metallic case.
With non-linear XC core correction and non-zero ecutsm and dilatmx.
Similar to test 92, except ecutsm and dilatmx, and no q(1/4 0 0).
The total energy of the 1-atom cell is -.29337493523588 Ha
The 2DTE with respect to a q(0 0 0) phonon is 5.5619d-8 .
The 2DTE with respect to a transverse q(1/2 0 0) phonon
is -0.12068188927 Ha." data-toggle="popover" href="/tests/v2/Input/t96.abi" target="_blank">t96.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Li, simple cubic, frozen-phonon calculations corresponding
to test 96 (with non-zero ecutsm and dilatmx).
The total energy of the double cell is -0.58674987047187 Ha,
in excellent agreement with the result of test 96.
The frozen-phonon (from energy) calculation of the 2DTE for the q(1/2 0 0)
case gives -0.1206807 Ha .
The frozen-phonon (from forces) calculation of the 2DTE for the q(1/2 0 0)
case gives -0.1206794 Ha . Both are in rather good agreement with
the result of test 96." data-toggle="popover" href="/tests/v2/Input/t97.abi" target="_blank">t97.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
Linear chain of Ge2 molecules (2 atoms per unit cell),
using a local (Starkloff-Joannopoulos) pseudopotential,
and no exchange-correlation (ixc=0). Uses 4 k-points.
Computation of the second derivative of the total energy
with respect to a atomic displacement along the chain,
with q(0 0 0) wavevector. Similar to test v2 #1, except
that it uses dilatmx=1.1 ." data-toggle="popover" href="/tests/v3/Input/t06.abi" target="_blank">t06.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Magnesium HCP. Optimize the cell volume and shape using different constraints.
Checks whether the stresses are symmetric, and whether the optimisation algorithm does
not break the symmetry.
Also checks the possibility to disable the stop after violation of the condition related to dilatmx.
This is done in dataset 6. The optimization delivers acell values slightly different from those of dataset 2." data-toggle="popover" href="/tests/v3/Input/t42.abi" target="_blank">t42.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98).
Compute responses at Gamma:
1) Ground state calculation, with a k-point sampling in the IBZ only;
2) Computation of the ddk response;
3) Computation of the dynamical matrix at Gamma, as well as the
Born effective charges, and the dielectric tensor ;
Perform five datasets:
* First dataset is the ground state calculation,
* Second dataset compute ddk perturbation
with the use of Legendre Polynomials for the nl operator,
* Third dataset compute phonon and elec. field perturbation
with the use of Legendre Polynomials for the nl operator,
* Fourth dataset compute ddk perturbation
with the use of spherical harmonics for the nl operator,
* Fifth dataset compute phonon and elec. field perturbation
with the use of spherical harmonics for the nl operator.
Datasets 2 and 3 must give the same results as 4 and 5." data-toggle="popover" href="/tests/v4/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the strain perturbation with GGA  and spin polarization
This is the same as t79, bcc Fe,
but with the PBE GGA.  This calculation is restricted to one
unaxial strain, which adequately tests the polarization-dependent
parts of the strain GGA code." data-toggle="popover" href="/tests/v4/Input/t59.abi" target="_blank">t59.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the strain perturbation with spin polarization and metallic
occupation of states
Fe in the bcc structure.  This tests both the introduction of spin
polarization into strain response function calculations and an
improved treatment of the first-order fermi energy for Q=0 metallic
calculations.  Similar data (with much larger cutoffs, tighter
convergence tolerances, and larger k sample) was used test the
elastic tensor in comparison with numerical derivatives of the
stress from GS calculations." data-toggle="popover" href="/tests/v4/Input/t79.abi" target="_blank">t79.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
White tin, body-centered tetragonal. This test checks that in an optcell 2 run, the symmetry
of the body-centered case is not broken. This test is run for two values of strprecon also,
to test the strprecon variable.
Coded and contributed by J. Zwanziger." data-toggle="popover" href="/tests/v5/Input/t77.abi" target="_blank">t77.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Make a simple geometry relaxation and validate the XML output (testing prtxml)." data-toggle="popover" href="/tests/v5/Input/t79.abi" target="_blank">t79.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrure of Lithium-Boron, in a face-centered monoclinic cell.
Test the recognition of the space group (C2/c (# 15)). Was lacking a case before Dec 2011." data-toggle="popover" href="/tests/v6/Input/t30.abi" target="_blank">t30.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density.
We test several improvements that are available when the NC file contains the PS valence charge.
More specifically, we test the initialization of the density from atomic quantities (initro.F90),
the correction to the forces due to the residuals and the extrapolation of the density
for structural relaxations (well, xred is fixed by symmetry).
We also test the treatment of the core charge in G-space in the case of GS calculations
By default, abinit handles XCCC in real space, here we use nc_xccc_gspace==1 to treat it
in G-space using the same approach as the one used in PAW." data-toggle="popover" href="/tests/v7/Input/t10.abi" target="_blank">t10.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Relaxation of FCC Argon crystal with DFT-D3 dispersion correction
Both the pair-wise term and the 3-body term are included
In this test the following properties are thus tested: contribution
to the energy and to the stress of DFT-D3 including the 3-body term" data-toggle="popover" href="/tests/v7/Input/t97.abi" target="_blank">t97.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Computation of the dynamical matrix at different q-vector including
vdW-DFT-D3 corrections with Becke-Jonhson damping (DFT-D3(BJ))
Only the pair-wise is computed (3-body term not implemented)
In this test the following properties are thus tested: contribution
to the energy and to the stress of DFT-D3(BJ) as well as
the contribution to the dynamical matrix" data-toggle="popover" href="/tests/v7/Input/t98.abi" target="_blank">t98.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 with in-plane strain.
Relaxing the third vector using optcell 6" data-toggle="popover" href="/tests/v9/Input/t104.abi" target="_blank">t104.abi</a> &hellip;</li>
<li>vdwxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-toggle="popover" href="/tests/vdwxc/Input/t10.abi" target="_blank">t10.abi</a> &hellip;</li>
</ul>
</details>
<p><strong>dilatmx</strong>  is an auxiliary variable used to book additional memory (see detailed description later) for possible
on-the-flight enlargement of the plane wave basis set, due to cell volume increase during geometry optimization by ABINIT.
Useful only when doing cell optimization, e.g. <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>/=0, usually with <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;bfgs&rdquo; or &ldquo;lbfgs&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> == 2 or 22).
Supposing that the starting (estimated) lattice parameters are already rather accurate (or likely to be too large),
then the recommended value of  <strong>dilatmx</strong>  is 1.05.
When you have no idea of evolution of the lattice parameters, and suspect that a large increase during geometry optimization is possible, while
you need an accurate estimation of the geometry, then make a first
run with <a class="wikilink codevar-wikilink" href="/variables/rlx#chkdilatmx">chkdilatmx</a>=0, producing an inaccurate, but much better estimation, followed by a second run using
the newly estimated geometry, with <a class="wikilink codevar-wikilink" href="/variables/rlx#chkdilatmx">chkdilatmx</a>=1 (the default) and  <strong>dilatmx</strong>  set to 1.05.
If you are not in search of an accurate estimation of the lattice parameters anyhow, then run with <a class="wikilink codevar-wikilink" href="/variables/rlx#chkdilatmx">chkdilatmx</a>=0 only once.</p>
<p>In the default mode (<a class="wikilink codevar-wikilink" href="/variables/rlx#chkdilatmx">chkdilatmx</a> = 1), when the  <strong>dilatmx</strong>  threshold is exceeded,
ABINIT will rescale uniformly the
tentative new primitive vectors to a value that leads at most to 90% of the
maximal allowed  <strong>dilatmx</strong>  deviation from 1. It will do this three times (to
prevent the geometry optimization algorithms to have taken a too large trial
step), but afterwards will stop and exit.</p>
<p>Setting <a class="wikilink codevar-wikilink" href="/variables/rlx#chkdilatmx">chkdilatmx</a> == 0 allows one to
book a larger planewave basis (if  <strong>dilatmx</strong>  is set to be bigger than 1), but will not rescale the tentative new primitive vectors
nor lead to an exit when the  <strong>dilatmx</strong>  threshold is exceeded.
The obtained optimized primitive vectors will not be exactly the ones corresponding to the planewave basis set
determined using <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> at the latter primitive vectors. Still, as an intermediate step in a geometry search
this might be sufficiently accurate. In such case,  <strong>dilatmx</strong>  might even be let at its default value 1.0.</p>
<p>Detailed explanation: The memory space for the planewave basis set is defined
by multiplying <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> by  <strong>dilatmx</strong>  squared (the result is an &ldquo;effective ecut&rdquo;, called
internally &ldquo;ecut_eff&rdquo;). Other uses of <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> are not modified when  <strong>dilatmx</strong>  &gt; 1.0.
Still, operations (like scalar products) are done by taking into account these fake (non-used) planewaves,
even if their coefficients are set to zero, thus slowing down the ABINIT execution.
Using  <strong>dilatmx</strong> &lt;1.0 is equivalent to changing <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> in all its uses. This
is allowed, although its meaning is no longer related to a maximal expected scaling.</p>
<p>Setting  <strong>dilatmx</strong>  to a large value leads to waste of CPU time and memory.
By default, ABINIT will not accept that you define  <strong>dilatmx</strong>  bigger than 1.15.
This behaviour will be overcome by using <a class="wikilink codevar-wikilink" href="/variables/rlx#chkdilatmx">chkdilatmx</a> == 0.
Supposing you think that the optimized <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a> values might be 5% larger
than your input values, use simply  <strong>dilatmx</strong>  1.05. This will lead to
an increase of the number of planewaves by a factor <span class="arithmatex">\((1.05)^3\)</span>, which is about <span class="arithmatex">\(1.158\)</span>, and a
corresponding increase in CPU time and memory.
It is possible to use  <strong>dilatmx</strong>  when <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> =0, but a value larger than
1.0 will be a waste.</p>
<h2 id="dtion"><strong>dtion</strong><a class="headerlink" href="#dtion" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Delta Time for IONs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [26/1314] in all abinit tests, [1/178] in abinit tutorials</summary>
<ul>
<li>etsf_io:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si2 molecule, static, spin-polarized.
Same system as test 49 of v2, except that nctime is non-zero.
Test ionmov=6 (Verlet) as well as ionmov=7." data-toggle="popover" href="/tests/etsf_io/Input/t21.abi" target="_blank">t21.abi</a> &hellip;</li>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="
21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-toggle="popover" href="/tests/fast/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
move with natfix.
Output also GEO files" data-toggle="popover" href="/tests/fast/Input/t29.abi" target="_blank">t29.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
He FCC solid in conventional cell (4 atoms).
Test the recursion algorithm (for high-temperature calculations).
Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t42.abi" target="_blank">t42.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Gold, 8 atom supercell, with PAW.
Test lotf_nitex, lotf_classic, lotf_version, lotf_nneigx.
These variables, concerning LOTF method, are taken into account only if the
enable_lotf=&quot;yes&quot; is used in configuration." data-toggle="popover" href="/tests/paral/Input/t41.abi" target="_blank">t41.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="Molecular dynamics for Al" data-toggle="popover" href="/tests/tutoparal/Input/tmoldyn_01.abi" target="_blank">tmoldyn_01.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Short MD traj to test restarxf -1" data-toggle="popover" href="/tests/v10/Input/t81.abi" target="_blank">t81.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si2 molecule, static, spin-polarized.
Same system as test 17 of fast, except lower ecut.
Test ionmov=6 (Verlet) as well as ionmov=7.
Cannot be executed in parallel:
--- !ERROR
message: |
the number of bands in the spin up case must be equal to
the number of bands in the spin down case.
This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3
Action : change nband, or use the sequential version of ABINIT.
src_file: chkinp.F90
src_line: 1181" data-toggle="popover" href="/tests/v2/Input/t49.abi" target="_blank">t49.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms)." data-toggle="popover" href="/tests/v4/Input/t97.abi" target="_blank">t97.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-toggle="popover" href="/tests/v5/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Simple test for double grid (fine/coarse) system, the cutting
radius and the initial fermi energy in recursion method
Corresponding variables: recgratio, recrcut,recefermi respectively
Corresponding code: 68_recursion
Other test: paral/tT,v5/t76" data-toggle="popover" href="/tests/v5/Input/t75.abi" target="_blank">t75.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Boron, 32 atoms - Thomas-Fermi method (without orbitals)
Test the temperature-dependent XC functional (ixc=50)
Ichimaru et al Phys. Rep. 149, 91-205 (1987)
Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-toggle="popover" href="/tests/v7/Input/t11.abi" target="_blank">t11.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Simple test of velocity verlet integrator" data-toggle="popover" href="/tests/v8/Input/t12.abi" target="_blank">t12.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Geometry optimization with FIRE (ionmov 15)" data-toggle="popover" href="/tests/v8/Input/t17.abi" target="_blank">t17.abi</a> &hellip;</li>
</ul>
</details>
<p>Used for controlling ion time steps. If <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> is set to &ldquo;viscous&rdquo;, &ldquo;quenched&rdquo; or &ldquo;fire&rdquo;
(<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> is set to 1, 7 or 15), or if <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> /= &ldquo;none&rdquo;, then
molecular dynamics is  used to update atomic positions in response to forces.
The parameter  <strong>dtion</strong>  is a time step in atomic units of time. (One atomic
time unit is 2.418884e-17 seconds, which is the value of Planck&rsquo;s constant in
hartree*sec.) In this case the atomic masses, in amu (given in array &rdquo; <a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>
&ldquo;), are used in Newton&rsquo;s equation and the viscosity (for <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;viscous&rdquo;) and
number of time steps are provided to the code using input variables &ldquo;<a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a>&rdquo;
and &ldquo;<a class="wikilink codevar-wikilink" href="/variables/rlx#ntime">ntime</a>&rdquo;. The code actually converts from masses in amu to masses in
atomic units (in units of electron masses) but the user enters masses in
<a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>. (The conversion from amu to atomic units (electron masses) is
1822.88851 electron masses/amu.)</p>
<p>A typical good value for  <strong>dtion</strong>  is about 100. The user must try several
values for  <strong>dtion</strong>  in order to establish the stable and efficient choice for
the accompanying amu, atom types and positions, and <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a> (viscosity).
For quenched dynamics (<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;quenched&rdquo;), a larger time step might be taken, for
example 200. No meaning for RF calculations.
It is also used in geometric relaxation calculation with the FIRE algorithm
(<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;fire&rdquo;), where the time is virtual. A small dtion should be set, for example 0.03.</p>
<h2 id="dynimage"><strong>dynimage</strong><a class="headerlink" href="#dynimage" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> list of DYNamic IMAGEs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a>)<br />
<em>Default value:</em> <em>1<br />
*Comment:</em> if <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> in [2,5] (String Method, NEB), <b>dynimage(1)</b>=0 and <b>dynimage(<a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a>)</b>=0.<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [14/1314] in all abinit tests, [2/178] in abinit tutorials</summary>
<ul>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_04.abi" target="_blank">timages_04.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>This input variable is relevant when sets of images are activated (see
<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>). Not all images might be required to evolve from one time step to
the other. Indeed, in the String Method or the Nudged Elastic Band, one might
impose that the extremal configurations of the string are fixed. In case
 <strong>dynimage</strong> (iimage)=0, the image with index &ldquo;iimage&rdquo; will be consider as
fixed. Thus, there is no need to compute forces and stresses for this image at
each time step. The purpose of defining extremal images is to make the input/output easier.</p>
<p>In order to save CPU time, the computation of properties of static images
( <strong>dynimage</strong> (iimage)=0) can be avoided: see <a class="wikilink codevar-wikilink" href="/variables/rlx#istatimg">istatimg</a> keyword.</p>
<h2 id="ecutsm"><strong>ecutsm</strong><a class="headerlink" href="#ecutsm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Energy CUToff SMearing<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">ENERGY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform numerically precise calculations with planewaves or projector- augmented waves and pseudopotentials" data-toggle="popover" href="/topics/Planewaves">topic_Planewaves</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [151/1314] in all abinit tests, [24/178] in abinit tutorials</summary>
<ul>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu/Input/t04.abi" target="_blank">t04.abi</a> &hellip;</li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu_omp/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Preparation for a nonlinear calculation : GS + linear response (ddk+phonon+efield), using GEMM nonlop.
Related tests: v8 83 and gpu_omp t25" data-toggle="popover" href="/tests/gpu_omp/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Preparation for a nonlinear calculation : GS + linear response (stresses), using GEMM nonlop.
Related tests: v8 t83 and gpu_omp t21 ." data-toggle="popover" href="/tests/gpu_omp/Input/t25.abi" target="_blank">t25.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/mpiio/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium,
to compare with the PSML results of t07 and t11" data-toggle="popover" href="/tests/psml/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t07.abi" target="_blank">t07.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=14 (finite reduced electric field calculation,relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_121.abi" target="_blank">tsv6_121.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=16 (finite reduced electric displacement field calculation,
relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_122.abi" target="_blank">tsv6_122.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=4 (finite electric field calculation, relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_123.abi" target="_blank">tsv6_123.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
AlP in hypothetical wurzite (hexagonal) structure
Response function calculation for:
* rigid-atom elastic tensor
* rigid-atom piezoelectric tensor
* interatomic force constants at gamma
* Born effective charges" data-toggle="popover" href="/tests/tutorespfn/Input/telast_2.abi" target="_blank">telast_2.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
AlP in hypothetical wurzite (hexagonal) structure
Finite-difference calculation for c-axis strain increment +/- 0.0001" data-toggle="popover" href="/tests/tutorespfn/Input/telast_4.abi" target="_blank">telast_4.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline silicon : computation of the optimal lattice parameter
Convergence with respect to the number of k points." data-toggle="popover" href="/tests/tutorial/Input/tbase3_4.abi" target="_blank">tbase3_4.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline aluminum : optimization of the lattice parameter
at fixed number of k points and broadening." data-toggle="popover" href="/tests/tutorial/Input/tbase4_1.abi" target="_blank">tbase4_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline aluminum : optimization of the lattice parameter

Convergence with respect to k points." data-toggle="popover" href="/tests/tutorial/Input/tbase4_2.abi" target="_blank">tbase4_2.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Test ecutsm. Determine a smooth etot vs acel curve
(one can try with ecutsm=0.0d0 to see the difference !)
Stress is correct. For example,
at acell=10.10 (volume=2.5757525E+02), one obtains etotal=-8.7850566628
at acell=10.12 (volume=2.5910843E+02), one obtains etotal=-8.7851433215,
giving a finite-difference estimate of the stress ( d(etotal)/d(volume) )
at 10.11 to be -5.6522d-5, while the code gives -5.6582015717E-05" data-toggle="popover" href="/tests/v1/Input/t76.abi" target="_blank">t76.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Optimization of unit cell volume (optcell=1).
Use 2 datasets, and the variable getcell." data-toggle="popover" href="/tests/v1/Input/t78.abi" target="_blank">t78.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a cell of variable size along the H2 chain =&gt; H polymer.
Optimization of cell size and atomic distances (optcell=4)." data-toggle="popover" href="/tests/v1/Input/t79.abi" target="_blank">t79.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Calculation of the Fan-Migdal + Debye-Waller self-energy matrix elements and Cumulant + DM mobilities in Silicon." data-toggle="popover" href="/tests/v10/Input/t107.abi" target="_blank">t107.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Arsenic in rhombohedral structure.
Optimization of unit cell size and shape.
With 4x4x4 k point grid, ecut 3.0, occopt 4 and tsmear 0.06d0,
one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277
These input parameters are too small, but still give a realistic
geometry, since with the much better parameters
12x12x12 k point grid, ecut 12.0, occopt 4 and tsmear 0.04d0,
one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296, while
the experimental values are
acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-toggle="popover" href="/tests/v2/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Li, simple cubic, to test accurately RF in the metallic case.
With non-linear XC core correction and non-zero ecutsm and dilatmx.
Similar to test 92, except ecutsm and dilatmx, and no q(1/4 0 0).
The total energy of the 1-atom cell is -.29337493523588 Ha
The 2DTE with respect to a q(0 0 0) phonon is 5.5619d-8 .
The 2DTE with respect to a transverse q(1/2 0 0) phonon
is -0.12068188927 Ha." data-toggle="popover" href="/tests/v2/Input/t96.abi" target="_blank">t96.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Li, simple cubic, frozen-phonon calculations corresponding
to test 96 (with non-zero ecutsm and dilatmx).
The total energy of the double cell is -0.58674987047187 Ha,
in excellent agreement with the result of test 96.
The frozen-phonon (from energy) calculation of the 2DTE for the q(1/2 0 0)
case gives -0.1206807 Ha .
The frozen-phonon (from forces) calculation of the 2DTE for the q(1/2 0 0)
case gives -0.1206794 Ha . Both are in rather good agreement with
the result of test 96." data-toggle="popover" href="/tests/v2/Input/t97.abi" target="_blank">t97.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
Magnesium HCP. Optimize the cell volume and shape using different constraints.
Checks whether the stresses are symmetric, and whether the optimisation algorithm does
not break the symmetry.
Also checks the possibility to disable the stop after violation of the condition related to dilatmx.
This is done in dataset 6. The optimization delivers acell values slightly different from those of dataset 2." data-toggle="popover" href="/tests/v3/Input/t42.abi" target="_blank">t42.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule : examine the rotational freedom.
The present test produces the following
vibrational frequencies (with degeneracies indicated):
56.89 i cm-1 (2)
0.41   cm-1 (2)
1.05   cm-1
3800    cm-1
The large frequency corresponds to the stretching
mode, and has the right order of magnitude.
The frequencies close to 1 cm-1 corresponds
to translation modes, and are small enough
for usual applications.
The  56.89 i cm-1 mode corresponds to rotation of
the H2 molecule. The magnitude of this
frequency might seem quite
large. Here are the results of tests made to understand
this phenomenon. First, note that
ecut 12  acell 3*14
Increasing the value of ecut to 25 decreases
the magnitude of the frequency to 36.8 cm-1 .
However, in order to continue to make it smaller,
the cell size must be increased , and an oscillatory
behaviour is observed :
3*16   45.6 i cm-1
3*18   22.7   cm-1
3*20   19.1 i cm-1
3*22   15.7 i cm-1
3*24   13.7   cm-1
Many other tests have been set up. In particular,
it was observed that the frequency of the oscillatory
behaviour changes with the ecut, and also that
using the Gamma point, instead of the 1/4 1/4 1/4 k point
(used in this test) degrades the convergence.
The overall picture is as follows.
There are different reasons for the translation
and rotation modes to acquire a non-zero frequency
when plane waves and supercells are used.
Still, as concerns translations, only the
existence of a discretization of the XC grid
is important. For rotations, supercell effects
are also present :
- alignement of dipole or quadrupoles
- interaction between tails of wavefunctions, accross cells
Since the convergence in supercell size is oscillatory, we infer
that the breaking of the rotational invariance is mostly
due to interaction between wavefunction tails.
This will be checked by confining the system in a spherical
well, in a forthcoming test." data-toggle="popover" href="/tests/v3/Input/t80.abi" target="_blank">t80.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98).
Compute responses at Gamma:
1) Ground state calculation, with a k-point sampling in the IBZ only;
2) Computation of the ddk response;
3) Computation of the dynamical matrix at Gamma, as well as the
Born effective charges, and the dielectric tensor ;
Perform five datasets:
* First dataset is the ground state calculation,
* Second dataset compute ddk perturbation
with the use of Legendre Polynomials for the nl operator,
* Third dataset compute phonon and elec. field perturbation
with the use of Legendre Polynomials for the nl operator,
* Fourth dataset compute ddk perturbation
with the use of spherical harmonics for the nl operator,
* Fifth dataset compute phonon and elec. field perturbation
with the use of spherical harmonics for the nl operator.
Datasets 2 and 3 must give the same results as 4 and 5." data-toggle="popover" href="/tests/v4/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the pspcod=8 input format (contributed by D.R. Hamann)
This format is primarily intended to offer a flexible platform
for those who wish to experiment with their own features in
norm-conserving pseudopotentials.  A full description and discussion
is given on the ABINIT wiki
https://wiki.abinit.org/doku.php?id=developers:pseudos:psp8_format
This test case, CaO, is an
example of self-interaction-corrected pseudopotentials in the
style of D. Vogel, P. Krueger, and J. Pollmann, Phys. Rev. B 52,
14 316 (1995).  The bandgap (direct) in the test example, 11.2eV,
is in excellent agreement with experiment, but is rather strongly
dependent on the radius at which the self-interaction-correction
of Ca is (smoothly) cut off.  The sic pseudopotentials used in
this test should not be considered anything but early-stage
experiments and should not be used for any other purpose." data-toggle="popover" href="/tests/v4/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Si/Ge heterojunction : 4 layer each. Output the density and potential.
Prepare the macroaverage analysis, see next tests." data-toggle="popover" href="/tests/v4/Input/t42.abi" target="_blank">t42.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-toggle="popover" href="/tests/v5/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline Silicon
Test the computation of second order eigenvalues for two qpoints, (0 0 0)
and (1/2 0 0), contained in the EIGR2D files. Parameters are far from
convergence, nband=5 and nkpt=16." data-toggle="popover" href="/tests/v5/Input/t26.abi" target="_blank">t26.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a> &hellip;</li>
<li>v67mbpt:  <a class="wikilink abifile-wikilink" data-tippy-content="Test the cd_custom_imfrq, cd_imfrqs and cd_full_grid options" data-toggle="popover" href="/tests/v67mbpt/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test new integration method and grid options for the integral
along the imaginary axis in contour deformation calculations." data-toggle="popover" href="/tests/v67mbpt/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density.
We test several improvements that are available when the NC file contains the PS valence charge.
More specifically, we test the initialization of the density from atomic quantities (initro.F90),
the correction to the forces due to the residuals and the extrapolation of the density
for structural relaxations (well, xred is fixed by symmetry).
We also test the treatment of the core charge in G-space in the case of GS calculations
By default, abinit handles XCCC in real space, here we use nc_xccc_gspace==1 to treat it
in G-space using the same approach as the one used in PAW." data-toggle="popover" href="/tests/v7/Input/t10.abi" target="_blank">t10.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Test presence of nuclear magnetic dipole moments on atomic sites" data-toggle="popover" href="/tests/v7/Input/t32.abi" target="_blank">t32.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Diamond.
Test temperature-dependent of the electronic structure, with
reduction of the number of q points to be computed, thanks to thmflag=7 .
The temperature-dependent shifts must be averaged over degenerate states.
This temperature-dependent calculation make use of a k-point interpolation to speed up
the convergence study with respect to the q-points integration. Two q-points are computed." data-toggle="popover" href="/tests/v7/Input/t50.abi" target="_blank">t50.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="Geometry optimization with LBFGS" data-toggle="popover" href="/tests/v8/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Computation of elastic tensor and internal strain in DFPT+PAW
Ground state is computed.
DFPT is computed" data-toggle="popover" href="/tests/v8/Input/t07.abi" target="_blank">t07.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Geometry optimization with FIRE (ionmov 15)" data-toggle="popover" href="/tests/v8/Input/t17.abi" target="_blank">t17.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
Preparatory run for the calculation of the Fan-Migdal self-energy
C in diamond structure. Very rough 2x2x2 q-point grid (3 qpoints); low ecut." data-toggle="popover" href="/tests/v9/Input/t50.abi" target="_blank">t50.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Calculation of the imaginary part of the Fan-Migdal self-energy matrix elements." data-toggle="popover" href="/tests/v9/Input/t53.abi" target="_blank">t53.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Calculation of the imaginary part of the Fan-Migdal self-energy matrix elements.
Use of the tetrahedron method for the electron linewidth" data-toggle="popover" href="/tests/v9/Input/t54.abi" target="_blank">t54.abi</a> &hellip;</li>
<li>vdwxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-toggle="popover" href="/tests/vdwxc/Input/t10.abi" target="_blank">t10.abi</a> &hellip;</li>
<li>wannier90:  <a class="wikilink abifile-wikilink" data-tippy-content="
Wannier function for bcc Fe.
To check whether wannier90 interface works properly with nsppol=2 and nspden=2.
Small accuracy parameters, small k-mesh and large test tolerance are used." data-toggle="popover" href="/tests/wannier90/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Wannier function for bcc Fe.
To check whether wannier90 interface works properly with nspinor=2  and nspden=4
Small accuracy parameters, small k-mesh and large test tolerance are used." data-toggle="popover" href="/tests/wannier90/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Argon FCC. Test the approach of Silvestrelli to access van der Waals interactions." data-toggle="popover" href="/tests/wannier90/Input/t11.abi" target="_blank">t11.abi</a> &hellip;</li>
</ul>
</details>
<p>This input variable is important when performing relaxation of unit cell size
and shape (non-zero <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>). Using a non-zero  <strong>ecutsm</strong> , the total
energy curves as a function of <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a>, or <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a>, can be smoothed,
keeping consistency with the stress (and automatically including the Pulay
stress).</p>
<p>The recommended value is 0.5 Ha in such a case (non-zero <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>).</p>
<p>Actually, when <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>/=0,
ABINIT requires  <strong>ecutsm</strong>  to be larger than zero. If you want to optimize
cell shape and size without smoothing the total energy curve (a dangerous
thing to do), use a very small  <strong>ecutsm</strong> , on the order of one microHartree.</p>
<p>Technical information:
See Appendix B of <a class="wikilink citation-wikilink" data-tippy-content="Precise effective masses from density functional perturbation theory" data-toggle="popover" href="/theory/bibliography#laflamme2016">[Laflamme2016]</a>.
 <strong>ecutsm</strong>  allows one to define an effective kinetic energy for plane waves, close
to, but lower than, the maximal kinetic energy <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a>. For kinetic energies
less than <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a>- <strong>ecutsm</strong> , nothing is modified, while between
<a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a>- <strong>ecutsm</strong>  and <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a>, the kinetic energy is multiplied by
<span class="arithmatex">\(1.0 / ( x^2(3+x-6x^2+3x^3 ))\)</span>,
where x = (<a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> - kinetic_energy)/ <strong>ecutsm</strong> .
Note that <span class="arithmatex">\(x^2(3+x-6x^2+3x^3)\)</span> is 0 at x=0, with vanishing derivative,
and that at x=1, it is 1, with also vanishing derivative.
If  <strong>ecutsm</strong>  is zero, the unmodified kinetic energy is used.
 <strong>ecutsm</strong>  can be specified in Ha (the default), Ry, eV or Kelvin, since
 <strong>ecutsm</strong>  has the <a class="wikilink" href="/guide/abinit#parameters">ENERGY</a> characteristics. (1 Ha = 27.2113845 eV).
A few test for Silicon (diamond structure, 2 k-points) have shown 0.5 Ha to be
largely enough for <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> between 2Ha and 6Ha, to get smooth curves. It is
likely that this value is OK as soon as <a class="wikilink codevar-wikilink" href="/variables/basic#ecut">ecut</a> is larger than 4Ha.</p>
<h2 id="friction"><strong>friction</strong><a class="headerlink" href="#friction" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> internal FRICTION coefficient<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.001<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [6/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Gives the internal friction coefficient (atomic units) for Langevin dynamics
(when <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_langevin&rdquo; or &ldquo;npt_langevin&rdquo; or <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 16 or <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9): fixed temperature simulations with random forces.</p>
<p>The equations of motion are:
<span class="arithmatex">\(M_I \frac{d^2 R_I}{dt^2}= F_I -\)</span> <strong>friction</strong> <span class="arithmatex">\(*M_I \frac{d R_I}{dt} - F_{random,I}\)</span>,
where <span class="arithmatex">\(F_{random,I}\)</span> is a Gaussian random force with average zero and variance  <strong>friction</strong> <span class="arithmatex">\(*2M_IkT/\)</span><a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>.
The atomic unit of  <strong>friction</strong>  is Hartree*Electronic mass*(atomic unit of Time)/Bohr^2. See <a class="wikilink citation-wikilink" data-tippy-content="The pseudopotential-density functional method applied to nanostructures" data-toggle="popover" href="/theory/bibliography#chelikowsky2000">[Chelikowsky2000]</a> and <a class="wikilink citation-wikilink" data-tippy-content="Langevin dynamics in constant pressure extended systems" data-toggle="popover" href="/theory/bibliography#quigley2004">[Quigley2004]</a> for additional information.</p>
<h2 id="frictionbar"><strong>frictionbar</strong><a class="headerlink" href="#frictionbar" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> internal FRICTION coefficient of the BARostat<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.001<br />
<em>Added in version:</em> v10.2  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
</ul>
</details>
<p>Gives the internal friction coefficient (atomic units) of the barostat for Langevin dynamics
(when <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;npt_langevin&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 16 and <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 2).
The mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>) must be given in addition.
See <a class="wikilink citation-wikilink" data-tippy-content="Langevin dynamics in constant pressure extended systems" data-toggle="popover" href="/theory/bibliography#quigley2004">[Quigley2004]</a> and <a class="wikilink citation-wikilink" data-tippy-content="Constant pressure Langevin dynamics: theory and application" data-toggle="popover" href="/theory/bibliography#quigley2005">[Quigley2005]</a> for additional information.</p>
<h2 id="fxcartfactor"><strong>fxcartfactor</strong><a class="headerlink" href="#fxcartfactor" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Forces to (X) CARTesian coordinates FACTOR<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1 (Bohr^2)/Hartree<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [10/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
</ul>
</details>
<p>The forces multiplied by  <strong>fxcartfactor</strong>  will be treated like difference in
cartesian coordinates in the process of optimization. This is a simple preconditioner.
TO BE UPDATED See (<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;bfgs&rdquo; or &ldquo;lbfgs&rdquo;, non-zero <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>). For example, the
stopping criterion defined by <a class="wikilink codevar-wikilink" href="/variables/rlx#tolmxf">tolmxf</a> relates to these scaled stresses.</p>
<h2 id="ga_algor"><strong>ga_algor</strong><a class="headerlink" href="#ga_algor" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm - ALGOrithm selection<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [0/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
</details>
<p>Choosing method to make the structure selection. Only the enthalpy is used now
but we plan to include, energy, electronic band gap and alchemical potentials.
Right now only value of 1 (enthalpy) works.</p>
<h2 id="ga_fitness"><strong>ga_fitness</strong><a class="headerlink" href="#ga_fitness" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm FITNESS function selection<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a></li>
</ul>
</details>
<p>Different methodologies to perform the roulette-wheel selection of parents.
Even though, the objective function is the crystalline enthalpy (H_i), the
weight of the population elements to be chosen from in a roulette-wheel
selection can be given through different functions. We consider the following cases.</p>
<ol>
<li>F = H_i / Sum H_i</li>
<li>F = exp(-(H_i-H_min)) / Sum exp(-(H_i-H_min))</li>
<li>F = (1/n_i) / Sum (1/n_i). Where n_i is the position in the ordered list of enthalpies</li>
</ol>
<h2 id="ga_n_rules"><strong>ga_n_rules</strong><a class="headerlink" href="#ga_n_rules" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm Number of RULES<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a></li>
</ul>
</details>
<p>Different genetic rules have been implemented and the user has the change to
choose between any of them. Right now we have 4 rules. See <a class="wikilink codevar-wikilink" href="/variables/rlx#ga_rules">ga_rules</a></p>
<h2 id="ga_opt_percent"><strong>ga_opt_percent</strong><a class="headerlink" href="#ga_opt_percent" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm OPTimal PERCENT<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.2<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a></li>
</ul>
</details>
<p>Percentage of the population that according to the fitness function passes to the following iteration.</p>
<h2 id="ga_rules"><strong>ga_rules</strong><a class="headerlink" href="#ga_rules" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm RULES<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a></li>
</ul>
</details>
<p>Different genetic rules have been implemented and the user has the change to
choose between any of them. The chosen number of rules have been defined in <a class="wikilink codevar-wikilink" href="/variables/rlx#ga_n_rules">ga_n_rules</a></p>
<p>Implemented rules are
1) crossover. Two parents are randomly chosen and two springs are mixed from
the two by (a) choosing randomly (through Fitness function) two parents and
then randomly rotating and shifting the coordinates withing that particular
cell. (b) Slice every one of the unit cell of the parents along a random
direction and creating the spring offs from the pieces of the two parents.
2) Vector flip mutation. From the coordinates from a given parent, a piece of
it is inverted.
3) Random strain. A random anisotropic deformation is given to the unit cell.
4) Coordinates mutation of &frac14; of the whole coordinates.</p>
<h2 id="geoopt"><strong>geoopt</strong><a class="headerlink" href="#geoopt" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GEOmetry OPTimization<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> string<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> none<br />
<em>Added in version:</em> 10.3  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [23/1314] in all abinit tests, [23/178] in abinit tutorials</summary>
<ul>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_01.abi" target="_blank">timages_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_02.abi" target="_blank">timages_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Finite difference calculation of the clamped-ion
piezoelectric constants of AlP" data-toggle="popover" href="/tests/tutorespfn/Input/tpolarization_5.abi" target="_blank">tpolarization_5.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file will optimize automatically the interatomic distance" data-toggle="popover" href="/tests/tutorial/Input/tbase1_3.abi" target="_blank">tbase1_3.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom." data-toggle="popover" href="/tests/tutorial/Input/tbase2_1.abi" target="_blank">tbase2_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom.
Here, a double loop has been used." data-toggle="popover" href="/tests/tutorial/Input/tbase2_2.abi" target="_blank">tbase2_2.abi</a> &hellip;</li>
</ul>
</details>
<p>Choice of algorithm to control the displacements of ions for geometry optimization, and possibly changes of cell shape and size (see <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>).
No meaning for RF calculations.</p>
<ul>
<li>
<p>&ldquo;none&rdquo; &rarr; Do not move ions (<strong>default behaviour</strong>)</p>
</li>
<li>
<p>&ldquo;viscous&rdquo; &rarr; Move atoms using molecular dynamics with optional viscous damping (friction linearly proportional to velocity).
  The viscous damping is controlled by the <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a> parameter.
  If undamped molecular dynamics is desired, set <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a> to 0.
  The implemented algorithm is the generalisation of the Numerov technique (6<sup>th</sup> order), but is NOT invariant upon time-reversal,
  so that the energy is not conserved.
  The time step is governed by <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>.
  <strong>Purpose:</strong> Molecular dynamics (if <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a> = 0), Structural optimization (if <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a> &gt;0)
  <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
  <strong>Related variables:</strong> Viscous parameter <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a>, time step <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>, index of atoms fixed <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a></p>
</li>
<li>
<p>&ldquo;bfgs&rdquo; &rarr; Conduct structural optimization using the Broyden-Fletcher-Goldfarb-Shanno minimization (BFGS).
  This is much more efficient for structural optimization than viscous damping, when there are
  less than about 10 degrees of freedom to optimize.
  <strong>Purpose:</strong> Structural optimization
  <strong>Cell optimization:</strong> Yes (if <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>/=0)</p>
</li>
<li>
<p>&ldquo;mdmin&rdquo; &rarr; Simple relaxation of ionic positions according to (converged) forces.
  Equivalent to <strong>geoopt</strong> = &ldquo;viscous&rdquo; with zero masses, albeit the relaxation coefficient is not <a class="wikilink codevar-wikilink" href="/variables/rlx#vis">vis</a>, but <a class="wikilink codevar-wikilink" href="/variables/dev#iprcfc">iprcfc</a>.
  <strong>Purpose:</strong> Structural optimization
  <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)</p>
</li>
<li>
<p>&ldquo;quenched&rdquo; &rarr; Quenched Molecular dynamics using the Verlet algorithm, and stopping each atom for which
  the scalar product of velocity and force is negative. The only related parameter is the time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>).
  The goal is not to produce a realistic dynamics, but to go as fast as possible to the minimum.
  For this purpose, it is advised to set all the masses to the same value
  (for example, use the Carbon mass, i.e. set <a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a> to 12 for all type of atoms).
  <strong>Purpose:</strong> Structural optimization
  <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
  <strong>Related variables:</strong> time step <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>, index of atoms fixed <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a></p>
</li>
<li>
<p>&ldquo;fire&rdquo; &rarr; Fast inertial relaxation engine (FIRE) algorithm proposed by Erik Bitzek, Pekka Koskinen, Franz Ghler,
    Michael Moseler, and Peter Gumbsch in <a class="wikilink citation-wikilink" data-tippy-content="Structural Relaxation Made Simple" data-toggle="popover" href="/theory/bibliography#bitzek2006">[Bitzek2006]</a>.
    According to the authors, the efficiency of this method is nearly the same as L-bfgs (<strong>ionmov</strong> = 22).
    It is based on conventional molecular dynamics with additional velocity modifications and adaptive time steps.
    The initial time step is set with <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>. Note that the physical meaning and unit of <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a> are different
    from the default ones.
    The purpose of this algorithm is relaxation, not molecular dynamics. <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a> governs the ion position changes,
    but the cell parameter changes as well.
    The positions are in reduced coordinates instead of in cartesian coordinates.
    The suggested first guess of dtion is 0.03.
    <strong>Purpose:</strong> Relaxation
    <strong>Cell optimization:</strong> Yes (if <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>/=0)
    <strong>Related variables:</strong> The initial time step <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a></p>
</li>
<li>
<p>&ldquo;lbfgs&rdquo; &rarr; Conduct structural optimization using the Limited-memory
    Broyden-Fletcher-Goldfarb-Shanno minimization (L-BFGS) <a class="wikilink citation-wikilink" data-tippy-content="Updating Quasi-Newton Matrices with Limited Storage" data-toggle="popover" href="/theory/bibliography#nocedal1980">[Nocedal1980]</a>.
    The routines are based on the original implementation by J. Nocedal available on netlib.org.
    This algorithm can be much better than the native implementation of BFGS in ABINIT (<strong>ionmov</strong> = 2)
    when one approaches convergence, perhaps because of better treatment of numerical details.
    <strong>Purpose:</strong> Structural optimization
    <strong>Cell optimization:</strong> Yes (if <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>/=0)</p>
</li>
</ul>
<h2 id="getcell"><strong>getcell</strong><a class="headerlink" href="#getcell" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET CELL parameters from&hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to set parameters for a multi dataset calculation" data-toggle="popover" href="/topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [5/1314] in all abinit tests, [1/178] in abinit tutorials</summary>
<ul>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu/Input/t04.abi" target="_blank">t04.abi</a></li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu_omp/Input/t01.abi" target="_blank">t01.abi</a></li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a></li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Optimization of unit cell volume (optcell=1).
Use 2 datasets, and the variable getcell." data-toggle="popover" href="/tests/v1/Input/t78.abi" target="_blank">t78.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Mg crystal, 1 atom per unit cell. Start with deformed FCC lattice,
optimize the lattice towards FCC (optcell=2).
Also test the output of the HIST.nc file (restartxf==-1)" data-toggle="popover" href="/tests/v1/Input/t80.abi" target="_blank">t80.abi</a></li>
</ul>
</details>
<p>This variable is typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="/variables/basic#ndtset">ndtset</a> &gt; 0). It describes from which dataset the output <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a>
and <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> are to be taken (implicitly also <a class="wikilink codevar-wikilink" href="/variables/basic#scalecart">scalecart</a>), as input of the present dataset.
The cell parameters are <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a> variables, for which such a chain of calculations is useful.</p>
<ul>
<li>If  <strong>getcell</strong> =0, no previously computed values are used.</li>
<li>If  <strong>getcell</strong> &gt;0, the value must be the index of the dataset from which the
cell data is to be used as input data. It must be the index of a dataset already
computed in the SAME run.</li>
<li>If  <strong>getcell</strong> =-1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has been computed in the same run.</li>
<li>If  <strong>getcell</strong> =another negative number, it indicates the number of datasets to go backward
to find the needed data (once again, going back beyond the first dataset is
equivalent to using a zero get variable).</li>
</ul>
<h2 id="getvel"><strong>getvel</strong><a class="headerlink" href="#getvel" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET VEL from&hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to set parameters for a multi dataset calculation" data-toggle="popover" href="/topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [0/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
</details>
<p>This variable is typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="/variables/basic#ndtset">ndtset</a> &gt; 0) since they describe from which dataset the
corresponding output variables are to be taken, as input of the present
dataset. The velocities are <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a> variables, for
which such a chain of calculation is useful.</p>
<ul>
<li>
<p>If  <strong>getvel</strong>  == 0, no such use of previously computed <a class="wikilink codevar-wikilink" href="/variables/rlx#vel">vel</a> is done.</p>
</li>
<li>
<p>If  <strong>getvel</strong>  is positive, its value gives the index of the dataset from which
the data are to be used as input data. It must be the index of a dataset
already computed in the SAME run.</p>
</li>
<li>
<p>If  <strong>getvel</strong>  is -1, the output data of the previous dataset must be taken,
which is a frequently occurring case.</p>
</li>
<li>
<p>If  <strong>getvel</strong>  is a negative number, it indicates the number of datasets to go
backward to find the needed data. Going back beyond the first dataset is equivalent to using zero for the get variable.</p>
</li>
</ul>
<p>Note: The use of  <strong>getvel</strong>  with <a class="wikilink codevar-wikilink" href="/variables/rlx#getxred">getxred</a> is allowed, despite the different coordinate system.</p>
<h2 id="getxcart"><strong>getxcart</strong><a class="headerlink" href="#getxcart" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET XCART from&hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to set parameters for a multi dataset calculation" data-toggle="popover" href="/topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [8/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule in a box of increasing size.
Use ecut=12 (see previous test).
5 different values of acell (convergence study). Use getxcart=-1." data-toggle="popover" href="/tests/v1/Input/t61.abi" target="_blank">t61.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule in a box of increasing size.
Same as test 61, except that it reads previous wavefunctions
(use getxcart=-1 and getwfk=-1). mkmem/=0 ." data-toggle="popover" href="/tests/v1/Input/t64.abi" target="_blank">t64.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule with 1 and 4 k points.
(use getxcart=-1 and getwfk=-1). Convergence study with k points." data-toggle="popover" href="/tests/v1/Input/t65.abi" target="_blank">t65.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule with 1 k points.
Use getxcart=-1 and getwfk=-1 : first perform a geometry optimization,
with minimal number of bands, then compute more bands at fixed geometry,
with an increased ecut." data-toggle="popover" href="/tests/v1/Input/t66.abi" target="_blank">t66.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule with 1 k points.
Convergence study, using a geometric series for ecut." data-toggle="popover" href="/tests/v1/Input/t74.abi" target="_blank">t74.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Same as test 66, H2 molecule with 1 k points, but uses the restartxf=-1 option
to test the output of the HIST.nc file. Note that HIST.nc is not read since
one should introduce a getrestartxf variable." data-toggle="popover" href="/tests/v1/Input/t75.abi" target="_blank">t75.abi</a></li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule : examine the rotational freedom.
The present test produces the following
vibrational frequencies (with degeneracies indicated):
56.89 i cm-1 (2)
0.41   cm-1 (2)
1.05   cm-1
3800    cm-1
The large frequency corresponds to the stretching
mode, and has the right order of magnitude.
The frequencies close to 1 cm-1 corresponds
to translation modes, and are small enough
for usual applications.
The  56.89 i cm-1 mode corresponds to rotation of
the H2 molecule. The magnitude of this
frequency might seem quite
large. Here are the results of tests made to understand
this phenomenon. First, note that
ecut 12  acell 3*14
Increasing the value of ecut to 25 decreases
the magnitude of the frequency to 36.8 cm-1 .
However, in order to continue to make it smaller,
the cell size must be increased , and an oscillatory
behaviour is observed :
3*16   45.6 i cm-1
3*18   22.7   cm-1
3*20   19.1 i cm-1
3*22   15.7 i cm-1
3*24   13.7   cm-1
Many other tests have been set up. In particular,
it was observed that the frequency of the oscillatory
behaviour changes with the ecut, and also that
using the Gamma point, instead of the 1/4 1/4 1/4 k point
(used in this test) degrades the convergence.
The overall picture is as follows.
There are different reasons for the translation
and rotation modes to acquire a non-zero frequency
when plane waves and supercells are used.
Still, as concerns translations, only the
existence of a discretization of the XC grid
is important. For rotations, supercell effects
are also present :
- alignement of dipole or quadrupoles
- interaction between tails of wavefunctions, accross cells
Since the convergence in supercell size is oscillatory, we infer
that the breaking of the rotational invariance is mostly
due to interaction between wavefunction tails.
This will be checked by confining the system in a spherical
well, in a forthcoming test." data-toggle="popover" href="/tests/v3/Input/t80.abi" target="_blank">t80.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a reasonably large box
Compute the Fan and Diagonal Debye-Waller corrections to the eigenenergies." data-toggle="popover" href="/tests/v6/Input/t68.abi" target="_blank">t68.abi</a></li>
</ul>
</details>
<p>This variable is typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="/variables/basic#ndtset">ndtset</a> &gt; 0) since it  describes from which dataset the
corresponding output variables are to be taken, as input of the present
dataset. The atomic positions are <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a> variables, for
which such a chain of calculation is useful.</p>
<p>Note that the use of  <strong>getxcart</strong>  and <a class="wikilink codevar-wikilink" href="/variables/rlx#getxred">getxred</a> differs when <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a> and
<a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> are different from one dataset to the other.</p>
<ul>
<li>
<p>If 0, no previously computed values are used.</p>
</li>
<li>
<p>If &gt;0, the integer must correspond to the index of the dataset from which the
data are to be used as input data. It must be the index of a dataset already
computed in the SAME run.</p>
</li>
<li>
<p>If equal to -1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has yet been computed in the same run.</p>
</li>
<li>
<p>If another negative number, it indicates the number of datasets to go backward
to find the needed data. Going back beyond the first dataset is equivalent to using a null get variable.</p>
</li>
</ul>
<p>Note: <a class="wikilink codevar-wikilink" href="/variables/rlx#getxred">getxred</a> and  <strong>getxcart</strong>  cannot be simultaneously non-zero for the
same dataset. On the other hand the use of <a class="wikilink codevar-wikilink" href="/variables/rlx#getvel">getvel</a> with <a class="wikilink codevar-wikilink" href="/variables/rlx#getxred">getxred</a> is
allowed, despite the different coordinate system.</p>
<h2 id="getxred"><strong>getxred</strong><a class="headerlink" href="#getxred" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET XRED from&hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to set parameters for a multi dataset calculation" data-toggle="popover" href="/topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [12/1314] in all abinit tests, [1/178] in abinit tutorials</summary>
<ul>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule NON-spin-polarized, in a big box, without tetragonal symmetry, in order to avoid
spurious degeneracy effects coupled to occupation numbers rapidly changing.
GPU
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization." data-toggle="popover" href="/tests/gpu_omp/Input/t28.abi" target="_blank">t28.abi</a></li>
<li>libxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule spin-polarized, in a big box.
Norm conserving : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with underconverged ecut (ecut 15) and box size (4 3 3 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation  (ecut 13, 5 4 4 box) : 1.138, 1.130, 1.131 Angstrom
Better  calculation  (ecut 18, 7 6 6 box) : 1.135, 1.124, 1.125 Angstrom
ABINIT Arnardi report (ecut acell unknown): 1.136, 1.124, 1.124 Angstrom
VASP Arnardi report (ecut acell unknown)  : 1.136, 1.124,  NA   Angstrom
Experimental : 1.120 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t68.abi" target="_blank">t68.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule spin-polarized, in a big box.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with slightly better converged ecut (ecut 15) and box size (6 5 5 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation (ecut 13, 5 4 4 box)     : 1.130, 1.121, 1.122 Angstrom
Better calculation (ecut 15, 6 5 5 box)      : 1.136, 1.125, 1.126 Angstrom
Even better calculation (ecut 18, 7 6 6 box) : 1.138, 1.126, 1.127 Angstrom
ABINIT Arnardi report (ecut acell unknown)   : 1.136, 1.124, 1.124 Angstrom
VASP Arnardi report (ecut acell unknown)     : 1.136, 1.124,  NA   Angstrom
Experimental : 1.120 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t69.abi" target="_blank">t69.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule NON-spin-polarized, in a big box, without tetragonal symmetry, in order to avoid
spurious degeneracy effects coupled to occupation numbers rapidly changing.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization." data-toggle="popover" href="/tests/libxc/Input/t70.abi" target="_blank">t70.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule NON-spin-polarized, in a big box, without tetragonal symmetry, in order to avoid
spurious degeneracy effects coupled to occupation numbers rapidly changing.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization." data-toggle="popover" href="/tests/libxc/Input/t71.abi" target="_blank">t71.abi</a></li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a></li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule in a box (psp PseudosTM_pwteter/1h.pspnc). acell constant, small (7 5 5).
10 different values of ecut (convergence study). Use getxred=-1." data-toggle="popover" href="/tests/v1/Input/t60.abi" target="_blank">t60.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule in a box of increasing size.
Same as test 61, except that it reads previous wavefunctions
(use getxcart=-1 and getwfk=-1). mkmem/=0 ." data-toggle="popover" href="/tests/v1/Input/t64.abi" target="_blank">t64.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test multi-dataset mode. H2 molecule with 1 k points.
Convergence study, using a geometric series for ecut." data-toggle="popover" href="/tests/v1/Input/t74.abi" target="_blank">t74.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Mg crystal, 1 atom per unit cell. Start with deformed FCC lattice,
optimize the lattice towards FCC (optcell=2).
Also test the output of the HIST.nc file (restartxf==-1)" data-toggle="popover" href="/tests/v1/Input/t80.abi" target="_blank">t80.abi</a></li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Arsenic in rhombohedral structure.
Optimization of unit cell size and shape.
With 4x4x4 k point grid, ecut 3.0, occopt 4 and tsmear 0.06d0,
one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277
These input parameters are too small, but still give a realistic
geometry, since with the much better parameters
12x12x12 k point grid, ecut 12.0, occopt 4 and tsmear 0.04d0,
one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296, while
the experimental values are
acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-toggle="popover" href="/tests/v2/Input/t44.abi" target="_blank">t44.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>This variable is typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="/variables/basic#ndtset">ndtset</a> &gt; 0) since it describes from which dataset the
corresponding output variables are to be taken, as input of the present
dataset. The atomic positions and velocities are <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a> variables, for
which such a chain of calculation is useful.</p>
<p>Note that the use of <a class="wikilink codevar-wikilink" href="/variables/rlx#getxcart">getxcart</a> and  <strong>getxred</strong>  differs when <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a> and
<a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> are different from one dataset to the other.</p>
<ul>
<li>
<p>If 0, no previously computed values are used.</p>
</li>
<li>
<p>If &gt;0, the integer must correspond to the index of the dataset from which the
data are to be used as input data. It must be the index of a dataset already
computed in the SAME run.</p>
</li>
<li>
<p>If equal to -1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has yet been computed in the same run.</p>
</li>
<li>
<p>If another negative number, it indicates the number of datasets to go backward
to find the needed data. Going back beyond the first dataset is equivalent to using a null get variable.</p>
</li>
</ul>
<p>Note:  <strong>getxred</strong>  and <a class="wikilink codevar-wikilink" href="/variables/rlx#getxcart">getxcart</a> cannot be simultaneously non-zero for the
same dataset. On the other hand the use of <a class="wikilink codevar-wikilink" href="/variables/rlx#getvel">getvel</a> with  <strong>getxred</strong>  is
allowed, despite the different coordinate system.</p>
<h2 id="goprecon"><strong>goprecon</strong><a class="headerlink" href="#goprecon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Geometry Optimization PRECONditioner equations<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of geometry optimization using using ionmov=1,4,5 and 7
Hydrogen molecule inside a box of 12 Bohr
This test was created to test ionmov=5" data-toggle="popover" href="/tests/v5/Input/t02.abi" target="_blank">t02.abi</a></li>
</ul>
</details>
<p>Set the kind of preconditioner to be used for Geometry Optimization
(Note: Under development now (2011.05.20))</p>
<ul>
<li><strong>goprecon</strong>  = 0: No preconditioner</li>
<li><strong>goprecon</strong>  = [1-9]: Linear preconditioner</li>
<li><strong>goprecon</strong>  = [11-19]: Non-linear preconditioner</li>
</ul>
<h2 id="goprecprm"><strong>goprecprm</strong><a class="headerlink" href="#goprecprm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Geometry Optimization PREconditioner PaRaMeters equations<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3)<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of geometry optimization using using ionmov=1,4,5 and 7
Hydrogen molecule inside a box of 12 Bohr
This test was created to test ionmov=5" data-toggle="popover" href="/tests/v5/Input/t02.abi" target="_blank">t02.abi</a></li>
</ul>
</details>
<p>Set the parameters use by the preconditioner to be used for Geometry
Optimization
(Note: Under development now (2011.06.06))</p>
<h2 id="hmcsst"><strong>hmcsst</strong><a class="headerlink" href="#hmcsst" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Hybrid Monte Carlo Strain Step Trajectory<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> == 25<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [0/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
</details>
<p>Number of strain teps per MC trial trajectory, for the Hybrid Monte Carlo algorithm <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a>=25.</p>
<h2 id="hmctt"><strong>hmctt</strong><a class="headerlink" href="#hmctt" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Hybrid Monte Carlo Trial Trajectory<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> == 25<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [0/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
</details>
<p>Number of steps per MC trial trajectory, for the Hybrid Monte Carlo algorithm <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a>=25.</p>
<h2 id="iatcon"><strong>iatcon</strong><a class="headerlink" href="#iatcon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms in CONstraint equations<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">NO_MULTI</a>, <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>,<a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>)<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Gives the indices of the atoms appearing in each of the <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a> independent
equations constraining the motion of atoms during structural optimization or
molecular dynamics (see <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>, and <a class="wikilink codevar-wikilink" href="/variables/rlx#wtatcon">wtatcon</a>).
(Note: combined with <a class="wikilink codevar-wikilink" href="/variables/rlx#wtatcon">wtatcon</a> to give internal representation of the latter)</p>
<h2 id="iatfix"><strong>iatfix</strong><a class="headerlink" href="#iatfix" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#natfix">natfix</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#natfix">natfix</a> &gt; 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [32/1314] in all abinit tests, [6/178] in abinit tutorials</summary>
<ul>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="
brdmin with natfix.
Output also DEN, POT and GEO files" data-toggle="popover" href="/tests/fast/Input/t28.abi" target="_blank">t28.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
move with natfix.
Output also GEO files" data-toggle="popover" href="/tests/fast/Input/t29.abi" target="_blank">t29.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium,
to compare with the PSML results of t07 and t11" data-toggle="popover" href="/tests/psml/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t07.abi" target="_blank">t07.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_01.abi" target="_blank">timages_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_02.abi" target="_blank">timages_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the option ionmov=4. 8-atom Si cube, non spin-polarized.
Same test as number 22a of fast, except for the change
of ionmov, no reading of wavefunctions, iscf to 2, and different
tolerance criteria." data-toggle="popover" href="/tests/v1/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom." data-toggle="popover" href="/tests/v8/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom along one direction." data-toggle="popover" href="/tests/v8/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
</ul>
</details>
<p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a> ) of each atom which is to be held
fixed for structural optimization or molecular dynamics. The variable
 <strong>iatfix</strong>  lists those fixed in the three directions, while the variables
<a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixx">iatfixx</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixy">iatfixy</a>, and <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixz">iatfixz</a>, allow to fix some atoms along x, y
or z directions, or a combination of these.</p>
<p>WARNING: The implementation is inconsistent !! For <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> set to &ldquo;viscous&rdquo; (or <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 1),
the fixing of directions was done in cartesian coordinates, while for the other values of
<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> or for any value of <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> /= 1), it was done in reduced coordinates.
Sorry for this.</p>
<p>There is no harm in fixing one atom in the three directions using  <strong>iatfix</strong> ,
then fixing it again in other directions by mentioning it in <strong>iatfixx</strong>,
<strong>iatfixy</strong> or <strong>iatfixz</strong>.
The internal representation of these input data is done by the mean of one
variable  <strong>iatfix</strong> (3,<a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>), defined for each direction and each atom,
being 0 if the atom is not fixed along the direction, and 1 if the atom is
fixed along the direction. When some atoms are fixed along 1 or 2 directions,
the use of symmetries is restricted to symmetry operations whose (3x3)
matrices <a class="wikilink codevar-wikilink" href="/variables/basic#symrel">symrel</a> are diagonal.
If the atom manipulator is used,  <strong>iatfix</strong>  will be related to the
preprocessed set of atoms, generated by the atom manipulator. The user must
thus foresee the effect of this atom manipulator (see <a class="wikilink codevar-wikilink" href="/variables/geo#objarf">objarf</a>).</p>
<h2 id="iatfixx"><strong>iatfixx</strong><a class="headerlink" href="#iatfixx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed along the X direction<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#natfixx">natfixx</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#natfixx">natfixx</a> &gt; 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a></li>
</ul>
</details>
<p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a> ) of each atom which is to be held
fixed ALONG THE X direction for structural optimization or molecular dynamics.
The variable <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the
variables  <strong>iatfixx</strong> , <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixy">iatfixy</a>, and <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixz">iatfixz</a>, allow to fix some atoms
along x, y or z directions, or a combination of these. See the variable
<a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a> for more information.</p>
<h2 id="iatfixy"><strong>iatfixy</strong><a class="headerlink" href="#iatfixy" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed along the Y direction<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#natfixy">natfixy</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#natfixy">natfixy</a> &gt; 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a></li>
</ul>
</details>
<p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a> ) of each atom which is to be held
fixed ALONG THE Y direction for structural optimization or molecular dynamics.
The variable <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the
variables <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixx">iatfixx</a>,  <strong>iatfixy</strong> , and <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixz">iatfixz</a>, allow to fix some atoms
along x, y or z directions, or a combination of these.
See the variable <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a> for more information.</p>
<h2 id="iatfixz"><strong>iatfixz</strong><a class="headerlink" href="#iatfixz" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed along the Z direction<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#natfixz">natfixz</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#natfixz">natfixz</a> &gt; 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom along one direction." data-toggle="popover" href="/tests/v8/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a> ) of each atom which is to be held
fixed ALONG THE Z direction for structural optimization or molecular dynamics.
The variable <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the
variables <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixx">iatfixx</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixy">iatfixy</a>, and  <strong>iatfixz</strong> , allow to fix some atoms
along x, y or z directions, or a combination of these. See the variable
<a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a> for more information.</p>
<h2 id="imgmov"><strong>imgmov</strong><a class="headerlink" href="#imgmov" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> IMaGe MOVEs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate crossing barriers" data-toggle="popover" href="/topics/CrossingBarriers">topic_CrossingBarriers</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [20/1314] in all abinit tests, [5/178] in abinit tutorials</summary>
<ul>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_04.abi" target="_blank">timages_04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at a set of decreasing I-I interatomic distances
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_02.abi" target="_blank">tpsic_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test starts from a reasonable I-I interatomic distance, and search for the optimized polaron geometry, without constraint.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_03.abi" target="_blank">tpsic_03.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Fixed occupation numbers. imgmov 6 + ionmov 2 algorithm.
dataset 1 : Broyden for one image
dataset 2 : Broyden for two images, but only the first one has non-zero weight.
dataset 3 : Broyden for two images, but only the second one has non-zero weight.
dataset 4 : Broyden for two images, with weights such that the first image has still lower energy than the second
dataset 5 : Broyden for two images, with weights such that the second image has now lower energy than the first.
Geometries that mimimize the linear combination of energy at fixed Lagrange parameters can thus be found." data-toggle="popover" href="/tests/v8/Input/t20.abi" target="_blank">t20.abi</a></li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>Control the collective changes of images (see <a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a>,<a class="wikilink codevar-wikilink" href="/variables/paral#npimage">npimage</a>,
<a class="wikilink codevar-wikilink" href="/variables/rlx#dynimage">dynimage</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#tolimg">tolimg</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#istatimg">istatimg</a>, <a class="wikilink codevar-wikilink" href="/variables/files#prtvolimg">prtvolimg</a>).
Similar to <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> in spirit, although here, a population of self-consistent
calculations for possibly different (evolving) geometries is managed, while with
<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a>, only self-consistent calculation for one (evolving) geometry is managed.
In this respect the maximal number of time step for image propagation is
<a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a>, corresponding to the input variable <a class="wikilink codevar-wikilink" href="/variables/rlx#ntime">ntime</a> of the single
geometry case. Also, the stopping criterion is governed by <a class="wikilink codevar-wikilink" href="/variables/rlx#tolimg">tolimg</a>,
corresponding to the input variable <a class="wikilink codevar-wikilink" href="/variables/basic#toldfe">toldfe</a> of the single geometry case.
The stopping condition is crude: the image propagation is stopped when the
mean value (over dynamic images) of the absolute difference of total energy
(previous and current time step) is less than <a class="wikilink codevar-wikilink" href="/variables/rlx#tolimg">tolimg</a>.</p>
<p>Actually, there might be combinations of <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> and  <strong>imgmov</strong>  in which
the two mechanisms are at work. Usually, however, only one mechanism will be
activated (so, usually, either <a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a> is bigger than one OR <a class="wikilink codevar-wikilink" href="/variables/rlx#ntime">ntime</a>
is bigger than one). In order for the user to acquire a mental representation
of the interplay between <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> and  <strong>imgmov</strong> , here is a F90 pseudo-code
presenting the interplay between the different above-mentioned input
variables, as well as with the parallelism (see input variable <a class="wikilink codevar-wikilink" href="/variables/paral#npimage">npimage</a>).</p>
<div class="highlight"><pre><span></span><code><span class="w">    </span><span class="k">do </span><span class="n">itimimage</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="n">ntimimage</span>
<span class="w">      </span><span class="k">do </span><span class="n">iimage</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="n">nimage</span>
<span class="w">        </span><span class="p">(</span><span class="n">possibly</span><span class="p">,</span><span class="w"> </span><span class="n">parallelisation</span><span class="w"> </span><span class="n">over</span><span class="w"> </span><span class="k">images</span><span class="p">)</span>
<span class="w">        </span><span class="k">do </span><span class="nb">itime</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="n">ntime</span>
<span class="w">          </span><span class="n">Compute</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">forces</span><span class="w"> </span><span class="nb">and </span><span class="n">stresses</span><span class="w"> </span><span class="n">for</span><span class="w"> </span><span class="n">image</span><span class="p">(</span><span class="n">iimage</span><span class="p">)</span>
<span class="w">          </span><span class="n">Examine</span><span class="w"> </span><span class="n">whether</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">stopping</span><span class="w"> </span><span class="n">criterion</span><span class="w"> </span><span class="n">defined</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="n">tolmxf</span><span class="w"> </span><span class="k">is </span><span class="n">fulfilled</span>
<span class="w">          </span><span class="n">Predict</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">next</span><span class="w"> </span><span class="n">geometry</span><span class="w"> </span><span class="n">for</span><span class="w"> </span><span class="n">image</span><span class="p">(</span><span class="n">iimage</span><span class="p">)</span><span class="w"> </span><span class="n">using</span><span class="w"> </span><span class="n">ionmov</span>
<span class="w">        </span><span class="k">enddo</span>
<span class="k">      enddo</span>
<span class="k">      </span><span class="n">Examine</span><span class="w"> </span><span class="n">whether</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">stopping</span><span class="w"> </span><span class="n">criterion</span><span class="w"> </span><span class="n">defined</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="n">tolimg</span><span class="w"> </span><span class="k">is </span><span class="n">fulfilled</span>
<span class="w">      </span><span class="n">Predict</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">next</span><span class="w"> </span><span class="n">geometries</span><span class="w"> </span><span class="n">for</span><span class="w"> </span><span class="k">all images </span><span class="n">using</span><span class="w"> </span><span class="n">imgmov</span>
<span class="w">    </span><span class="k">enddo</span>
</code></pre></div>
<ul>
<li>= 0 &rarr; simply <strong>copy</strong> images from previous itimimage step.</li>
<li>= 1 &rarr; move images according to <strong>Steepest Descent</strong> following the (scaled) forces,
    the scaling factor being <a class="wikilink codevar-wikilink" href="/variables/rlx#fxcartfactor">fxcartfactor</a>.</li>
<li>= 2 &rarr; <strong>String Method</strong> for finding Minimal Energy Path (MEP) connecting two minima
    (see <a class="wikilink citation-wikilink" data-tippy-content="String Method for the study of rare events" data-toggle="popover" href="/theory/bibliography#weinan2002">[Weinan2002]</a>), or even two configurations that are not local minima; the algorithm variant can be selected with the <a class="wikilink codevar-wikilink" href="/variables/rlx#string_algo">string_algo</a> keyword
    (Simplified String Method by default). The solver for the Ordinary Differential Equation (ODE)
    can be selected with <a class="wikilink codevar-wikilink" href="/variables/rlx#mep_solver">mep_solver</a> (steepest-descent by default). See also <a class="wikilink codevar-wikilink" href="/variables/rlx#mep_mxstep">mep_mxstep</a> keyword.</li>
<li>= 3 &rarr; (tentatively, not yet coded) <strong>Metadynamics</strong>.</li>
<li>= 4 &rarr; (tentatively, not yet coded) <strong>Genetic Algorithm</strong>.</li>
<li>= 5 &rarr; <strong>Nudged Elastic Band (NEB)</strong> for finding Minimal Energy Path (MEP) connecting two minima;
    the algorithm variant can be selected with the <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_algo">neb_algo</a> keyword (NEB+improved tangent by default).
    The solver for the Ordinary Differential Equation (ODE) can be selected with <a class="wikilink codevar-wikilink" href="/variables/rlx#mep_solver">mep_solver</a> (steepest-descent by default).
    It is possible (optionally) to vary the vectors of the primitive cell along the path (variable-cell NEB) by using the variable <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_cell_algo">neb_cell_algo</a>.
    The spring constant connecting images along the path is defined by <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_spring">neb_spring</a>. See also <a class="wikilink codevar-wikilink" href="/variables/rlx#mep_mxstep">mep_mxstep</a> keyword.</li>
<li>= 6 &rarr; <strong>Linear Combination of Constrained DFT Energies</strong>. The images can have different electronic structure (<a class="wikilink codevar-wikilink" href="/variables/gstate#occ">occ</a> can differ),
    and their total energies are combined linearly using the factors in <a class="wikilink codevar-wikilink" href="/variables/rlx#mixesimgf">mixesimgf</a>, giving the actual total energy of the ensemble
    of constrained DFT images. The geometry is the same for all images, forces and stresses are computed, and all usual
    algorithms for MD or geometry optimization are allowed, using <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> (instead of  <strong>imgmov</strong> , this is the exception to the rule)
    and related variables.</li>
<li>= 9 or 13 &rarr; <strong>Path-Integral Molecular Dynamics</strong> (see e.g. <a class="wikilink citation-wikilink" data-tippy-content="Ab initio path integral molecular dynamics: Basic ideas" data-toggle="popover" href="/theory/bibliography#marx1996">[Marx1996]</a>).
    Will use 9 for <strong>Langevin thermostat</strong> (associated friction coefficient given by <a class="wikilink codevar-wikilink" href="/variables/rlx#friction">friction</a>)
    and 13 for <strong>Nose-Hoover thermostat chains</strong> (associated input variables are the number of thermostats in the chains,
    <a class="wikilink codevar-wikilink" href="/variables/rlx#nnos">nnos</a>, and the masses of these thermostats <a class="wikilink codevar-wikilink" href="/variables/rlx#qmass">qmass</a>). <a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a> is the Trotter number
    (no use of <a class="wikilink codevar-wikilink" href="/variables/rlx#dynimage">dynimage</a>); possible transformations of coordinates are defined by <a class="wikilink codevar-wikilink" href="/variables/rlx#pitransform">pitransform</a>;
    Fictitious masses of the atoms (possibly different from the true masses given by <a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>) can be specified by <a class="wikilink codevar-wikilink" href="/variables/rlx#pimass">pimass</a>.
    At present, it is only possible to perform calculations in the (N,V,T) ensemble (<a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0).</li>
<li>= 10 &rarr; <strong>Path-Integral Quantum Thermal Bath (PI-QTB)</strong> (see <a class="wikilink citation-wikilink" data-tippy-content="Quantum Thermal Bath for Path Integral Molecular Dynamics Simulation" data-toggle="popover" href="/theory/bibliography#brieuc2016">[Brieuc2016]</a>,<a class="wikilink citation-wikilink" data-tippy-content="Quantum Thermal Bath for Molecular Dynamics Simulation" data-toggle="popover" href="/theory/bibliography#dammak2009">[Dammak2009]</a>).
    QTB-MD is an efficient but approximate method when dealing with strongly anharmonic systems,
    while path integral molecular dynamics (PIMD) gives exact results but in a huge amount of computation time.
    The QTB and PIMD methods have been combined in order to improve the PIMD convergence
    or correct the failures of the QTB-MD technique. Warning: this option need a previoulsy generated file
    containing Gaussian random forces for each time step. Very experimental !!!</li>
</ul>
<p>No meaning for RF calculations.</p>
<h2 id="imgwfstor"><strong>imgwfstor</strong><a class="headerlink" href="#imgwfstor" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> IMaGe WaveFunction STORage<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate crossing barriers" data-toggle="popover" href="/topics/CrossingBarriers">topic_CrossingBarriers</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/dev#extrapwf">extrapwf</a> == 0 and <a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a> &gt; 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [6/1314] in all abinit tests, [3/178] in abinit tutorials</summary>
<ul>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at a set of decreasing I-I interatomic distances
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_02.abi" target="_blank">tpsic_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test starts from a reasonable I-I interatomic distance, and search for the optimized polaron geometry, without constraint.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_03.abi" target="_blank">tpsic_03.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Test the input variable imgwfstor.
In itimimage=2, the SCF cycle and the Broyden optimization are indeed immediately fulfilled." data-toggle="popover" href="/tests/v8/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Fixed occupation numbers. imgmov 6 + ionmov 2 algorithm.
dataset 1 : Broyden for one image
dataset 2 : Broyden for two images, but only the first one has non-zero weight.
dataset 3 : Broyden for two images, but only the second one has non-zero weight.
dataset 4 : Broyden for two images, with weights such that the first image has still lower energy than the second
dataset 5 : Broyden for two images, with weights such that the second image has now lower energy than the first.
Geometries that mimimize the linear combination of energy at fixed Lagrange parameters can thus be found." data-toggle="popover" href="/tests/v8/Input/t20.abi" target="_blank">t20.abi</a></li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>Govern the storage of wavefunctions at the level of the loop over images, see <a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a>.
Possible values of  <strong>imgwfstor</strong>  are 0 or 1.
If  <strong>imgwfstor</strong>  is 1, the wavefunctions for each image are stored in a big array of
size <a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a> more than the storage needed for one set of wavefunctions.
When the specific computation (optimization/SCF cycle etc) for this image is started,
the past wavefunctions are used, to speed up the computation. If  <strong>imgwfstor</strong> ==0,
the wavefunctions are reinitialised, either at random or from the initial wavefunction file (so, without
any modification to take into account the computations at the previous value of itimimage.</p>
<p>If <a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a> is large, the increase of memory need can be problematic, unless the wavefunctions
are spread over many processors, which happens when <a class="wikilink codevar-wikilink" href="/variables/paral#paral_kgb">paral_kgb</a> == 1.
For some algorithms, e.g. when some geometry optimization
is performed, <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>==2 or 5, the gain in speed of choosing  <strong>imgwfstor</strong> =1 can be quite large, e.g. two to four.
For algorithms of the molecular dynamics type, <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>==9 or 13, the expected gain is smaller.
Of course, with adequate memory resources,  <strong>imgwfstor</strong> ==1 should always be preferred.</p>
<h2 id="istatimg"><strong>istatimg</strong><a class="headerlink" href="#istatimg" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Integer governing the computation of STATic IMaGes<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a></li>
</ul>
</details>
<p>This input variable is relevant when sets of images are activated (see <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>).
Not all images might be required to evolve from one time step to the other
(see<a class="wikilink codevar-wikilink" href="/variables/rlx#dynimage">dynimage</a>): these are static images.
If  <strong>istatimg</strong>  = 0, the total energy of static images is not computed (but
static images are used to make the dynamic images evolve).
This can be useful to save CPU time.
If  <strong>istatimg</strong>  = 1, the total energy of static images is computed.</p>
<h2 id="mdtemp"><strong>mdtemp</strong><a class="headerlink" href="#mdtemp" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Molecular Dynamics TEMPeratures<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (2)<br />
<em>Default value:</em> [300, 300]<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [15/1314] in all abinit tests, [1/178] in abinit tutorials</summary>
<ul>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
He FCC solid in conventional cell (4 atoms).
Test the recursion algorithm (for high-temperature calculations).
Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t42.abi" target="_blank">t42.abi</a></li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Gold, 8 atom supercell, with PAW.
Test lotf_nitex, lotf_classic, lotf_version, lotf_nneigx.
These variables, concerning LOTF method, are taken into account only if the
enable_lotf=&quot;yes&quot; is used in configuration." data-toggle="popover" href="/tests/paral/Input/t41.abi" target="_blank">t41.abi</a></li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="Molecular dynamics for Al" data-toggle="popover" href="/tests/tutoparal/Input/tmoldyn_01.abi" target="_blank">tmoldyn_01.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Short MD traj to test restarxf -1" data-toggle="popover" href="/tests/v10/Input/t81.abi" target="_blank">t81.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Short MD traj restarting from HIST file using restarxf -1" data-toggle="popover" href="/tests/v10/Input/t82.abi" target="_blank">t82.abi</a></li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a></li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Simple test for double grid (fine/coarse) system, the cutting
radius and the initial fermi energy in recursion method
Corresponding variables: recgratio, recrcut,recefermi respectively
Corresponding code: 68_recursion
Other test: paral/tT,v5/t76" data-toggle="popover" href="/tests/v5/Input/t75.abi" target="_blank">t75.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test for an electron gas by recursion method (Vtrial is set to zero)
Corresponding variable: rectesteg
Corresponding code: 68_recursion
Other test: paral/tT,v5/t76" data-toggle="popover" href="/tests/v5/Input/t76.abi" target="_blank">t76.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom." data-toggle="popover" href="/tests/v8/Input/t21.abi" target="_blank">t21.abi</a></li>
</ul>
</details>
<p>Give the initial and final temperature of the Nose-Hoover thermostat <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_nose&rdquo;
(<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 8), Langevin dynamics <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_langevin&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 16),
Martyna thermostats <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;npt_martyna&rdquo; or &ldquo;nst_martyna&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 13), in Kelvin. This
temperature will change linearly from the initial temperature <strong>mdtemp(1)</strong> at
itime=1 to the final temperature <strong>mdtemp(2)</strong> at the end of the <a class="wikilink codevar-wikilink" href="/variables/rlx#ntime">ntime</a> timesteps.</p>
<p>In the case of the isokinetic molecular dynamics <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_isokin&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 12), <strong>mdtemp(1)</strong> allows ABINIT
to generate velocities (<a class="wikilink codevar-wikilink" href="/variables/rlx#vel">vel</a>) to start the run if they are not provided by the user or if they all vanish.
However <strong>mdtemp(2)</strong> is not used (even if it must be defined to please the parser). If some velocities are non-zero,
<strong>mdtemp</strong> is not used, the kinetic energy computed from the velocities is kept constant during the run.</p>
<h2 id="mdwall"><strong>mdwall</strong><a class="headerlink" href="#mdwall" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Molecular Dynamics WALL location<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 10000.0<br />
<em>Comment:</em> the walls are extremely far away<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a></li>
</ul>
</details>
<p>Gives the location (atomic units) of walls on which the atoms will bounce
back when <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;quenched&rdquo; or <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nve_verlet&rdquo; (also when <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 6, 7, 8 or 9).
For each cartesian direction idir=1, 2 or 3, there is a pair of walls with coordinates xcart(idir)=-wall and
xcart(idir)=rprimd(idir,idir)+wall. Supposing the particle will cross the
wall, its velocity normal to the wall is reversed, so that it bounces back.
By default, given in Bohr atomic units (1 Bohr=0.5291772108 Angstroms),
although Angstrom can be specified, if preferred, since  <strong>mdwall</strong>  has the <a class="wikilink" href="/guide/abinit#parameters">LENGTH</a> characteristics.</p>
<h2 id="mep_mxstep"><strong>mep_mxstep</strong><a class="headerlink" href="#mep_mxstep" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Minimal Energy Path search: MaXimum allowed STEP size<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">LENGTH</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.4 if <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 5,
100.0 otherwise.</p>
<p><em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
</ul>
</details>
<p>Relevant only when <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 1 (Steepest-Descent), 2 (String Method) or 5
(Nudged Elastic Band).
The optimizer used to solve the Ordinary Differential Equation (ODE) can be
constrained with a maximum allowed step size for each image. By default this
feature is only activated for Nudged Elastic Band (NEB) and the value is
inspired by <a class="wikilink citation-wikilink" data-tippy-content="Optimization methods for finding minimum energy paths" data-toggle="popover" href="/theory/bibliography#sheppard2008">[Sheppard2008]</a>.
Note that the step size is defined for each image as <em>step = SQRT[SUM(R_i dot
R_i)]</em> where the <em>R_i</em> are the positions of the atoms in the cell.</p>
<h2 id="mep_solver"><strong>mep_solver</strong><a class="headerlink" href="#mep_solver" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Minimal Energy Path ordinary differential equation SOLVER<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> None<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [4/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
</ul>
</details>
<p>Relevant only when <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 2 (String Method) or 5 (Nudged Elastic Band).
Gives the algorithm used to solve the Ordinary Differential Equation (ODE)
when searching for a Minimal Energy Path (MEP).
Possible values can be:</p>
<ul>
<li>
<p>0 &rarr; <strong>Steepest-Descent algorithm</strong> following the (scaled) forces, the scaling factor being <a class="wikilink codevar-wikilink" href="/variables/rlx#fxcartfactor">fxcartfactor</a> (forward Euler method).
Compatible with all MEP search methods.</p>
</li>
<li>
<p>1 &rarr; <strong>Quick-min optimizer</strong> following the (scaled) forces, the scaling factor being <a class="wikilink codevar-wikilink" href="/variables/rlx#fxcartfactor">fxcartfactor</a>. The &ldquo;quick minimizer&rdquo; improves upon the steepest-descent method by accelerating the system in the direction of the forces. The velocity (of the image) is projected long the force and cancelled if antiparallel to it.
Compatible only with Nudged Elastic Band (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 5).
See <a class="wikilink citation-wikilink" data-tippy-content="Optimization methods for finding minimum energy paths" data-toggle="popover" href="/theory/bibliography#sheppard2008">[Sheppard2008]</a>.</p>
</li>
<li>
<p>2 &rarr; <strong>Local Broyden-Fletcher-Goldfarb-Shanno (L-BFGS) algorithm</strong>; each image along the band is minimized with a different instance of the BFGS optimizer.
Compatible only with Nudged Elastic Band (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 5).
See <a class="wikilink citation-wikilink" data-tippy-content="Optimization methods for finding minimum energy paths" data-toggle="popover" href="/theory/bibliography#sheppard2008">[Sheppard2008]</a>.
IN <a class="wikilink" href="/guide/abinit#parameters">DEVELOP</a>PMENT - NOT RELIABLE</p>
</li>
<li>
<p>3 &rarr; <strong>Global Broyden-Fletcher-Goldfarb-Shanno (GL-BFGS) algorithm</strong>; all images along the band are minimized with a single instance of the BFGS optimizer.
Compatible only with Nudged Elastic Band (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 5).
See <a class="wikilink citation-wikilink" data-tippy-content="Optimization methods for finding minimum energy paths" data-toggle="popover" href="/theory/bibliography#sheppard2008">[Sheppard2008]</a>.
IN <a class="wikilink" href="/guide/abinit#parameters">DEVELOP</a>PMENT - NOT RELIABLE</p>
</li>
<li>
<p>4 &rarr; <strong>Fourth-order Runge-Kutta method</strong>; the images along the band are moved every four steps (1 &lt;=istep&lt;=<a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a>) following the Runge-Kutta algorithm, the time step being <a class="wikilink codevar-wikilink" href="/variables/rlx#fxcartfactor">fxcartfactor</a>.
Compatible only with Simplified String Method (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 2 and
<a class="wikilink codevar-wikilink" href="/variables/rlx#string_algo">string_algo</a> = 1 or 2).
See <a class="wikilink citation-wikilink" data-tippy-content="Simplified and improved string method for computing the minimum energy paths in barrier-crossing events" data-toggle="popover" href="/theory/bibliography#weinan2007">[Weinan2007]</a>.</p>
</li>
</ul>
<p>All of the optimizers can be constrained with a maximum allowed step size for
each image; see <a class="wikilink codevar-wikilink" href="/variables/rlx#mep_mxstep">mep_mxstep</a>. This is by default the case of the Nudged
Elastic Band (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 5).</p>
<h2 id="mixesimgf"><strong>mixesimgf</strong><a class="headerlink" href="#mixesimgf" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> MIXing Electronic Structure IMAGE Factors<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate crossing barriers" data-toggle="popover" href="/topics/CrossingBarriers">topic_CrossingBarriers</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a>)<br />
<em>Default value:</em> None<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [5/1314] in all abinit tests, [3/178] in abinit tutorials</summary>
<ul>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at a set of decreasing I-I interatomic distances
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_02.abi" target="_blank">tpsic_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test starts from a reasonable I-I interatomic distance, and search for the optimized polaron geometry, without constraint.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_03.abi" target="_blank">tpsic_03.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Fixed occupation numbers. imgmov 6 + ionmov 2 algorithm.
dataset 1 : Broyden for one image
dataset 2 : Broyden for two images, but only the first one has non-zero weight.
dataset 3 : Broyden for two images, but only the second one has non-zero weight.
dataset 4 : Broyden for two images, with weights such that the first image has still lower energy than the second
dataset 5 : Broyden for two images, with weights such that the second image has now lower energy than the first.
Geometries that mimimize the linear combination of energy at fixed Lagrange parameters can thus be found." data-toggle="popover" href="/tests/v8/Input/t20.abi" target="_blank">t20.abi</a></li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>Used in the algorithm Linear Combination of Constrained DFT Energies, that is, when <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>==6.</p>
<p>This array gives, for each one of the <a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a> images, the factor
by which the total energies for systems with same geometry but different electronic structures (occupation numbers) are linearly combined.
The sum of these factors must equal 1.</p>
<h2 id="moldyn"><strong>moldyn</strong><a class="headerlink" href="#moldyn" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> MOLecular DYNamics<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> string<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> none<br />
<em>Added in version:</em> 10.3  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [1/178] in abinit tutorials</summary>
<ul>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="Molecular dynamics for Al" data-toggle="popover" href="/tests/tutoparal/Input/tmoldyn_01.abi" target="_blank">tmoldyn_01.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
</ul>
</details>
<p>Choice of algorithm for the molecular dynamics simulation, and possibly changes of cell shape and size (see <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>).</p>
<ul>
<li>
<p>&ldquo;none&rdquo; &rarr; Do not move ions (<strong>default behaviour</strong>)</p>
</li>
<li>
<p>&ldquo;nve_verlet&rdquo; &rarr; Molecular dynamics using the Verlet algorithm, see <a class="wikilink citation-wikilink" data-tippy-content="Computer Simulation of Liquids" data-toggle="popover" href="/theory/bibliography#allen1987a">[Allen1987a]</a> p 81].
    The only related parameter is the time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>).
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
    <strong>Related variables:</strong> time step <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>, index of atoms fixed <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a></p>
</li>
<li>
<p>&ldquo;nvt_isokin&rdquo; &rarr; Isokinetic ensemble molecular dynamics.
    The equations of motion of the ions in contact with a thermostat are solved with the
    algorithm proposed in <a class="wikilink citation-wikilink" data-tippy-content="Operator-splitting integrators for constant-temperature molecular dynamics" data-toggle="popover" href="/theory/bibliography#zhang1997">[Zhang1997]</a>, as worked out in <a class="wikilink citation-wikilink" data-tippy-content="Algorithms and novel applications based on the isokinetic ensemble. I. Biophysical and path integral molecular dynamics" data-toggle="popover" href="/theory/bibliography#minary2003">[Minary2003]</a>.
    The conservation of the kinetic energy is obtained within machine precision at each step.
    As in <a class="wikilink citation-wikilink" data-tippy-content="The isothermal/isobaric molecular dynamics ensemble" data-toggle="popover" href="/theory/bibliography#evans1983">[Evans1983]</a>, when there is no fixing of atoms, the number of degrees of freedom in which the
    microscopic kinetic energy is hosted is 3*<a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a> - 4.
    Indeed, the total kinetic energy is constrained, which accounts for
    minus one degree of freedom (also mentioned in <a class="wikilink citation-wikilink" data-tippy-content="Algorithms and novel applications based on the isokinetic ensemble. I. Biophysical and path integral molecular dynamics" data-toggle="popover" href="/theory/bibliography#minary2003">[Minary2003]</a>), but also there are three degrees of freedom
    related to the total momentum in each direction, that cannot be counted as microscopic degrees of freedom, since the
    total momentum is also preserved (but this is not mentioned in <a class="wikilink citation-wikilink" data-tippy-content="Algorithms and novel applications based on the isokinetic ensemble. I. Biophysical and path integral molecular dynamics" data-toggle="popover" href="/theory/bibliography#minary2003">[Minary2003]</a>).
    When some atom is fixed in one or more direction,
    e.g. using <a class="wikilink codevar-wikilink" href="/variables/rlx#natfix">natfix</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#natfixx">natfixx</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#natfixy">natfixy</a>, or <a class="wikilink codevar-wikilink" href="/variables/rlx#natfixz">natfixz</a>, the number of degrees of freedom is decreased accordingly,
    albeit taking into account that the total momentum is not preserved
    anymore (e.g. fixing the position of one atom gives 3*natom-4, like in the non-fixed case).
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
    <strong>Related variables:</strong> time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>) and the first temperature in <a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a> in case
    the velocities <a class="wikilink codevar-wikilink" href="/variables/rlx#vel">vel</a> are not initialized, or all initialized to zero.</p>
</li>
<li>
<p>&ldquo;nvt_nose&rdquo; &rarr; Isothermal/isochoric ensemble. The equations of motion of the ions in contact with a thermostat
    with the algorithm proposed in <a class="wikilink citation-wikilink" data-tippy-content="Explict reversible integrators for extended systems dynamics" data-toggle="popover" href="/theory/bibliography#martyna1996">[Martyna1996]</a>.
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
    <strong>Related variables:</strong> The time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>), the temperatures (<a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a>),
    the number of thermostats (<a class="wikilink codevar-wikilink" href="/variables/rlx#nnos">nnos</a>), and the masses of thermostats (<a class="wikilink codevar-wikilink" href="/variables/rlx#qmass">qmass</a>).</p>
</li>
<li>
<p>&ldquo;nvt_langevin&rdquo; &rarr; Langevin molecular dynamics. The equations of motion of the ions are described
    with the algorithms proposed in <a class="wikilink citation-wikilink" data-tippy-content="Langevin dynamics in constant pressure extended systems" data-toggle="popover" href="/theory/bibliography#quigley2004">[Quigley2004]</a> and <a class="wikilink citation-wikilink" data-tippy-content="Constant pressure Langevin dynamics: theory and application" data-toggle="popover" href="/theory/bibliography#quigley2005">[Quigley2005]</a>.
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
    <strong>Related variables:</strong> time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>), temperatures (<a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a>) and friction coefficient (<a class="wikilink codevar-wikilink" href="/variables/rlx#friction">friction</a>).</p>
</li>
<li>
<p>&ldquo;npt_langevin&rdquo; &rarr; Langevin molecular dynamics at constant pressure. The equations of motion of the ions are described
    with the algorithms proposed in <a class="wikilink citation-wikilink" data-tippy-content="Langevin dynamics in constant pressure extended systems" data-toggle="popover" href="/theory/bibliography#quigley2004">[Quigley2004]</a> and <a class="wikilink citation-wikilink" data-tippy-content="Constant pressure Langevin dynamics: theory and application" data-toggle="popover" href="/theory/bibliography#quigley2005">[Quigley2005]</a>.
    The mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>) must be given in addition.
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> Yes (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 2 only)
    <strong>Related variables:</strong> time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>), temperatures (<a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a>), friction coefficient (<a class="wikilink codevar-wikilink" href="/variables/rlx#friction">friction</a>),
    friction coefficient of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#frictionbar">frictionbar</a>), and the mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>).</p>
</li>
<li>
<p>&ldquo;npt_martyna&rdquo; &rarr; Isothermal/isobaric ensemble. The equations of motion of the ions in contact with a thermostat
    and a barostat are solved with the algorithm proposed in <a class="wikilink citation-wikilink" data-tippy-content="Explict reversible integrators for extended systems dynamics" data-toggle="popover" href="/theory/bibliography#martyna1996">[Martyna1996]</a>.
    The mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>) must be given in addition.
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> Yes (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 1 only)
    <strong>Related variables:</strong> The time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>), the temperatures (<a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a>),
    the number of thermostats (<a class="wikilink codevar-wikilink" href="/variables/rlx#nnos">nnos</a>), and the masses of thermostats (<a class="wikilink codevar-wikilink" href="/variables/rlx#qmass">qmass</a>),
    and the mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>).</p>
</li>
<li>
<p>&ldquo;nst_martyna&rdquo; &rarr; Isothermal/isobaric ensemble. The equations of motion of the ions in contact with a thermostat
    and a barostat are solved with the algorithm proposed in <a class="wikilink citation-wikilink" data-tippy-content="Explict reversible integrators for extended systems dynamics" data-toggle="popover" href="/theory/bibliography#martyna1996">[Martyna1996]</a>.
    The mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>) must be given in addition.
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> Yes (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 2 only)
    <strong>Related variables:</strong> The time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>), the temperatures (<a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a>),
    the number of thermostats (<a class="wikilink codevar-wikilink" href="/variables/rlx#nnos">nnos</a>), and the masses of thermostats (<a class="wikilink codevar-wikilink" href="/variables/rlx#qmass">qmass</a>),
    and the mass of the barostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#bmass">bmass</a>).</p>
</li>
<li>
<p>&ldquo;nve_velverlet&rdquo; &rarr; Simple constant energy molecular dynamics using
    the velocity Verlet symplectic algorithm (second order), see <a class="wikilink citation-wikilink" data-tippy-content="Geometric numerical integration illustrated by the Stomer-Verlet Method" data-toggle="popover" href="/theory/bibliography#hairer2003">[Hairer2003]</a>.
    The only related parameter is the time step (<a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a>).
    <strong>Purpose:</strong> Molecular dynamics
    <strong>Cell optimization:</strong> No (Use <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 0 only)
    <strong>Related variables:</strong> time step <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a></p>
</li>
</ul>
<h2 id="natcon"><strong>natcon</strong><a class="headerlink" href="#natcon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of AToms in CONstraint equations<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">NO_MULTI</a>, <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>)<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Gives the number of atoms appearing in each of the <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a> independent
equations constraining the motion of atoms during structural optimization or
molecular dynamics (see <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#iatcon">iatcon</a>, and <a class="wikilink codevar-wikilink" href="/variables/rlx#wtatcon">wtatcon</a>).</p>
<h2 id="natfix"><strong>natfix</strong><a class="headerlink" href="#natfix" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Comment:</em> (no atoms held fixed)<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [33/1314] in all abinit tests, [6/178] in abinit tutorials</summary>
<ul>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="
8-atom Si cube to set up brdmin and MD runs with natfix.
Output also DEN, POT and GEO files" data-toggle="popover" href="/tests/fast/Input/t27.abi" target="_blank">t27.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
brdmin with natfix.
Output also DEN, POT and GEO files" data-toggle="popover" href="/tests/fast/Input/t28.abi" target="_blank">t28.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
move with natfix.
Output also GEO files" data-toggle="popover" href="/tests/fast/Input/t29.abi" target="_blank">t29.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium,
to compare with the PSML results of t07 and t11" data-toggle="popover" href="/tests/psml/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t07.abi" target="_blank">t07.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_01.abi" target="_blank">timages_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_02.abi" target="_blank">timages_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the option ionmov=4. 8-atom Si cube, non spin-polarized.
Same test as number 22a of fast, except for the change
of ionmov, no reading of wavefunctions, iscf to 2, and different
tolerance criteria." data-toggle="popover" href="/tests/v1/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom." data-toggle="popover" href="/tests/v8/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom along one direction." data-toggle="popover" href="/tests/v8/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
</ul>
</details>
<p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>) which are to be held fixed
during a structural optimization or molecular dynamics.
When  <strong>natfix</strong>  &gt; 0,  <strong>natfix</strong>  entries should be provided in array <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfix">iatfix</a>
.</p>
<h2 id="natfixx"><strong>natfixx</strong><a class="headerlink" href="#natfixx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed along the X direction<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a></li>
</ul>
</details>
<p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>) which are to be held fixed
along the X direction during a structural optimization or molecular dynamics.
When  <strong>natfixx</strong>  &gt; 0,  <strong>natfixx</strong>  entries should be provided in array <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixx">iatfixx</a>.</p>
<h2 id="natfixy"><strong>natfixy</strong><a class="headerlink" href="#natfixy" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed along the Y direction<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a></li>
</ul>
</details>
<p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>) which are to be held fixed
along the Y direction during a structural optimization or molecular dynamics.
When  <strong>natfixy</strong>  &gt; 0,  <strong>natfixy</strong>  entries should be provided in array <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixy">iatfixy</a></p>
<h2 id="natfixz"><strong>natfixz</strong><a class="headerlink" href="#natfixz" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed along the Z direction<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : optimize the geometry with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=2
Permutes the three axes x,y,z in three datasets.
Since the initialization is different, the three datasets deliver slightly different
results, though quite close to each others. Should be identical at convergence." data-toggle="popover" href="/tests/v1/Input/t85.abi" target="_blank">t85.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
SiH4 : molecular dynamics with partially constrained atomic positions.
1 special k-point, in a box 8x8x8 (too small), and 8Ha cut-off.
Test ionmov=1" data-toggle="popover" href="/tests/v1/Input/t86.abi" target="_blank">t86.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom along one direction." data-toggle="popover" href="/tests/v8/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>) which are to be held fixed
along the Z direction during a structural optimization or molecular dynamics.
When  <strong>natfixz</strong>  &gt; 0,  <strong>natfixz</strong>  entries should be provided in array <a class="wikilink codevar-wikilink" href="/variables/rlx#iatfixz">iatfixz</a>.</p>
<h2 id="nconeq"><strong>nconeq</strong><a class="headerlink" href="#nconeq" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of CONstraint EQuations<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">NO_MULTI</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Gives the number of independent equations constraining the motion of atoms
during structural optimization or molecular dynamics (see <a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#iatcon">iatcon</a>, and <a class="wikilink codevar-wikilink" href="/variables/rlx#wtatcon">wtatcon</a>).</p>
<h2 id="neb_algo"><strong>neb_algo</strong><a class="headerlink" href="#neb_algo" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Nudged Elastic Band ALGOrithm<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 5<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
</ul>
</details>
<p>Gives the variant of the NEB method used. Possible values can be:</p>
<ul>
<li>
<p>0 &rarr; <strong>Original NEB method</strong>.
See <a class="wikilink citation-wikilink" data-tippy-content="Classical and Quantum Dynamics in Condensed Phase Simulations" data-toggle="popover" href="/theory/bibliography#berne1998">[Berne1998]</a> pp. 385-404</p>
</li>
<li>
<p>1 &rarr; <strong>NEB + improved tangent</strong>.
The Improved Tangent Method builds on the NEB with an improved estimate of the
tangent direction and a resulting change of the component of the spring force
acting on the images.
See <a class="wikilink citation-wikilink" data-tippy-content="Improved tangent estimate in the nudged elastic band method for finding minimum energy paths and saddle points" data-toggle="popover" href="/theory/bibliography#henkelman2000">[Henkelman2000]</a></p>
</li>
<li>
<p>2 &rarr; <strong>Climbing-Image NEB (CI-NEB)</strong>.
The CI-NEB method constitutes a small modification to the NEB method.
Information about the shape of the MEP is retained, but a rigorous convergence
to a saddle point is also obtained. By default the spring constants are
variable (see <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_spring">neb_spring</a>). As the image with the highest energy has to be
identified, the calculation begins with several iterations of the standard NEB
algorithm. The effective CI-NEB begins at the <a class="wikilink codevar-wikilink" href="/variables/rlx#cineb_start">cineb_start</a> iteration.
See <a class="wikilink citation-wikilink" data-tippy-content="A climbing image nudged elastic band method for finding saddle points and minimum energy paths" data-toggle="popover" href="/theory/bibliography#henkelman2000a">[Henkelman2000a]</a></p>
</li>
</ul>
<p>Note that, in all cases, it is possible to define the value of the spring
constant connecting images with <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_spring">neb_spring</a>, keeping it constant or
allowing it to vary between 2 values (to have higher resolution close to the saddle point).<br />
All these methods are also compatible with the variable cell NEB (evolution of lattice vectors along the path), which can be applied using <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_cell_algo">neb_cell_algo</a>.</p>
<h2 id="neb_cell_algo"><strong>neb_cell_algo</strong><a class="headerlink" href="#neb_cell_algo" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Nudged Elastic Band variable CELL ALGOrithm<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 5<br />
<em>Added in version:</em> v10  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
</ul>
</details>
<p>This variable must be enabled to perform Variable Cell Nudged Elastic Band (NEB) calculations,
i.e. to allow the primitive cell vectors evolve along the path.<br />
The possible values are:</p>
<ul>
<li>
<p>0 &rarr; <strong>No cell evolution</strong> (default value)</p>
</li>
<li>
<p>1 &rarr; <strong>Generalized Solid-State NEB (GSS-NEB)</strong><br />
  Implements the method described in <a class="wikilink citation-wikilink" data-tippy-content="A generalized solid-state nudged elastic band method" data-toggle="popover" href="/theory/bibliography#sheppard2012">[Sheppard2012]</a>.</p>
</li>
<li>
<p>2 &rarr; <strong>Variable-Cell NEB (VC-NEB)</strong><br />
  Implements the method described in <a class="wikilink citation-wikilink" data-tippy-content="Variable cell nudged elastic band method for studying solidsolid structural phase transitions" data-toggle="popover" href="/theory/bibliography#qian2013">[Qian2013]</a>.<br />
    WARNING: Not yet usable! (to be fixed)</p>
</li>
</ul>
<blockquote>
<p>Note: variable-cell NEB is only compatible with the steepest-descent
Minimum Energy Path search algorithm (<a class="wikilink codevar-wikilink" href="/variables/rlx#mep_solver">mep_solver</a>=0).</p>
</blockquote>
<h2 id="neb_spring"><strong>neb_spring</strong><a class="headerlink" href="#neb_spring" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Nudged Elastic Band: SPRING constant<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (2)<br />
<em>Default value:</em> [0.02, 0.15] Hartree/Bohr^2 if <a class="wikilink codevar-wikilink" href="/variables/rlx#neb_algo">neb_algo</a> == 2,
[0.05, 0.05] Hartree/Bohr^2 otherwise.</p>
<p><em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> == 5<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
</ul>
</details>
<p>Gives the minimal and maximal values (in Hartree/Bohr^2) of the spring constant connecting images
for the NEB method.
In the standard &ldquo;Nudged Elastic Band&rdquo; method, the spring constant is constant
along the path, but, in order to have higher resolution close to the saddle
point, it can be better to have stronger springs close to it.
See <a class="wikilink citation-wikilink" data-tippy-content="A climbing image nudged elastic band method for finding saddle points and minimum energy paths" data-toggle="popover" href="/theory/bibliography#henkelman2000a">[Henkelman2000a]</a></p>
<h2 id="nimage"><strong>nimage</strong><a class="headerlink" href="#nimage" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of IMAGEs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [23/1314] in all abinit tests, [5/178] in abinit tutorials</summary>
<ul>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_04.abi" target="_blank">timages_04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Test the input variable imgwfstor.
In itimimage=2, the SCF cycle and the Broyden optimization are indeed immediately fulfilled." data-toggle="popover" href="/tests/v8/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Fixed occupation numbers. imgmov 6 + ionmov 2 algorithm.
dataset 1 : Broyden for one image
dataset 2 : Broyden for two images, but only the first one has non-zero weight.
dataset 3 : Broyden for two images, but only the second one has non-zero weight.
dataset 4 : Broyden for two images, with weights such that the first image has still lower energy than the second
dataset 5 : Broyden for two images, with weights such that the second image has now lower energy than the first.
Geometries that mimimize the linear combination of energy at fixed Lagrange parameters can thus be found." data-toggle="popover" href="/tests/v8/Input/t20.abi" target="_blank">t20.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
</ul>
</details>
<p>Give the number of images (or replicas) of the system, for which the forces
and stresses might be computed independently, in the context of string
or NEB method, genetic algorithm, hyperdynamics, Path-Integral Molecular
Dynamics, linear combination of images, pSIC, genetic algorithm, etc, depending on the value of <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>). Related input variables:
<a class="wikilink codevar-wikilink" href="/variables/rlx#dynimage">dynimage</a>, <a class="wikilink codevar-wikilink" href="/variables/paral#npimage">npimage</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a> and <a class="wikilink codevar-wikilink" href="/variables/files#prtvolimg">prtvolimg</a>.
If  <strong>nimage</strong> &gt;1, the default choice for printing many files is set to zero, and the user might
want to manually reestablish the printing, using, e.g. <a class="wikilink codevar-wikilink" href="/variables/files#prtgsr">prtgsr</a>, <a class="wikilink codevar-wikilink" href="/variables/files#prtwf">prtwf</a>, <a class="wikilink codevar-wikilink" href="/variables/gstate#prtebands">prtebands</a>, <a class="wikilink codevar-wikilink" href="/variables/files#prteig">prteig</a>, etc.</p>
<p>Images might differ by the position of atoms in the unit cell, their velocity,
as well as by their cell geometry. The following input variables might be used
to define the images:</p>
<ul>
<li><a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/basic#angdeg">angdeg</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/gstate#cellcharge">cellcharge</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/paw#dmatpawu">dmatpawu</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/paw#jpawu">jpawu</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/gstate#mixalch">mixalch</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/gstate#occ">occ</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/basic#scalecart">scalecart</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/paw#upawu">upawu</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/rlx#vel">vel</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/rlx#vel_cell">vel_cell</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/basic#xcart">xcart</a></li>
<li><a class="wikilink codevar-wikilink" href="/variables/basic#xred">xred</a></li>
</ul>
<p>These input variables, non-modified, will be used to define the image with
index 1. For the image with the last index, the input file might specify the
values of such input variables, appended with &ldquo;_lastimg&rdquo;, e.g.:</p>
<ul>
<li>acell_lastimg</li>
<li>rprim_lastimg</li>
<li>xcart_lastimg</li>
<li>&hellip;</li>
</ul>
<p>By default, these values will be interpolated linearly to define values for
the other images, unless there exist specific values for some images, for
which the string &ldquo;last&rdquo; has to be replaced by the index of the image, e.g. for
the image number 4:</p>
<ul>
<li>acell_4img</li>
<li>rprim_4img</li>
<li>xcart_4img</li>
<li>&hellip;</li>
</ul>
<p>It is notably possible to specify the starting point and the end point of the
path (of images), while specifying intermediate points.</p>
<p>It usually happens that the images do not have the same symmetries and space
groups. ABINIT has not been designed to use different sets of symmetries for
different images. ABINIT will use the symmetry set and space group of the image
number 2, that is expected to have a low number of symmetries. This might lead
to erroneous calculations, in case some image has even less symmetry. By
contrast, there is no problem if some other image has more symmetries than
those of the second image.</p>
<h2 id="nnos"><strong>nnos</strong><a class="headerlink" href="#nnos" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of NOSe masses<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
</ul>
</details>
<p>Gives the number of thermostats in the chain of oscillators thermostats as proposed in <a class="wikilink citation-wikilink" data-tippy-content="Explict reversible integrators for extended systems dynamics" data-toggle="popover" href="/theory/bibliography#martyna1996">[Martyna1996]</a>.
The thermostat chains can be used either to performMolecular Dynamics (MD) (<a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_nose&rdquo;, &ldquo;npt_martyna&rdquo; or &ldquo;nst_martyna&rdquo;)
or to perform Path Integral Molecular Dynamics (PIMD) (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 13).
The mass of these thermostats is given by <a class="wikilink codevar-wikilink" href="/variables/rlx#qmass">qmass</a>.</p>
<h2 id="noseinert"><strong>noseinert</strong><a class="headerlink" href="#noseinert" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> NOSE thermostat INERTia factor<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100000<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> == 8<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a></li>
</ul>
</details>
<p>Give the inertia factor WT of the Nose-Hoover thermostat (when <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 8),
in atomic units of weight*length2, that is (electron mass)*(Bohr)2. The
equations of motion are: MI d2RI/dt2= FI - dX/dt MI dRI/dt and WT d2X/dt2=
Sum(I) MI (dRI/dt)2 - 3NkBT where I represent each nucleus, MI is the mass of
each nucleus (see <a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>), RI is the coordinate of each nucleus (see
<a class="wikilink codevar-wikilink" href="/variables/basic#xcart">xcart</a>), dX/dt is a dynamical friction coefficient, and T is the
temperature of the thermostat (see <a class="wikilink codevar-wikilink" href="/variables/rlx#mdtemp">mdtemp</a>).</p>
<h2 id="ntime"><strong>ntime</strong><a class="headerlink" href="#ntime" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of TIME steps<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform Real-Time TDDFT calculations" data-toggle="popover" href="/topics/RTTDDFT">topic_RTTDDFT</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0 if ionmvov == 0, set to 1000 if ionvmov != 0 and imgmov != 0 and the variable is not specified.<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [158/1314] in all abinit tests, [23/178] in abinit tutorials</summary>
<ul>
<li>bigdft:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule with geometry optimisation using BFGS (test the reformating part of the code)." data-toggle="popover" href="/tests/bigdft/Input/t10.abi" target="_blank">t10.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2O in isolated boundary conditions (plane waves, NC), try tu use Damiens Poisson cutoff.
Water molecule with HGH pseudo-potentials. This checks that a negative
value in rhor will not make the Poisson solver crashes on XC computation." data-toggle="popover" href="/tests/bigdft/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
<li>built-in:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule, determination of the H-H distance by the Broyden algorithm." data-toggle="popover" href="/tests/built-in/Input/testin_fast.abi" target="_blank">testin_fast.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Ca atom with PAW, writing a ETSF_IO wavefunction file." data-toggle="popover" href="/tests/built-in/Input/testin_etsf_io.abi" target="_blank">testin_etsf_io.abi</a> &hellip;</li>
<li>etsf_io:  <a class="wikilink abifile-wikilink" data-tippy-content="Ca atom with PAW, writing a ETSF_IO wavefunction file.
Very quick built-in test, to check that ETSF_IO is installed." data-toggle="popover" href="/tests/etsf_io/Input/t00.abi" target="_blank">t00.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ca atom in PAW. Test the output of density in PAW formalism with ETSF_IO.
Also test the output of the wavefunctions in PAW formalism with ETSF_IO." data-toggle="popover" href="/tests/etsf_io/Input/t09.abi" target="_blank">t09.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Si2 molecule, static, spin-polarized.
Same system as test 49 of v2, except that nctime is non-zero.
Test ionmov=6 (Verlet) as well as ionmov=7." data-toggle="popover" href="/tests/etsf_io/Input/t21.abi" target="_blank">t21.abi</a> &hellip;</li>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule, determination of the H-H distance by the Broyden algorithm.
Very quick built-in test, to check that ABINIT is working." data-toggle="popover" href="/tests/fast/Input/t00.abi" target="_blank">t00.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 diatomic molecule, static, to check accurate forces." data-toggle="popover" href="/tests/fast/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Conduct relaxation of molecule starting from wf.19, using ionmov=2. (Broyden method)." data-toggle="popover" href="/tests/fast/Input/t20.abi" target="_blank">t20.abi</a> &hellip;</li>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu/Input/t04.abi" target="_blank">t04.abi</a> &hellip;</li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu_omp/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule NON-spin-polarized, in a big box, without tetragonal symmetry, in order to avoid
spurious degeneracy effects coupled to occupation numbers rapidly changing.
GPU
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization." data-toggle="popover" href="/tests/gpu_omp/Input/t28.abi" target="_blank">t28.abi</a> &hellip;</li>
<li>libxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
N2 molecule non-spin-polarized, in a big box.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with underconverged ecut (ecut 18) and box size (5 4 4 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation  (ecut 18, 5 4 4 box) : 1.101, 1.084, 1.085 Angstrom
Better  calculation  (ecut 22, 6 5 5 box) : 1.102, 1.086, 1.086 Angstrom
ABINIT Arnardi report (ecut acell unknown): 1.103, 1.091, 1.092 Angstrom
VASP Arnardi report (ecut acell unknown)  : 1.103, 1.089,  NA   Angstrom
Experimental : 1.098 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t67.abi" target="_blank">t67.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule spin-polarized, in a big box.
Norm conserving : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with underconverged ecut (ecut 15) and box size (4 3 3 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation  (ecut 13, 5 4 4 box) : 1.138, 1.130, 1.131 Angstrom
Better  calculation  (ecut 18, 7 6 6 box) : 1.135, 1.124, 1.125 Angstrom
ABINIT Arnardi report (ecut acell unknown): 1.136, 1.124, 1.124 Angstrom
VASP Arnardi report (ecut acell unknown)  : 1.136, 1.124,  NA   Angstrom
Experimental : 1.120 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t68.abi" target="_blank">t68.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule spin-polarized, in a big box.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with slightly better converged ecut (ecut 15) and box size (6 5 5 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation (ecut 13, 5 4 4 box)     : 1.130, 1.121, 1.122 Angstrom
Better calculation (ecut 15, 6 5 5 box)      : 1.136, 1.125, 1.126 Angstrom
Even better calculation (ecut 18, 7 6 6 box) : 1.138, 1.126, 1.127 Angstrom
ABINIT Arnardi report (ecut acell unknown)   : 1.136, 1.124, 1.124 Angstrom
VASP Arnardi report (ecut acell unknown)     : 1.136, 1.124,  NA   Angstrom
Experimental : 1.120 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t69.abi" target="_blank">t69.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si, Bulk, 2 atoms , paral_kgb.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/mpiio/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si, Bulk, 2 atoms , paral_kgb.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/paral/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium,
to compare with the PSML results of t07 and t11" data-toggle="popover" href="/tests/psml/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t07.abi" target="_blank">t07.abi</a> &hellip;</li>
<li>rttddft_suite:  <a class="wikilink abifile-wikilink" data-tippy-content="
FCC Ni, nonmagnetic, GGA, PAW.
Tests RT-TDDFT using ER and EMR propagator in PAW." data-toggle="popover" href="/tests/rttddft_suite/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
FCC Al, GGA, NC.
Tests RT-TDDFT using ER and EMR propagator in NC." data-toggle="popover" href="/tests/rttddft_suite/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
FCC Al, GGA, NC.
Tests RT-TDDFT restart capabilities." data-toggle="popover" href="/tests/rttddft_suite/Input/t03.abi" target="_blank">t03.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=14 (finite reduced electric field calculation,relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_121.abi" target="_blank">tsv6_121.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=16 (finite reduced electric displacement field calculation,
relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_122.abi" target="_blank">tsv6_122.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=4 (finite electric field calculation, relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_123.abi" target="_blank">tsv6_123.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_01.abi" target="_blank">timages_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_02.abi" target="_blank">timages_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Molecular dynamics for Al" data-toggle="popover" href="/tests/tutoparal/Input/tmoldyn_01.abi" target="_blank">tmoldyn_01.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Finite difference calculation of the clamped-ion
piezoelectric constants of AlP" data-toggle="popover" href="/tests/tutorespfn/Input/tpolarization_5.abi" target="_blank">tpolarization_5.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file will optimize automatically the interatomic distance" data-toggle="popover" href="/tests/tutorial/Input/tbase1_3.abi" target="_blank">tbase1_3.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom." data-toggle="popover" href="/tests/tutorial/Input/tbase2_1.abi" target="_blank">tbase2_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom.
Here, a double loop has been used." data-toggle="popover" href="/tests/tutorial/Input/tbase2_2.abi" target="_blank">tbase2_2.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the option ionmov=4. 8-atom Si cube, non spin-polarized.
Same test as number 22a of fast, except for the change
of ionmov, no reading of wavefunctions, iscf to 2, and different
tolerance criteria." data-toggle="popover" href="/tests/v1/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the option ionmov=4. Si2 molecule, spin-polarized.
Same test as number 16 of fast, except for the change
of ionmov, no reading of wavefunctions, iscf to 2, and different
tolerance criteria.

Cannot be executed in parallel:
--- !ERROR
message: |
the number of bands in the spin up case must be equal to
the number of bands in the spin down case.
This is not the case for the k point number : 1
The number of bands spin up and down are : 5 3 Action : change nband, or use the sequential
version of ABINIT.
src_file: chkinp.F90
src_line: 1181
..." data-toggle="popover" href="/tests/v1/Input/t45.abi" target="_blank">t45.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Si2 molecule, spin-polarized. Take the wavefunctions from test 45,
and converge them further for initialisation of test 47.
Cannot be executed in parallel:
--- !ERROR
message: |
the number of bands in the spin up case must be equal to
the number of bands in the spin down case.
This is not the case for the k point number : 1
The number of bands spin up and down are : 5 3 Action : change nband, or use the sequential
version of ABINIT.
src_file: chkinp.F90
src_line: 1181
..." data-toggle="popover" href="/tests/v1/Input/t46.abi" target="_blank">t46.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the post identification of more symmetric structures obtained after geometry optimization
Consider diamond and graphene, with low ecut and number of k points." data-toggle="popover" href="/tests/v10/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Compare cprj_in_memory=1 with cprj_in_memory=0 (lobpcg2 and chebfi2) for cell optimization. PAW,istwfk=1." data-toggle="popover" href="/tests/v10/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Arsenic in rhombohedral structure.
Optimization of unit cell size and shape.
With 4x4x4 k point grid, ecut 3.0, occopt 4 and tsmear 0.06d0,
one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277
These input parameters are too small, but still give a realistic
geometry, since with the much better parameters
12x12x12 k point grid, ecut 12.0, occopt 4 and tsmear 0.04d0,
one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296, while
the experimental values are
acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-toggle="popover" href="/tests/v2/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box.
Comparison of the modified Broyden algorithm (ionmov=3) with the
original one (ionmov=2). Start with different values
of xcart, from 0.6 to 1.1, by step of 0.1 . The number of Broyden
steps needed to reach acceptable residual forces
with the ionmov=3 algorithm are : 3, 2, 2, 3, 3, 4, while
with the ionmov=2 algorithms, one get : 4, 3, 2, 5, and then,
either the algorithm does not converge within 8 steps, or it
converges to a saddle point of the energy !
This test was hard to make portable. This is why the tolerance
for fldiff is very large." data-toggle="popover" href="/tests/v2/Input/t48.abi" target="_blank">t48.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Si2 molecule, static, spin-polarized.
Same system as test 17 of fast, except lower ecut.
Test ionmov=6 (Verlet) as well as ionmov=7.
Cannot be executed in parallel:
--- !ERROR
message: |
the number of bands in the spin up case must be equal to
the number of bands in the spin down case.
This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3
Action : change nband, or use the sequential version of ABINIT.
src_file: chkinp.F90
src_line: 1181" data-toggle="popover" href="/tests/v2/Input/t49.abi" target="_blank">t49.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
BCC Tantalum (1 atoms per unit cell),
using the HGH pseudopotential, within LDA.
Fixed cell dimensions: 2 slightly different
cell sizes.
Use only one k point. Test whether the stress
is correctly given, even with spin-orbit coupling.
Difference of total energies : 0.0000588977 Ha
Difference of unit cell volume : 0.05861 Bohr^3
Stress from finite difference : 1.0049 Ha/Bohr^3
Average stress : 1.0062 Ha/Bohr^3" data-toggle="popover" href="/tests/v3/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Magnesium HCP. Optimize the cell volume and shape using different constraints.
Checks whether the stresses are symmetric, and whether the optimisation algorithm does
not break the symmetry.
Also checks the possibility to disable the stop after violation of the condition related to dilatmx.
This is done in dataset 6. The optimization delivers acell values slightly different from those of dataset 2." data-toggle="popover" href="/tests/v3/Input/t42.abi" target="_blank">t42.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
1. Si diamond. Usual geometry, k points, psps. One of the two
atoms is displaced in order to have non-zeroforces.
Test whether energy and forces (and soon stresses) are correctly
given, even with use of spherical harmonics.
Perform first computation with the use of Legendre polynomials;
Then perform a computation with the use of spherical harmonics.
Both must give the same results." data-toggle="popover" href="/tests/v4/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ca in orthorhombic structure. 1 k point, low ecut.
Perform a test of GS total energy computation, using an
experimental PAW pseudopotential (pspcod=7)." data-toggle="popover" href="/tests/v4/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O in orthorhombic structure. 1 k point, low ecut. Experimental PAW pseudopotential.
The aim of this test is to verify that different values
of input variable &#x27;ortalg&#x27; give the same results for
total energy. &#x27;ortalg&#x27; is use in the conjugate gradient
algorithm which solves, in the case of PAW calculations,
a generalized eigenproblem (H.|Psi&gt;=Lambda.S|Psi&gt;).
Three datasets are perfomed ; they must give the same
total energies values (when converged)." data-toggle="popover" href="/tests/v4/Input/t05.abi" target="_blank">t05.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of geometry optimization using using ionmov=1,4,5 and 7
Hydrogen molecule inside a box of 12 Bohr
This test was created to test ionmov=5" data-toggle="popover" href="/tests/v5/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-toggle="popover" href="/tests/v5/Input/t03.abi" target="_blank">t03.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Delocalized internals test. Si diamond bulk with displaced atom
Unconverged in all variables" data-toggle="popover" href="/tests/v6/Input/t23.abi" target="_blank">t23.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the extrapolation of then wave functions from
one structural relaxation (time) step to the other:
extrapwf=1
WARNING: this feature is stil experimental and probably buggy.
Si, Bulk, 2 atoms" data-toggle="popover" href="/tests/v7/Input/t09.abi" target="_blank">t09.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density.
We test several improvements that are available when the NC file contains the PS valence charge.
More specifically, we test the initialization of the density from atomic quantities (initro.F90),
the correction to the forces due to the residuals and the extrapolation of the density
for structural relaxations (well, xred is fixed by symmetry).
We also test the treatment of the core charge in G-space in the case of GS calculations
By default, abinit handles XCCC in real space, here we use nc_xccc_gspace==1 to treat it
in G-space using the same approach as the one used in PAW." data-toggle="popover" href="/tests/v7/Input/t10.abi" target="_blank">t10.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Boron, 32 atoms - Thomas-Fermi method (without orbitals)
Test the temperature-dependent XC functional (ixc=50)
Ichimaru et al Phys. Rep. 149, 91-205 (1987)
Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-toggle="popover" href="/tests/v7/Input/t11.abi" target="_blank">t11.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="Geometry optimization with LBFGS" data-toggle="popover" href="/tests/v8/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Simple test of velocity verlet integrator" data-toggle="popover" href="/tests/v8/Input/t12.abi" target="_blank">t12.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Geometry optimization with FIRE (ionmov 15)" data-toggle="popover" href="/tests/v8/Input/t17.abi" target="_blank">t17.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 with in-plane strain.
Relaxing the third vector using optcell 6" data-toggle="popover" href="/tests/v9/Input/t104.abi" target="_blank">t104.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the post identification of more symmetric structures obtained after geometry optimization
Consider diamond and graphene, with low ecut and number of k points." data-toggle="popover" href="/tests/v9/Input/t189.abi" target="_blank">t189.abi</a> &hellip;</li>
<li>vdwxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-toggle="popover" href="/tests/vdwxc/Input/t10.abi" target="_blank">t10.abi</a> &hellip;</li>
</ul>
</details>
<p>Gives the maximum number of molecular dynamics or electron dynamics time steps or structural
optimization steps to be done if <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> or <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> are not &ldquo;none&rdquo; or <a class="wikilink codevar-wikilink" href="/variables/gstate#optdriver">optdriver</a> is 9
(real-time TDDFT).
Starting with Abinit9, ntime is automatically set to 1000, if <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> or <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> are not &ldquo;none&rdquo;,
<a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a> is zero and  <strong>ntime</strong>  is not specified in the input file.
Users are encouraged to pass a <strong>timelimit</strong> to Abinit using the command line and the syntax:</p>
<div class="codehilite"><pre><span></span><code>    abinit --timelimit hours:minutes:seconds
</code></pre></div>

<p>so that the code will try to stop smoothly before the timelimit and produce the DEN and the WFK files
that may be used to restart the calculation.</p>
<p>Note that at the present the option <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> = &ldquo;viscous&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 1) is initialized with four
Runge-Kutta steps which costs some overhead in the startup. By contrast, the
initialisation of other <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> or <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> values is only one SCF call.
Note that <strong>ntime</strong> is ignored if <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> or <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> are set to &ldquo;none&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 0).</p>
<h2 id="ntimimage"><strong>ntimimage</strong><a class="headerlink" href="#ntimimage" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of TIME steps for IMAGE propagation<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [22/1314] in all abinit tests, [5/178] in abinit tutorials</summary>
<ul>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_04.abi" target="_blank">timages_04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Test the input variable imgwfstor.
In itimimage=2, the SCF cycle and the Broyden optimization are indeed immediately fulfilled." data-toggle="popover" href="/tests/v8/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Fixed occupation numbers. imgmov 6 + ionmov 2 algorithm.
dataset 1 : Broyden for one image
dataset 2 : Broyden for two images, but only the first one has non-zero weight.
dataset 3 : Broyden for two images, but only the second one has non-zero weight.
dataset 4 : Broyden for two images, with weights such that the first image has still lower energy than the second
dataset 5 : Broyden for two images, with weights such that the second image has now lower energy than the first.
Geometries that mimimize the linear combination of energy at fixed Lagrange parameters can thus be found." data-toggle="popover" href="/tests/v8/Input/t20.abi" target="_blank">t20.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
</ul>
</details>
<p>Gives the maximal number of molecular dynamics time steps or structural
optimization steps to be done for the set of images, referred to as &lsquo;image-
timesteps&rsquo;. At each image-timestep, all the images are propagated
simultaneously, each according to the algorithm determined by <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> and
the usual accompanying input variables, and then the next positions and
velocities for each image are determined from the set of results obtained for all images.</p>
<h2 id="optcell"><strong>optcell</strong><a class="headerlink" href="#optcell" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> OPTimize the CELL shape and dimensions<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [69/1314] in all abinit tests, [10/178] in abinit tutorials</summary>
<ul>
<li>bigdft:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule with geometry optimisation using BFGS (test the reformating part of the code)." data-toggle="popover" href="/tests/bigdft/Input/t10.abi" target="_blank">t10.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2O in isolated boundary conditions (plane waves, NC), try tu use Damiens Poisson cutoff.
Water molecule with HGH pseudo-potentials. This checks that a negative
value in rhor will not make the Poisson solver crashes on XC computation." data-toggle="popover" href="/tests/bigdft/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
<li>built-in:  <a class="wikilink abifile-wikilink" data-tippy-content="Ca atom with PAW, writing a ETSF_IO wavefunction file." data-toggle="popover" href="/tests/built-in/Input/testin_etsf_io.abi" target="_blank">testin_etsf_io.abi</a> &hellip;</li>
<li>etsf_io:  <a class="wikilink abifile-wikilink" data-tippy-content="Ca atom with PAW, writing a ETSF_IO wavefunction file.
Very quick built-in test, to check that ETSF_IO is installed." data-toggle="popover" href="/tests/etsf_io/Input/t00.abi" target="_blank">t00.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ca atom in PAW. Test the output of density in PAW formalism with ETSF_IO.
Also test the output of the wavefunctions in PAW formalism with ETSF_IO." data-toggle="popover" href="/tests/etsf_io/Input/t09.abi" target="_blank">t09.abi</a> &hellip;</li>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu/Input/t04.abi" target="_blank">t04.abi</a> &hellip;</li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu_omp/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/mpiio/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/paral/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 4 procs.
Similar to test paral_22 but test the automatic parallelisation" data-toggle="popover" href="/tests/paral/Input/t26.abi" target="_blank">t26.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium,
to compare with the PSML results of t07 and t11" data-toggle="popover" href="/tests/psml/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Partial cut-off energy convergence study of bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t07.abi" target="_blank">t07.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=14 (finite reduced electric field calculation,relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_121.abi" target="_blank">tsv6_121.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=16 (finite reduced electric displacement field calculation,
relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_122.abi" target="_blank">tsv6_122.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=4 (finite electric field calculation, relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_123.abi" target="_blank">tsv6_123.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_01.abi" target="_blank">timages_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_02.abi" target="_blank">timages_02.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Finite difference calculation of the clamped-ion
piezoelectric constants of AlP" data-toggle="popover" href="/tests/tutorespfn/Input/tpolarization_5.abi" target="_blank">tpolarization_5.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline silicon : computation of the optimal lattice parameter
Convergence with respect to the number of k points." data-toggle="popover" href="/tests/tutorial/Input/tbase3_4.abi" target="_blank">tbase3_4.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline aluminum : optimization of the lattice parameter
at fixed number of k points and broadening." data-toggle="popover" href="/tests/tutorial/Input/tbase4_1.abi" target="_blank">tbase4_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline aluminum : optimization of the lattice parameter

Convergence with respect to k points." data-toggle="popover" href="/tests/tutorial/Input/tbase4_2.abi" target="_blank">tbase4_2.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Optimization of unit cell volume (optcell=1).
Use 2 datasets, and the variable getcell." data-toggle="popover" href="/tests/v1/Input/t78.abi" target="_blank">t78.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a cell of variable size along the H2 chain =&gt; H polymer.
Optimization of cell size and atomic distances (optcell=4)." data-toggle="popover" href="/tests/v1/Input/t79.abi" target="_blank">t79.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Mg crystal, 1 atom per unit cell. Start with deformed FCC lattice,
optimize the lattice towards FCC (optcell=2).
Also test the output of the HIST.nc file (restartxf==-1)" data-toggle="popover" href="/tests/v1/Input/t80.abi" target="_blank">t80.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="Compare cprj_in_memory=1 with cprj_in_memory=0 (lobpcg2 and chebfi2) for cell optimization. PAW,istwfk=1." data-toggle="popover" href="/tests/v10/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Arsenic in rhombohedral structure.
Optimization of unit cell size and shape.
With 4x4x4 k point grid, ecut 3.0, occopt 4 and tsmear 0.06d0,
one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277
These input parameters are too small, but still give a realistic
geometry, since with the much better parameters
12x12x12 k point grid, ecut 12.0, occopt 4 and tsmear 0.04d0,
one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296, while
the experimental values are
acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-toggle="popover" href="/tests/v2/Input/t44.abi" target="_blank">t44.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
Magnesium HCP. Optimize the cell volume and shape using different constraints.
Checks whether the stresses are symmetric, and whether the optimisation algorithm does
not break the symmetry.
Also checks the possibility to disable the stop after violation of the condition related to dilatmx.
This is done in dataset 6. The optimization delivers acell values slightly different from those of dataset 2." data-toggle="popover" href="/tests/v3/Input/t42.abi" target="_blank">t42.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
1. Si diamond. Usual geometry, k points, psps. One of the two
atoms is displaced in order to have non-zeroforces.
Test whether energy and forces (and soon stresses) are correctly
given, even with use of spherical harmonics.
Perform first computation with the use of Legendre polynomials;
Then perform a computation with the use of spherical harmonics.
Both must give the same results." data-toggle="popover" href="/tests/v4/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ca in orthorhombic structure. 1 k point, low ecut.
Perform a test of GS total energy computation, using an
experimental PAW pseudopotential (pspcod=7)." data-toggle="popover" href="/tests/v4/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O in orthorhombic structure. 1 k point, low ecut. Experimental PAW pseudopotential.
The aim of this test is to verify that different values
of input variable &#x27;ortalg&#x27; give the same results for
total energy. &#x27;ortalg&#x27; is use in the conjugate gradient
algorithm which solves, in the case of PAW calculations,
a generalized eigenproblem (H.|Psi&gt;=Lambda.S|Psi&gt;).
Three datasets are perfomed ; they must give the same
total energies values (when converged)." data-toggle="popover" href="/tests/v4/Input/t05.abi" target="_blank">t05.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test to prepare a conductivity calculation (test v5#t10)
Ground state calculation for 4 Al atoms
The number of bands is unrealistic. This test produces files that are inputs for test 10.
It tests the following parts of the code:
optic_paw, optic_paw_core
Note that, in addition to the psp file, the core wave functions file is needed." data-toggle="popover" href="/tests/v5/Input/t09.abi" target="_blank">t09.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
White tin, body-centered tetragonal. This test checks that in an optcell 2 run, the symmetry
of the body-centered case is not broken. This test is run for two values of strprecon also,
to test the strprecon variable.
Coded and contributed by J. Zwanziger." data-toggle="popover" href="/tests/v5/Input/t77.abi" target="_blank">t77.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrure of Lithium-Boron, in a face-centered monoclinic cell.
Test the recognition of the space group (C2/c (# 15)). Was lacking a case before Dec 2011." data-toggle="popover" href="/tests/v6/Input/t30.abi" target="_blank">t30.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the extrapolation of then wave functions from
one structural relaxation (time) step to the other:
extrapwf=1
WARNING: this feature is stil experimental and probably buggy.
Si, Bulk, 2 atoms" data-toggle="popover" href="/tests/v7/Input/t09.abi" target="_blank">t09.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density.
We test several improvements that are available when the NC file contains the PS valence charge.
More specifically, we test the initialization of the density from atomic quantities (initro.F90),
the correction to the forces due to the residuals and the extrapolation of the density
for structural relaxations (well, xred is fixed by symmetry).
We also test the treatment of the core charge in G-space in the case of GS calculations
By default, abinit handles XCCC in real space, here we use nc_xccc_gspace==1 to treat it
in G-space using the same approach as the one used in PAW." data-toggle="popover" href="/tests/v7/Input/t10.abi" target="_blank">t10.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test to prepare a SOC conductivity calculation (test v9#t101)
Ground state calculation for 4 Al atoms
The number of bands is unrealistic. This test produces files that are inputs for test 101.
It tests the following parts of the code:
optic_paw, optic_paw_core, optic_paw_soc
Note that, in addition to the psp file, the spinor core wave functions file is needed." data-toggle="popover" href="/tests/v9/Input/t100.abi" target="_blank">t100.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test to prepare a SOC conductivity calculation (test v9#t102)
Ground state calculation for 4 Al atoms
The number of bands is unrealistic. This test produces files that are inputs for test 103.
It tests the following parts of the code:
optic_paw_core
Note that, in addition to the psp file, the spinor core wave functions file is needed." data-toggle="popover" href="/tests/v9/Input/t102.abi" target="_blank">t102.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 with in-plane strain.
Relaxing the third vector using optcell 6" data-toggle="popover" href="/tests/v9/Input/t104.abi" target="_blank">t104.abi</a> &hellip;</li>
<li>vdwxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-toggle="popover" href="/tests/vdwxc/Input/t10.abi" target="_blank">t10.abi</a> &hellip;</li>
</ul>
</details>
<p>Allows one to optimize the unit cell shape and dimensions (for <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> &gt;=2).
The configuration for which the stress almost vanishes is iteratively
determined, by using the same algorithms as for the nuclei positions.
May modify <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a> and/or <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a>. The ionic positions are ALWAYS
updated, according to the forces. A target stress tensor might be defined, see <a class="wikilink codevar-wikilink" href="/variables/rlx#strtarget">strtarget</a>.
In most cell relaxations ( <strong>optcell</strong> /=0), the user should define <a class="wikilink codevar-wikilink" href="/variables/rlx#ecutsm">ecutsm</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#dilatmx">dilatmx</a>, and <a class="wikilink codevar-wikilink" href="/variables/basic#tolrff">tolrff</a>, because the default values
are NOT aimed at cell relaxation runs. The user should also be aware that convergence studies might need to modify the value of <a class="wikilink codevar-wikilink" href="/variables/rlx#tolmxf">tolmxf</a>, the default being decent, but possibly not sufficiently stringent.</p>
<ul>
<li><strong>optcell</strong> = 0: modify nuclear positions, but no cell shape and dimension optimisation.</li>
<li><strong>optcell</strong> = 1: optimisation of volume only (do not modify <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a>, and allow an homogeneous dilatation of the three components of <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a>)</li>
<li><strong>optcell</strong> = 2: full optimization of cell geometry (modify <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a> and <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> - normalize the vectors of <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> to generate the <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a>). This is the usual mode for cell shape and volume optimization. It takes into account the symmetry of the system, so that only the effectively relevant degrees of freedom are optimized.</li>
<li><strong>optcell</strong> = 3: constant-volume optimization of cell geometry (modify <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a> and <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> under constraint - normalize the vectors of <a class="wikilink codevar-wikilink" href="/variables/basic#rprim">rprim</a> to generate the <a class="wikilink codevar-wikilink" href="/variables/basic#acell">acell</a>)</li>
<li><strong>optcell</strong> = 4, 5 or 6: optimize (both length and angle) the first, second or third vector, respectively (only works if, in the case <strong>optcell</strong>=6, the two other vectors lie in the xy cartesian plane; similarly for <strong>optcell</strong>=4 and <strong>optcell</strong>=5, the two clamped vectors must lie in the yz and xz plane respectively).</li>
<li><strong>optcell</strong> = 7, 8 or 9: optimize the cell geometry while keeping the first, second or third vector unchanged (only works if the two other vectors are orthogonal to the one left unchanged, the latter being along its cartesian axis).</li>
</ul>
<p>A few details require attention when performing unit cell optimisation:</p>
<ul>
<li>one has to get rid of the discontinuities due to discrete changes of plane wave number with cell size, by using a suitable value of <a class="wikilink codevar-wikilink" href="/variables/rlx#ecutsm">ecutsm</a>;</li>
<li>one has to allow for the possibility of a larger sphere of plane waves, by using <a class="wikilink codevar-wikilink" href="/variables/rlx#dilatmx">dilatmx</a>;</li>
<li>one might have to adjust the scale of stresses to the scale of forces, by using <a class="wikilink codevar-wikilink" href="/variables/rlx#strfact">strfact</a>.</li>
<li>if all the reduced coordinates of atoms are fixed by symmetry, one cannot use <a class="wikilink codevar-wikilink" href="/variables/basic#tolrff">tolrff</a> (or <a class="wikilink codevar-wikilink" href="/variables/basic#toldff">toldff</a>) to stop the SCF cycle. (Suggestion: use <a class="wikilink codevar-wikilink" href="/variables/basic#tolvrs">tolvrs</a> with a small value, like 1.0d-14)</li>
</ul>
<p>It is STRONGLY suggested first to optimize the ionic positions without cell
shape and size optimization (<strong>optcell</strong> = 0), then start the cell shape and
size optimization from the cell with relaxed ionic positions, possibly from the previous dataset, using <a class="wikilink codevar-wikilink" href="/variables/rlx#getxred">getxred</a> or <a class="wikilink codevar-wikilink" href="/variables/rlx#getxcart">getxcart</a>. Presently
(v3.1), one cannot restart (<a class="wikilink codevar-wikilink" href="/variables/rlx#restartxf">restartxf</a>) a calculation with a non-zero
<strong>optcell</strong> value from the (x,f) history of another run with a different non-
zero <strong>optcell</strong> value. There are still a few problems at that level.</p>
<h2 id="pimass"><strong>pimass</strong><a class="headerlink" href="#pimass" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Path Integral fictitious MASSes<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/basic#ntypat">ntypat</a>)<br />
<em>Default value:</em> <a class="wikilink codevar-wikilink" href="/variables/basic#ntypat">ntypat</a><br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9 or 13<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
</ul>
</details>
<p>Only relevant if <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9 or 13 (Path-Integral Molecular Dynamics).
Gives the fictitious masses ( <a class="wikilink citation-wikilink" data-tippy-content="Ab initio path integral molecular dynamics: Basic ideas" data-toggle="popover" href="/theory/bibliography#marx1996">[Marx1996]</a>) in atomic mass units for each kind of atom in cell. These masses are the inertial masses used in performing Path Integral Molecular
Dynamics (PIMD), they are different from the true masses (<a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>) used to
define the quantum spring that relates the different beads in PIMD. They can
be chosen arbitrarily, but an appropriate choice will lead the different
variables to move on the same time scale in order to optimize the sampling
efficiency of the PIMD trajectory.
If <a class="wikilink codevar-wikilink" href="/variables/rlx#pitransform">pitransform</a> = 1 (normal mode transformation), or <a class="wikilink codevar-wikilink" href="/variables/rlx#pitransform">pitransform</a> = 2
(staging transformation),  <strong>pimass</strong>  is automatically set to its optimal value.</p>
<h2 id="pimd_constraint"><strong>pimd_constraint</strong><a class="headerlink" href="#pimd_constraint" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Path-Integral Molecular Dynamics: CONSTRAINT to be applied on a reaction coordinate<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9 or 13<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Only relevant for Path-Integral Molecular Dynamics.
Selects a constraint to be applied during the PIMD trajectory. The constraint
is holonomic (it is a relation between the position variables). In practice,
the total forces applied to the atomic positions are modified so as to respect
the constraint.</p>
<p>To date, the available constraints are:</p>
<ul>
<li><strong>0</strong>: no constraint</li>
<li><strong>1</strong>: <em>&ldquo;Blue Moon Ensemble&rdquo; method</em>.
The constraint is a linear combination of the positions of atomic centroids
(this linear combination is kept constant during the simulation).
Sum[W_i * X_i] = constant
The X_i are the coordinates of the atomic centroids. The weights W_i have to
be specified with the <a class="wikilink codevar-wikilink" href="/variables/rlx#wtatcon">wtatcon</a>(3,<a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>,<a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>),
<a class="wikilink codevar-wikilink" href="/variables/rlx#iatcon">iatcon</a>(<a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>) and <a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a> input parameters (where <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a> is fixed to 1).
More details on the implementation in <a class="wikilink citation-wikilink" data-tippy-content="Implementation of the blue moon ensemble method" data-toggle="popover" href="/theory/bibliography#komeiji2007">[Komeiji2007]</a>.</li>
</ul>
<h2 id="pitransform"><strong>pitransform</strong><a class="headerlink" href="#pitransform" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Path Integral coordinate TRANSFORMation<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [4/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Only relevant if <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9 or 13 (Path-Integral Molecular Dynamics).
Coordinate transformation used in the integration of the Path Integral
Molecular Dynamics equations of motion. The transformation, with an
appropriate choice of fictitious masses (<a class="wikilink codevar-wikilink" href="/variables/rlx#pimass">pimass</a>), is used to force the
different modes to move on the same time scale, and thus optimize the
efficiency of the statistical sampling in the corresponding statistical
ensemble. Available with a Langevin thermostat (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9) or with Nose-
Hoover chains (<a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 13). See <a class="wikilink citation-wikilink" data-tippy-content="Efficient and general algorithms for path integral CarParrinello molecular dynamics" data-toggle="popover" href="/theory/bibliography#tuckerman1996">[Tuckerman1996]</a>.</p>
<p>If equal to 0, no transformation is applied (primitive coordinates).
If equal to 1, normal mode transformation (in that case, <a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a> must be absolutely EVEN).
If equal to 2, staging transformation.</p>
<h2 id="prtatlist"><strong>prtatlist</strong><a class="headerlink" href="#prtatlist" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> PRinT by ATom LIST of ATom<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">NO_MULTI</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to print some useful quantities" data-toggle="popover" href="/topics/printing">topic_printing</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to tune the output of computed quantities" data-toggle="popover" href="/topics/Output">topic_Output</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>)<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms)." data-toggle="popover" href="/tests/v4/Input/t97.abi" target="_blank">t97.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom." data-toggle="popover" href="/tests/v8/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom along one direction." data-toggle="popover" href="/tests/v8/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>This is an array of the numbers associated to the index atoms that the user
want to print in the output or log files, this is useful when you have a large
number of atoms and you are only interested to follow specific atoms, the
numbers associated should be consistent with the list in <a class="wikilink codevar-wikilink" href="/variables/basic#xcart">xcart</a> or
<a class="wikilink codevar-wikilink" href="/variables/basic#xred">xred</a>. This input variable does not affect the contents of the &ldquo;OUT.nc&rdquo; or
&ldquo;HIST.nc&rdquo;, those are NetCDF files containing the information about all the atoms.</p>
<h2 id="qmass"><strong>qmass</strong><a class="headerlink" href="#qmass" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Q thermostat MASS<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="/variables/rlx#nnos">nnos</a>)<br />
<em>Default value:</em> <em>10.0<br />
*Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a></li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation
in (N,V,T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested
together with the three coordinate systems (primitive, normal mode, staging).
NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with
--- Pseudopotential description --" data-toggle="popover" href="/tests/v7/Input/t08.abi" target="_blank">t08.abi</a></li>
</ul>
</details>
<p>This are the masses of the chains of <a class="wikilink codevar-wikilink" href="/variables/rlx#nnos">nnos</a> thermostats to be used when
<a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_nose&rdquo; or &ldquo;npt_martyna&rdquo; or &ldquo;nst_martyna&rdquo; (Molecular Dynamics) or <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 13 (Path Integral Molecular
Dynamics).</p>
<p>If <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> = &ldquo;nvt_nose&rdquo; or &ldquo;npt_martyna&rdquo; or &ldquo;nst_martyna&rdquo; (Molecular Dynamics), this temperature control can be used
with  <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> =0, 1 (homogeneous cell deformation) or 2 (full cell deformation).
If <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 13 (Path Integral Molecular Dynamics), this temperature control
can be used with  <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> =0 (NVT ensemble) or 2 (fully flexible NPT
ensemble). In that case, <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> = 2 is NOT USABLE yet.</p>
<h2 id="random_atpos"><strong>random_atpos</strong><a class="headerlink" href="#random_atpos" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> RANDOM ATomic POSitions<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to specify a crystal, with atomic positions and symmetries" data-toggle="popover" href="/topics/crystal">topic_crystal</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Genetic algorithm structure random search.
Hydrogen has been used as an example. No physical meaning.
Number of images is only 10 but experience shows that at
least 20 are fine." data-toggle="popover" href="/tests/v6/Input/t27.abi" target="_blank">t27.abi</a></li>
</ul>
</details>
<p>Control the inner coordinates, which can be generated randomly by using 4
different methods depending ont its value
(0) if zero, no random generation and xred are taken as they have been introduced by the user
(1) if one, particles are generated completely random within the unit cell.
(2) if two, particles are generated randomly but the inner particle distance
is always larger than a factor of the sum of the covalent bonds between the
atoms (note: this is incompatible with the definition of alchemical mixing,
in which <a class="wikilink codevar-wikilink" href="/variables/basic#ntypat">ntypat</a> differs from <a class="wikilink codevar-wikilink" href="/variables/gstate#npsp">npsp</a>)</p>
<h2 id="restartxf"><strong>restartxf</strong><a class="headerlink" href="#restartxf" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> RESTART from (X,F) history<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [5/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Same as test 66, H2 molecule with 1 k points, but uses the restartxf=-1 option
to test the output of the HIST.nc file. Note that HIST.nc is not read since
one should introduce a getrestartxf variable." data-toggle="popover" href="/tests/v1/Input/t75.abi" target="_blank">t75.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Mg crystal, 1 atom per unit cell. Start with deformed FCC lattice,
optimize the lattice towards FCC (optcell=2).
Also test the output of the HIST.nc file (restartxf==-1)" data-toggle="popover" href="/tests/v1/Input/t80.abi" target="_blank">t80.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="Short MD traj to test restarxf -1" data-toggle="popover" href="/tests/v10/Input/t81.abi" target="_blank">t81.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Short MD traj restarting from HIST file using restarxf -1" data-toggle="popover" href="/tests/v10/Input/t82.abi" target="_blank">t82.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Control the restart of a molecular dynamics or structural optimization job.</p>
<p><strong>restartxf = 0 (Default)</strong>: No restart procedure is enabled and the code will start a
Molecular dynamics or structural optimization from scratch.</p>
<p><strong>restartxf = -1 (New)</strong>: Use the HIST.nc file to reconstruct a partial
calculation. It will reconstruct the different configurations using the forces
and the stresses stored in the HIST.nc file, instead of calling the SCF procedure.
Using <strong>restartxf = -1</strong> from the beginning is harmless. The only condition is
to keep the input file the same in such a way that the same predictor is used
and it will predict the same structure recorded in the HIST.nc file.
This option will always compute extra <a class="wikilink codevar-wikilink" href="/variables/rlx#ntime">ntime</a> iterations independent of the
number of iterations recovered previously.</p>
<p><strong>restartxf = -2 (New)</strong>: Read the HIST.nc file and select the atomic positions and
cell parameters with the lowest energy. Forget all the history and start the
calculation using those values. The original atomic coordinates and cell
parameters are irrelevant in this case.</p>
<p><strong>restartxf = -3 (New)</strong>: Read <strong>ONLY</strong> the last required atomic positions and cell parameters
in the HIST.nc file to restart the Molecular dynamics or structural optimization.</p>
<p>NOTES:</p>
<ul>
<li>You can use <strong>restartxf=-1, -2 or -3</strong> for all predictors that make no use of random numbers.</li>
<li>You can use <strong>restartxf=-1, -2 or -3</strong> to restart a calculation that was not completed. The HIST.nc file is written on each iteration. So you always have something to recover from.</li>
<li>You can take advantage of the appropriate values of <a class="wikilink codevar-wikilink" href="/variables/files#irdwfk">irdwfk</a> or <a class="wikilink codevar-wikilink" href="/variables/files#getwfk">getwfk</a> to get a good wave function to continue your job.</li>
</ul>
<p>Previously, positive values of restartxf were allowed, but these are deprecated since ABINITv8.</p>
<h2 id="signperm"><strong>signperm</strong><a class="headerlink" href="#signperm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> SIGN of PERMutation potential<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [2/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-toggle="popover" href="/tests/v2/Input/t87.abi" target="_blank">t87.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a></li>
</ul>
</details>
<p>+1 favors alternation of species -1 favors segregation</p>
<h2 id="strfact"><strong>strfact</strong><a class="headerlink" href="#strfact" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRess FACTor<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [9/1314] in all abinit tests, [2/178] in abinit tutorials</summary>
<ul>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=14 (finite reduced electric field calculation,relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_121.abi" target="_blank">tsv6_121.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=16 (finite reduced electric displacement field calculation,
relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_122.abi" target="_blank">tsv6_122.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=4 (finite electric field calculation, relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_123.abi" target="_blank">tsv6_123.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=6 (finite electric displacement field calculation,
relaxing cell perameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_124.abi" target="_blank">tsv6_124.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test berryopt=17 (mixed finite electric field and electric displacement
field boundary condition, relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_125.abi" target="_blank">tsv6_125.abi</a></li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a></li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.
Optimization of unit cell volume (optcell=1).
Use 2 datasets, and the variable getcell." data-toggle="popover" href="/tests/v1/Input/t78.abi" target="_blank">t78.abi</a></li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
Make a simple geometry relaxation and validate the XML output (testing prtxml)." data-toggle="popover" href="/tests/v5/Input/t79.abi" target="_blank">t79.abi</a></li>
</ul>
</details>
<p>The stresses (in atomic units) multiplied by  <strong>strfact</strong>  will be treated like forces (in atomic units) in the
algorithms for optimization (<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> == &ldquo;bfgs&rdquo; or &ldquo;lbfgs&rdquo;, non-zero <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>).
For example, the stopping criterion defined by <a class="wikilink codevar-wikilink" href="/variables/rlx#tolmxf">tolmxf</a> relates to these
scaled stresses.</p>
<h2 id="string_algo"><strong>string_algo</strong><a class="headerlink" href="#string_algo" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRING method ALGOrithm<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a></li>
</ul>
</details>
<p>Relevant only when <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 2 (String Method).
Gives the variant of the String Method method used.
Possible values can be:</p>
<ul>
<li>
<p>0 &rarr;  <strong>Original String Method</strong>.
NOT YET IMPLEMENTED.
See <a class="wikilink citation-wikilink" data-tippy-content="String Method for the study of rare events" data-toggle="popover" href="/theory/bibliography#weinan2002">[Weinan2002]</a></p>
</li>
<li>
<p>1 &rarr; <strong>Simplified String Method</strong> with parametrization by <strong>equal arc length</strong>.
Instead of using the normal force (wrt the band), the full force is used; the
reparametrization is enforced by keeping the points of the string equally
spaced.
See <a class="wikilink citation-wikilink" data-tippy-content="Simplified and improved string method for computing the minimum energy paths in barrier-crossing events" data-toggle="popover" href="/theory/bibliography#weinan2007">[Weinan2007]</a></p>
</li>
<li>
<p>2 &rarr; <strong>Simplified String Method</strong> with parametrization by <strong>energy-weighted arc length</strong>.
A variant of the Simplified String Method (like 1-); the reparametrization is
done by using energy-weight arc-lengths, giving a finer distribution near the saddle point.
See <a class="wikilink citation-wikilink" data-tippy-content="Simplified and improved string method for computing the minimum energy paths in barrier-crossing events" data-toggle="popover" href="/theory/bibliography#weinan2007">[Weinan2007]</a> and <a class="wikilink citation-wikilink" data-tippy-content="Transition state-finding strategies for use with the growing string method" data-toggle="popover" href="/theory/bibliography#goodrow2009">[Goodrow2009]</a></p>
</li>
</ul>
<h2 id="strprecon"><strong>strprecon</strong><a class="headerlink" href="#strprecon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRess PRECONditioner<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to tune the computation of forces and stresses" data-toggle="popover" href="/topics/ForcesStresses">topic_ForcesStresses</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1.0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [4/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>bigdft:  <a class="wikilink abifile-wikilink" data-tippy-content="Hypothetical NaTi molecule. Test the diagonalisation scheme with wavelets" data-toggle="popover" href="/tests/bigdft/Input/t14.abi" target="_blank">t14.abi</a></li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
White tin, body-centered tetragonal. This test checks that in an optcell 2 run, the symmetry
of the body-centered case is not broken. This test is run for two values of strprecon also,
to test the strprecon variable.
Coded and contributed by J. Zwanziger." data-toggle="popover" href="/tests/v5/Input/t77.abi" target="_blank">t77.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
HI 8 molecule per cell, the cell is triclinic (has to change tolsym as default is too permissive)
Prepare the density for the next Bader analysis." data-toggle="popover" href="/tests/v6/Input/t18.abi" target="_blank">t18.abi</a></li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 with in-plane strain.
Relaxing the third vector using optcell 6" data-toggle="popover" href="/tests/v9/Input/t104.abi" target="_blank">t104.abi</a></li>
</ul>
</details>
<p>This is a scaling factor to initialize the part of the Hessian related to the
treatment of the stresses (optimisation of the unit cell). In case there is an
instability, decrease the default value, e.g. set it to 0.1.</p>
<h2 id="strtarget"><strong>strtarget</strong><a class="headerlink" href="#strtarget" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRess TARGET<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to tune the computation of forces and stresses" data-toggle="popover" href="/topics/ForcesStresses">topic_ForcesStresses</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (6)<br />
<em>Default value:</em> [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [4/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Mg crystal, 1 atom per unit cell. Start with deformed FCC lattice,
optimize the lattice towards FCC (optcell=2).
Also test the output of the HIST.nc file (restartxf==-1)" data-toggle="popover" href="/tests/v1/Input/t80.abi" target="_blank">t80.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: MD simulation with Langevin thermostat." data-toggle="popover" href="/tests/v10/Input/t21.abi" target="_blank">t21.abi</a></li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
This is a test, in the norm conserving case, of the Molecular Dynamics flag
ionmov==13. With the values of optcell=0, 1, 2 (isothermal (0) , isothermal/isenthalpic
with homogeneous cell deformations (1) or full cell deformations (2).)
This is the reversible integrator of Martyna, Tuckerman, Tobias and Klein.
(Mol. Phys., 1996, p 1117).
XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent, while
the bmass varies by a factor of 2 ?" data-toggle="popover" href="/tests/v5/Input/t01.abi" target="_blank">t01.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
HI 8 molecule per cell, the cell is triclinic (has to change tolsym as default is too permissive)
Prepare the density for the next Bader analysis." data-toggle="popover" href="/tests/v6/Input/t18.abi" target="_blank">t18.abi</a></li>
</ul>
</details>
<p>Define the target stress to be obtained by optimization of the cell.
The components of the stress tensor must be stored according to: (1,1) &rarr;1;
(2,2) &rarr; 2; (3,3) &rarr; 3; (2,3) &rarr; 4; (3,1) &rarr; 5; (1,2) &rarr;6. The conversion factor
between Ha/Bohr**3 (the atomic unit for stress) and GPa is: 1 Ha/Bohr**3 = 29421.033 GPa.
If a hydrostatic stress (so, a pressure P) is wanted, define  <strong>strtarget</strong>  as
-P -P -P 0 0 0 .
Not used if <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> == 0.</p>
<h2 id="tolimg"><strong>tolimg</strong><a class="headerlink" href="#tolimg" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> TOLerance on the mean total energy for IMaGes<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">ENERGY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to calculate transition paths" data-toggle="popover" href="/topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 5e-05<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [17/1314] in all abinit tests, [5/178] in abinit tutorials</summary>
<ul>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the string method within parallelization over images
Inspired by test v6#22.
Hydrogen diatomic molecule in a cell, close to BCC
7 images, exploring the transition path.
Three datasets, testing each value of prtvolimg.
Processors distribution automatically determined:" data-toggle="popover" href="/tests/paral/Input/t08.abi" target="_blank">t08.abi</a></li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_03.abi" target="_blank">timages_03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
String method
Moving the proton from H2O to NH3
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_04.abi" target="_blank">timages_04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at a set of decreasing I-I interatomic distances
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_02.abi" target="_blank">tpsic_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test starts from a reasonable I-I interatomic distance, and search for the optimized polaron geometry, without constraint.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_03.abi" target="_blank">tpsic_03.abi</a></li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the GSS-NEB method (Nudged Elastic Band with variable cell)
Cubic cristal of beryllium
7 images, exploring the distortion of a cell when moving an atom" data-toggle="popover" href="/tests/v10/Input/t83.abi" target="_blank">t83.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path. Similar to test 22 (but with string)." data-toggle="popover" href="/tests/v6/Input/t24.abi" target="_blank">t24.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the string method : 6 images, exploring the transition
path but keeping X coordinated fixed. Similar to test 24" data-toggle="popover" href="/tests/v6/Input/t25.abi" target="_blank">t25.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the NEB method : 6 images, exploring the transition
path. Similar to test 22 (but with NEB).
First dataset with neb_algo=0 (original NEB method), and second dataset with neb_algo=1 (improved tangent)." data-toggle="popover" href="/tests/v6/Input/t26.abi" target="_blank">t26.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC
Test the CI-NEB method : 7 images, exploring the transition
path. Use the climbing image method to refine the position of the saddle point.
Except the number of images, similar to test 22 (but with NEB)." data-toggle="popover" href="/tests/v6/Input/t29.abi" target="_blank">t29.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Fixed occupation numbers. imgmov 6 + ionmov 2 algorithm.
dataset 1 : Broyden for one image
dataset 2 : Broyden for two images, but only the first one has non-zero weight.
dataset 3 : Broyden for two images, but only the second one has non-zero weight.
dataset 4 : Broyden for two images, with weights such that the first image has still lower energy than the second
dataset 5 : Broyden for two images, with weights such that the second image has now lower energy than the first.
Geometries that mimimize the linear combination of energy at fixed Lagrange parameters can thus be found." data-toggle="popover" href="/tests/v8/Input/t20.abi" target="_blank">t20.abi</a></li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a></li>
</ul>
</details>
<p>Sets a maximal absolute energy tolerance (in hartree, averaged over dynamic
images) below which iterations on images (the one governed by the
<a class="wikilink codevar-wikilink" href="/variables/rlx#ntimimage">ntimimage</a> input variable) will stop.
This is to be used when trying to optimize a population of structures to their
lowest energy configuration, taking into account the particular algorithm
defined by <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a>
A value of about 5.0d-5 hartree or smaller is suggested (this corresponds to about 3.7d-7 eV).
No meaning for RF calculations.</p>
<h2 id="tolmxde"><strong>tolmxde</strong><a class="headerlink" href="#tolmxde" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> TOLerance on the MaXimal Difference in Energy<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">ENERGY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [1/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test tolmxde input variable. For silicon supercell with atomic positions randomly displaced" data-toggle="popover" href="/tests/v7/Input/t15.abi" target="_blank">t15.abi</a></li>
</ul>
</details>
<p>Sets a maximal difference in energy with respect to the two previous steps
below which BFGS structural relaxation iterations will stop.
A value of about 0.0005 eV/atom or smaller is suggested.
In order to use tolmxde, you should explicitly set tolmxf to 0.0.
No meaning for RF calculations.</p>
<h2 id="tolmxf"><strong>tolmxf</strong><a class="headerlink" href="#tolmxf" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> TOLerance on the MaXimal Force<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 5e-05<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [99/1314] in all abinit tests, [18/178] in abinit tutorials</summary>
<ul>
<li>bigdft:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2O in isolated boundary conditions (plane waves, NC), try tu use Damiens Poisson cutoff.
Water molecule with HGH pseudo-potentials. This checks that a negative
value in rhor will not make the Poisson solver crashes on XC computation." data-toggle="popover" href="/tests/bigdft/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
<li>built-in:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule, determination of the H-H distance by the Broyden algorithm." data-toggle="popover" href="/tests/built-in/Input/testin_fast.abi" target="_blank">testin_fast.abi</a> &hellip;</li>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule, determination of the H-H distance by the Broyden algorithm.
Very quick built-in test, to check that ABINIT is working." data-toggle="popover" href="/tests/fast/Input/t00.abi" target="_blank">t00.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="H2 diatomic molecule, static, to check accurate forces." data-toggle="popover" href="/tests/fast/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
2 special k point Si in 2-atom diamond unit cell.  Uses symmetry.
mkmem=mkpt, mffmem=1 (&quot;in core&quot; solution),
This calculation is also iterated to convergence.  ecut too small.
Uses original Teter extended norm conserving psp 14si.psp." data-toggle="popover" href="/tests/fast/Input/t03.abi" target="_blank">t03.abi</a> &hellip;</li>
<li>gpu:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu/Input/t04.abi" target="_blank">t04.abi</a> &hellip;</li>
<li>gpu_omp:  <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon - diamond structure - PAW - volume relaxation
Test the use of GPU within Projector Augmented-Wave formalism.
Test structure relaxation (multiple calls to gstate routine)." data-toggle="popover" href="/tests/gpu_omp/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule NON-spin-polarized, in a big box, without tetragonal symmetry, in order to avoid
spurious degeneracy effects coupled to occupation numbers rapidly changing.
GPU
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization." data-toggle="popover" href="/tests/gpu_omp/Input/t28.abi" target="_blank">t28.abi</a> &hellip;</li>
<li>libxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
N2 molecule non-spin-polarized, in a big box.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with underconverged ecut (ecut 18) and box size (5 4 4 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation  (ecut 18, 5 4 4 box) : 1.101, 1.084, 1.085 Angstrom
Better  calculation  (ecut 22, 6 5 5 box) : 1.102, 1.086, 1.086 Angstrom
ABINIT Arnardi report (ecut acell unknown): 1.103, 1.091, 1.092 Angstrom
VASP Arnardi report (ecut acell unknown)  : 1.103, 1.089,  NA   Angstrom
Experimental : 1.098 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t67.abi" target="_blank">t67.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule spin-polarized, in a big box.
Norm conserving : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with underconverged ecut (ecut 15) and box size (4 3 3 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation  (ecut 13, 5 4 4 box) : 1.138, 1.130, 1.131 Angstrom
Better  calculation  (ecut 18, 7 6 6 box) : 1.135, 1.124, 1.125 Angstrom
ABINIT Arnardi report (ecut acell unknown): 1.136, 1.124, 1.124 Angstrom
VASP Arnardi report (ecut acell unknown)  : 1.136, 1.124,  NA   Angstrom
Experimental : 1.120 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t68.abi" target="_blank">t68.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
CH molecule spin-polarized, in a big box.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with slightly better converged ecut (ecut 15) and box size (6 5 5 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation (ecut 13, 5 4 4 box)     : 1.130, 1.121, 1.122 Angstrom
Better calculation (ecut 15, 6 5 5 box)      : 1.136, 1.125, 1.126 Angstrom
Even better calculation (ecut 18, 7 6 6 box) : 1.138, 1.126, 1.127 Angstrom
ABINIT Arnardi report (ecut acell unknown)   : 1.136, 1.124, 1.124 Angstrom
VASP Arnardi report (ecut acell unknown)     : 1.136, 1.124,  NA   Angstrom
Experimental : 1.120 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t69.abi" target="_blank">t69.abi</a> &hellip;</li>
<li>mpiio:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si, Bulk, 2 atoms , paral_kgb.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/mpiio/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/mpiio/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
Si, Bulk, 2 atoms , paral_kgb.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test the triple parallelisation.
We cannot check the distribution npband*npfft*np_spkpt=2*2*2=8 processors,
this number being not allowed in the test procedure.
Here, we only test the parallelisation over FFT and kpoints:
npband*npfft*np_spkpt=1*2*2=4 processors.
In test tY.in we check other distributions for guarantee." data-toggle="popover" href="/tests/paral/Input/t24.abi" target="_blank">t24.abi</a> &hellip;</li>
<li>psml:  <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium, to compare with the
PSML results of t08 and t12" data-toggle="popover" href="/tests/psml/Input/t04.abi" target="_blank">t04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Total energy vs. cell size for bulk FCC aluminium" data-toggle="popover" href="/tests/psml/Input/t12.abi" target="_blank">t12.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=14 (finite reduced electric field calculation,relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_121.abi" target="_blank">tsv6_121.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=16 (finite reduced electric displacement field calculation,
relaxing cell parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_122.abi" target="_blank">tsv6_122.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 in the tetragonal geometry.
Test berryopt=4 (finite electric field calculation, relaxing cell
parameters, e.g. optcell=2)
(XG120616 : tranferred to tests/seq)" data-toggle="popover" href="/tests/seq/Input/tsv6_123.abi" target="_blank">tsv6_123.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_01.abi" target="_blank">timages_01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Hydronium ion + NH3 molecule
Ground state calculation
keeping O and H atoms fixed" data-toggle="popover" href="/tests/tutoparal/Input/timages_02.abi" target="_blank">timages_02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing.
The present test at the fixed, symmetric geometry, for reference. Of course, the cell is not primitive.
Parallelism until 12 procs is quite efficient. However, setting npband 2 does not bring noticeable improvement.
Although this is still subject for further investigation, the hypothesis is that the density/xc/potential part
is not improved with npband 2, while it is actually a major contribution, even with 32 bands.
Timing analysis is to be performed." data-toggle="popover" href="/tests/tutoparal/Input/tpsic_01.abi" target="_blank">tpsic_01.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
AlAs in hypothetical wurzite (hexagonal) structure
Structural optimization run" data-toggle="popover" href="/tests/tutorespfn/Input/telast_1.abi" target="_blank">telast_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Al fcc metal - elastic constant calculation" data-toggle="popover" href="/tests/tutorespfn/Input/telast_6.abi" target="_blank">telast_6.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Finite difference calculation of the clamped-ion
piezoelectric constants of AlP" data-toggle="popover" href="/tests/tutorespfn/Input/tpolarization_5.abi" target="_blank">tpolarization_5.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file will optimize automatically the interatomic distance" data-toggle="popover" href="/tests/tutorial/Input/tbase1_3.abi" target="_blank">tbase1_3.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom." data-toggle="popover" href="/tests/tutorial/Input/tbase2_1.abi" target="_blank">tbase2_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom.
Here, a double loop has been used." data-toggle="popover" href="/tests/tutorial/Input/tbase2_2.abi" target="_blank">tbase2_2.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Mo surface, with 10 atom per cell. Not very large, but the
number of bands make it a medium-size problem : 36 bands." data-toggle="popover" href="/tests/v1/Input/t41.abi" target="_blank">t41.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the option ionmov=4. 8-atom Si cube, non spin-polarized.
Same test as number 22a of fast, except for the change
of ionmov, no reading of wavefunctions, iscf to 2, and different
tolerance criteria." data-toggle="popover" href="/tests/v1/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the option ionmov=4. Si2 molecule, spin-polarized.
Same test as number 16 of fast, except for the change
of ionmov, no reading of wavefunctions, iscf to 2, and different
tolerance criteria.

Cannot be executed in parallel:
--- !ERROR
message: |
the number of bands in the spin up case must be equal to
the number of bands in the spin down case.
This is not the case for the k point number : 1
The number of bands spin up and down are : 5 3 Action : change nband, or use the sequential
version of ABINIT.
src_file: chkinp.F90
src_line: 1181
..." data-toggle="popover" href="/tests/v1/Input/t45.abi" target="_blank">t45.abi</a> &hellip;</li>
<li>v10:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test the post identification of more symmetric structures obtained after geometry optimization
Consider diamond and graphene, with low ecut and number of k points." data-toggle="popover" href="/tests/v10/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Arsenic in rhombohedral structure.
Optimization of unit cell size and shape.
With 4x4x4 k point grid, ecut 3.0, occopt 4 and tsmear 0.06d0,
one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277
These input parameters are too small, but still give a realistic
geometry, since with the much better parameters
12x12x12 k point grid, ecut 12.0, occopt 4 and tsmear 0.04d0,
one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296, while
the experimental values are
acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-toggle="popover" href="/tests/v2/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box.
Comparison of the modified Broyden algorithm (ionmov=3) with the
original one (ionmov=2). Start with different values
of xcart, from 0.6 to 1.1, by step of 0.1 . The number of Broyden
steps needed to reach acceptable residual forces
with the ionmov=3 algorithm are : 3, 2, 2, 3, 3, 4, while
with the ionmov=2 algorithms, one get : 4, 3, 2, 5, and then,
either the algorithm does not converge within 8 steps, or it
converges to a saddle point of the energy !
This test was hard to make portable. This is why the tolerance
for fldiff is very large." data-toggle="popover" href="/tests/v2/Input/t48.abi" target="_blank">t48.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Magnesium HCP. Optimize the cell volume and shape using different constraints.
Checks whether the stresses are symmetric, and whether the optimisation algorithm does
not break the symmetry.
Also checks the possibility to disable the stop after violation of the condition related to dilatmx.
This is done in dataset 6. The optimization delivers acell values slightly different from those of dataset 2." data-toggle="popover" href="/tests/v3/Input/t42.abi" target="_blank">t42.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule : examine the rotational freedom.
The present test produces the following
vibrational frequencies (with degeneracies indicated):
56.89 i cm-1 (2)
0.41   cm-1 (2)
1.05   cm-1
3800    cm-1
The large frequency corresponds to the stretching
mode, and has the right order of magnitude.
The frequencies close to 1 cm-1 corresponds
to translation modes, and are small enough
for usual applications.
The  56.89 i cm-1 mode corresponds to rotation of
the H2 molecule. The magnitude of this
frequency might seem quite
large. Here are the results of tests made to understand
this phenomenon. First, note that
ecut 12  acell 3*14
Increasing the value of ecut to 25 decreases
the magnitude of the frequency to 36.8 cm-1 .
However, in order to continue to make it smaller,
the cell size must be increased , and an oscillatory
behaviour is observed :
3*16   45.6 i cm-1
3*18   22.7   cm-1
3*20   19.1 i cm-1
3*22   15.7 i cm-1
3*24   13.7   cm-1
Many other tests have been set up. In particular,
it was observed that the frequency of the oscillatory
behaviour changes with the ecut, and also that
using the Gamma point, instead of the 1/4 1/4 1/4 k point
(used in this test) degrades the convergence.
The overall picture is as follows.
There are different reasons for the translation
and rotation modes to acquire a non-zero frequency
when plane waves and supercells are used.
Still, as concerns translations, only the
existence of a discretization of the XC grid
is important. For rotations, supercell effects
are also present :
- alignement of dipole or quadrupoles
- interaction between tails of wavefunctions, accross cells
Since the convergence in supercell size is oscillatory, we infer
that the breaking of the rotational invariance is mostly
due to interaction between wavefunction tails.
This will be checked by confining the system in a spherical
well, in a forthcoming test." data-toggle="popover" href="/tests/v3/Input/t80.abi" target="_blank">t80.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of geometry optimization using using ionmov=1,4,5 and 7
Hydrogen molecule inside a box of 12 Bohr
This test was created to test ionmov=5" data-toggle="popover" href="/tests/v5/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Al2 molecule in a big box. Treat 8 bands, occupied using occopt=4.
Compare tolrff and toldff stopping criteria" data-toggle="popover" href="/tests/v5/Input/t41.abi" target="_blank">t41.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
White tin, body-centered tetragonal. This test checks that in an optcell 2 run, the symmetry
of the body-centered case is not broken. This test is run for two values of strprecon also,
to test the strprecon variable.
Coded and contributed by J. Zwanziger." data-toggle="popover" href="/tests/v5/Input/t77.abi" target="_blank">t77.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Delocalized internals test. Si diamond bulk with displaced atom
Unconverged in all variables" data-toggle="popover" href="/tests/v6/Input/t23.abi" target="_blank">t23.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 Molecule in a small box
Second-order eigenvalue calculation using a small number of bands, testing bdeigrf." data-toggle="popover" href="/tests/v6/Input/t61.abi" target="_blank">t61.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test tolmxde input variable. For silicon supercell with atomic positions randomly displaced" data-toggle="popover" href="/tests/v7/Input/t15.abi" target="_blank">t15.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Relaxation of FCC Argon crystal with DFT-D3 dispersion correction
Both the pair-wise term and the 3-body term are included
In this test the following properties are thus tested: contribution
to the energy and to the stress of DFT-D3 including the 3-body term" data-toggle="popover" href="/tests/v7/Input/t97.abi" target="_blank">t97.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="Geometry optimization with FIRE (ionmov 15)" data-toggle="popover" href="/tests/v8/Input/t17.abi" target="_blank">t17.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 (spin-polarized, non-linear XC core correction, LDA)
Test the input variable imgwfstor.
In itimimage=2, the SCF cycle and the Broyden optimization are indeed immediately fulfilled." data-toggle="popover" href="/tests/v8/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
LiH molecule
test the use of spatial chemical potential. The molecule starts aligned along the x direction,
while the Li chemical potential pulls the Li nucleus toward -z, and the H chemical potential
pushes the H nucleus toward +z. See whether the optimization runs properly." data-toggle="popover" href="/tests/v8/Input/t31.abi" target="_blank">t31.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
NaI with a sligthly expanded lattice, to allow self-localisation of a hole polaron, using pSIC technique.
Realistic ecut and reasonable k point sampling, but small cell of only 8 atoms for testing." data-toggle="popover" href="/tests/v9/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PbTiO3 with in-plane strain.
Relaxing the third vector using optcell 6" data-toggle="popover" href="/tests/v9/Input/t104.abi" target="_blank">t104.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test the post identification of more symmetric structures obtained after geometry optimization
Consider diamond and graphene, with low ecut and number of k points." data-toggle="popover" href="/tests/v9/Input/t189.abi" target="_blank">t189.abi</a> &hellip;</li>
</ul>
</details>
<p>Sets a maximal absolute force tolerance (in hartree/Bohr) below which BFGS
structural relaxation iterations will stop.
Can also control tolerance on stresses, when <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> /=0, using the
conversion factor <a class="wikilink codevar-wikilink" href="/variables/rlx#strfact">strfact</a>. This tolerance applies to any particular
cartesian component of any atom, excluding fixed ones. See the parameter
<a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a>.
This is to be used when trying to equilibrate a structure to its lowest energy
configuration.
A value of about 5.0d-5 hartree/Bohr or smaller is suggested (this corresponds
to about 2.5d-3 eV/Angstrom).
No meaning for RF calculations.</p>
<h2 id="vel"><strong>vel</strong><a class="headerlink" href="#vel" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> VELocity<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-tippy-content="How to perform a geometry optimization" data-toggle="popover" href="/topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3,<a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>)<br />
<em>Commentdims:</em> It is represented internally as <a class="wikilink codevar-wikilink" href="/variables/rlx#vel">vel</a>(3,<a class="wikilink codevar-wikilink" href="/variables/basic#natom">natom</a>,<a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a>)<br />
<em>Default value:</em> <em>0<br />
*Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> or <a class="wikilink codevar-wikilink" href="/variables/rlx#moldyn">moldyn</a> /= none (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> &gt; 0)<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [107/1314] in all abinit tests, [16/178] in abinit tutorials</summary>
<ul>
<li>bigdft:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule with default parameters. No forces are computed, with steepest decent
for the direct minimisation of the wavefunctions." data-toggle="popover" href="/tests/bigdft/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule with default parameters. No forces are computed, with DIIS
for the direct minimisation of the wavefunctions." data-toggle="popover" href="/tests/bigdft/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule with default parameters. Forces are computed." data-toggle="popover" href="/tests/bigdft/Input/t03.abi" target="_blank">t03.abi</a> &hellip;</li>
<li>bigdft_paral:  <a class="wikilink abifile-wikilink" data-tippy-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0)
Test: DIIS, potential/density mixing and parallelism" data-toggle="popover" href="/tests/bigdft_paral/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0)
Test: DIIS, potential/density mixing and parallelism" data-toggle="popover" href="/tests/bigdft_paral/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="CO2 in a box (PAW). ABINIT routines are used (wvl_bigdft_comp=0)
Test: DIIS, density mixing" data-toggle="popover" href="/tests/bigdft_paral/Input/t02.abi" target="_blank">t02.abi</a> &hellip;</li>
<li>gwr_suite:  <a class="wikilink abifile-wikilink" data-tippy-content="Hartree-Fock with time-reversal, then CC4S output" data-toggle="popover" href="/tests/gwr_suite/Input/t10.abi" target="_blank">t10.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Hartree-Fock without time-reversal, then CC4S output" data-toggle="popover" href="/tests/gwr_suite/Input/t11.abi" target="_blank">t11.abi</a> &hellip;</li>
<li>libxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
Isolated Carbon atom
Test the mGGA functionals generating directly XC potentials.
To make the test more portable, we generate GGA-PBE wave functions as starting point.
Similar to test 08, but with smaller acell and ecut.
Without the use of a &quot;large&quot; xc_denpos, Tran-Blaha is hard to converge. Hypothesis : there is a strong non-linear
region, preventing the Anderson method to work effectively.
Note that the default iscf (=7) does not allow to converge with the default xc_denpos !
By contrast, things are much better behaved with xc_denpos on the order of 1.0e-7 ." data-toggle="popover" href="/tests/libxc/Input/t13.abi" target="_blank">t13.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Diamond: G0W0 @ scHSE06 calculation. Monitor the direct gap at Gamma.

First, with the scGW methodology based on a Kohn-Sham basis,
then doing the scHSE06 using the planewave basis, followed by a one-shot G0W0.
The agreement is reasonable with the parameters used in the automatic test,
but can be improved with better parameters (esp. nband), see later.
At the PBE level, the (KS) band gap is 5.231 eV,
At the scHSE06 level, the Kohn-Sham basis delivers 6.834 eV,
while the plane wave basis delivers 6.850 eV.
At the G0W0-scHSE06 level, the Kohn-Sham basis delivers 7.314 eV,
while the plane wave basis delivers 7.344 eV.
The macroscopic dielectric constant (at Gamma) is 9.3698 from PBE,
5.8309 from scHSE06(KS) and 5.8202 from scHSE06(planewaves).

These calculations have also been done with better
parameters, in order to observe a better agreement between the KS basis set and the
planewave basis set (ecut 20  ecutsigx 20  nband 30 gw_qprange 30 - note however that ecuteps 2 is low),
at the expense of CPU time..
At the PBE level, the (KS) band gap was 5.661 eV,
At the scHSE06 level, the Kohn-Sham basis delivered 7.340 eV,
while the plane wave basis delivered 7.341 eV.
At the G0W0-scHSE06 level, the Kohn-Sham basis delivered 7.806 eV,
while the plane wave basis delivered 7.807 eV.
The macroscopic dielectric constant (at Gamma) was 7.845 from PBE,
5.033 from scHSE06(KS) and 5.032 from scHSE06(planewaves)." data-toggle="popover" href="/tests/libxc/Input/t44.abi" target="_blank">t44.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
N2 molecule non-spin-polarized, in a big box.
PAW : first PBE, then PBE0, then HSE06. Interatomic distance optimization.
Test the geometry convergence in the hybrid functional case
The results obtained with underconverged ecut (ecut 18) and box size (5 4 4 Angstrom)
compare favourably with those mentioned in Arnardi&#x27;s report, as well as with those of VASP.
Values for the distance d, in Angstrom, for PBE, PBE0 and HSE06 respectively :
Present calculation  (ecut 18, 5 4 4 box) : 1.101, 1.084, 1.085 Angstrom
Better  calculation  (ecut 22, 6 5 5 box) : 1.102, 1.086, 1.086 Angstrom
ABINIT Arnardi report (ecut acell unknown): 1.103, 1.091, 1.092 Angstrom
VASP Arnardi report (ecut acell unknown)  : 1.103, 1.089,  NA   Angstrom
Experimental : 1.098 Angstrom." data-toggle="popover" href="/tests/libxc/Input/t67.abi" target="_blank">t67.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
C-diamond, Bulk, 2 atoms, paral_kgb, with PAW.
Test of ground state with different occs(7 and 0),
and also ionmov 2. Only with 0 and 4 procs." data-toggle="popover" href="/tests/paral/Input/t22.abi" target="_blank">t22.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Test local susceptibility from CTQMC at scalar level. Not realistic" data-toggle="popover" href="/tests/paral/Input/t100.abi" target="_blank">t100.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Test local susceptibility from CTQMC at spin-orbit level. Not realistic" data-toggle="popover" href="/tests/paral/Input/t101.abi" target="_blank">t101.abi</a> &hellip;</li>
<li>seq:  <a class="wikilink abifile-wikilink" data-tippy-content="
81. Nitrogen atom in a big box, spin-polarized.
Compute Berry phase for atom at different positions." data-toggle="popover" href="/tests/seq/Input/tsv2_81.abi" target="_blank">tsv2_81.abi</a> &hellip;</li>
<li>tutoparal:  <a class="wikilink abifile-wikilink" data-tippy-content="BaTiO3 linear response calculation" data-toggle="popover" href="/tests/tutoparal/Input/tdfpt_04.abi" target="_blank">tdfpt_04.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Input for PARAL_GSWVL tutorial
14 atom boron cluster, parallel calculation" data-toggle="popover" href="/tests/tutoparal/Input/tgswvl_1.abi" target="_blank">tgswvl_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Input for PARAL_GSWVL tutorial
14 atom boron cluster, parallel calculation" data-toggle="popover" href="/tests/tutoparal/Input/tgswvl_1.abi" target="_blank">tgswvl_1.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="Temperature dependence calculation of diamond." data-toggle="popover" href="/tests/tutorespfn/Input/teph_tdep_legacy_1.abi" target="_blank">teph_tdep_legacy_1.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Temperature dependence calculation of diamond. Quick determination of the k-point grid in the IBZ." data-toggle="popover" href="/tests/tutorespfn/Input/teph_tdep_legacy_2.abi" target="_blank">teph_tdep_legacy_2.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Temperature dependence calculation of diamond." data-toggle="popover" href="/tests/tutorespfn/Input/teph_tdep_legacy_3.abi" target="_blank">teph_tdep_legacy_3.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box
This file to optimize the H2 bond length, compute the associated total
energy, then to compute the total energy of the isolated H atom.
Here, the ecut and acell are fixed : the double loop reduces
effectively to a single loop." data-toggle="popover" href="/tests/tutorial/Input/tbase2_4.abi" target="_blank">tbase2_4.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
O2 molecule, treated like a metal (there is a degeneracy at the Fermi level),
occopt=4 and tsmear=0.04 , with nsppol=2 .
Use spinat to polarize the molecule at start.
Gives total energy of -32.09317 Hartree
This test can be used to check the accuracy of stresses in the
spin-polarized case. Setting acell(1:2) to 7.005 and 6.995
gives ETOT -32.090846193972 and -32.095501753750,
ucvol  441.63023 and 440.37023
so that the estimation of stress by d(Etot)/d(Vol) gives 3.6948887E-03,
to be compared with sigma(1)=3.6949713786E-03
and sigma(2)=3.6949681083E-03 from the present test
(the degeneracy is slightly lifted because of incomplete convergence,
using tolvrs 1.0d-12 instead of toldfe gives at least 8 identical digits for both
sigma values)." data-toggle="popover" href="/tests/v1/Input/t08.abi" target="_blank">t08.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Helium FCC solid, using bare potential, but a relatively low cut-off.
Cell parameter is 6.505 Bohr , 2 k points are used.  PBE GGA.
Combined with test 29, it is used to test the accuracy of
the stress calculation. Output of the code are as follows :
Energy=-2.835212586 Ha, volume=68.814809 Bohr^3,
stress=6.03381838E-04 Ha/Bohr^3" data-toggle="popover" href="/tests/v1/Input/t28.abi" target="_blank">t28.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Isolated tin atom, with PseudosTM_pwteter/50sn.pspnc pseudopotential (Troullier-Martins),
to be compared with the result of test 32.
The size of the box is 12x12x12, One k-point (1/4 1/4 1/4) is used.
8 elements of symmetry are present. The cut-off is 14 Hartree.
The energy levels (eV) are found at -7.162, -0.273 and -0.228 (two-fold deg)
The difference between these energy levels is 6.889 and 0.045.
Going to larger unit cells decrease the latter difference
(14x14x14 makes it 0.010)." data-toggle="popover" href="/tests/v1/Input/t31.abi" target="_blank">t31.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
H2 molecule in a big box : compute VERY accurately
the derivatives of the energy, by both symmetric finite-differences and
direct computation of forces and 2DTE.
Also test the interplay between istwfk/=1 in the GS calculation
and istwfk==1 in the RF calculation (istwfk/=1 is not yet-991020-
allowed for RF, which is a shame)
1) Computation of the first-order derivative of the total energy
With delta(xred)=0.0002, one gets delta(etot)/delta(xred)=-3.145846551
With delta(xred)=0.0001, one gets delta(etot)/delta(xred)=-3.145836932
The combination of both results, in a higher-order finite difference
formula gives -3.145833726 . The direct computation of forces
at the target geometry gives -3.145833725869 . The agreement is perfect,
taking into account the &quot;limited&quot; number of digits (10) of the
finite-difference result.
2) Computation of the second-order derivative of the total energy
With delta(xred)=0.0002, one gets delta(dedt)/delta(xred)=188.73875
With delta(xred)=0.0001, one gets delta(dedt)/delta(xred)=188.73837
The combination of both results, in a higher-order finite difference
formula gives 188.73824613 . The direct computation of 2DTE
at the target geometry gives 188.73824613046 . The agreement at the
level of 11 digits is also perfect." data-toggle="popover" href="/tests/v2/Input/t33.abi" target="_blank">t33.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Aluminum in FCC structure.
At fixed tsmear (=0.04), check the convergence with respect
to the number of k points, for different occopt.
The grids that are tested have respectively 10, 28, and 60 k points
in the IZB (ngkpt values are 4 4 4 4, 6 6 6 4 and 8 8 8 4
For occopt=4, the total energy (or the enthalpy)
is -2.079770 Ha, -2.080534 Ha, -2.080672 Ha ;
for occopt=5, it is -2.079787 Ha, -2.080563 Ha, -2.080693 Ha ;
for occopt=7, it is -2.084211 Ha, -2.084861 Ha, -2.084916 Ha.
The latter values can be corrected by computing the mean of the
total energy and the internal energy, as described in Marzari&#x27;s
thesis, for example. Thanks to this procedure, one gets :
-2.079732 Ha, -2.080470 Ha, and -2.080626 Ha.
For information, with a 12 12 12 4 grid (182 k points)
and occopt=4, one gets -2.080648 Ha." data-toggle="popover" href="/tests/v2/Input/t46.abi" target="_blank">t46.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ta, single atom in a box (1k point). HGH pseudopotential
Ecut and acell are NOT realistic. One would need 20 Ha
and a 20x20x20 box, while here we have 5 Ha and a 12x12x12 box.
The small box size changes the degeneracy of levels.
The all-electron values derived from another code, for comparison, are
level  degeneracy       energy (Ha)  spin-orbit splitting
6s      2               -0.194573
5d3/2   4               -0.141779
5d5/2   6               -0.119933       0.0218
6p1/2   2               -0.050542
6p3/2   4               -0.031281       0.0193" data-toggle="popover" href="/tests/v2/Input/t84.abi" target="_blank">t84.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
Fe, one atom in a big box.
Test spin-polarized GS and RF calculation, at fixed occupation
number. Check acoustic sum rule.
The ecut is 18 Hartree. It seems quite low, but gives
phonon frequency of 3.83i cm-1, which is quite small,
while increasing it to 22 Ha or 26 Ha, respectively,
gives 4.73i cm-1 and 1.32i cm-1, respectively." data-toggle="popover" href="/tests/v3/Input/t19.abi" target="_blank">t19.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
O2 molecule, treated like a metal (there is a degeneracy at the
Fermi level), occopt=4 and tsmear=0.04 , with nsppol=2 .
Use spinat to polarize the molecule at start.
Same as test v1 #8, except use different energy and length units.
Test the use of dimensional and logical input
variables." data-toggle="popover" href="/tests/v3/Input/t53.abi" target="_blank">t53.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test on splitting screening calculation using nqptdm and qptdm
Si in 2-atom diamond unit cell (nkpt 2; ecut 6)
In dataset 1, ABINIT finds the ground state (tolwfr 1.0d-16)
the eigenvectors necessary for the GW calculation being stored in _WFK.
In datasets 2-4, ABINIT computes the eps^-1 matrix for 1 qpt , 2 qpts, 3qpts respectively" data-toggle="popover" href="/tests/v3/Input/t87.abi" target="_blank">t87.abi</a> &hellip;</li>
<li>v4:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the pspcod=8 input format (contributed by D.R. Hamann)
This format is primarily intended to offer a flexible platform
for those who wish to experiment with their own features in
norm-conserving pseudopotentials.  A full description and discussion
is given on the ABINIT wiki
https://wiki.abinit.org/doku.php?id=developers:pseudos:psp8_format
This test case, CaO, is an
example of self-interaction-corrected pseudopotentials in the
style of D. Vogel, P. Krueger, and J. Pollmann, Phys. Rev. B 52,
14 316 (1995).  The bandgap (direct) in the test example, 11.2eV,
is in excellent agreement with experiment, but is rather strongly
dependent on the radius at which the self-interaction-correction
of Ca is (smoothly) cut off.  The sic pseudopotentials used in
this test should not be considered anything but early-stage
experiments and should not be used for any other purpose." data-toggle="popover" href="/tests/v4/Input/t20.abi" target="_blank">t20.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ta, single atom in a box (1k point). Same as test v2 #84
HGH pseudopotential
Ecut and acell are NOT realistic. One would need 20 Ha
and a 20x20x20 box, while here we have 5 Ha and a 12x12x12 box.
Prepare the analysis of next test
level  degeneracy       energy (Ha)  spin-orbit splitting
6s      2               -0.194573
5d3/2   4               -0.141779
5d5/2   6               -0.119933       0.0218
6p1/2   2               -0.050542
6p3/2   4               -0.031281       0.0193" data-toggle="popover" href="/tests/v4/Input/t36.abi" target="_blank">t36.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the strain perturbation with the GGA
Note that this test and the following one, 59,
address more advanced features of strain perturbation than the ones
going with number 60 to 79, and were developed later, but were
placed here for convenience).
AlAs in the zincblende structure, with a choice of
pseudopotenials that include model core charges.  In the LDA, the
strain perturbation simply introduces perturbing charge densities
leading to perturbing first-order potentials in a straightforward
manner.  (This perturbation does act differently on the valence
and model core charges.)   The GGA is a non-trivial extension,
since strain perturbs the gradient operator itself.  This test
is a symmetrized and stripped-down version of a test case used in
development which yielded agreement between such RF calculations
and numerical derivatives of GS stresses and forces to 1.e-8 to
1.e-9 atomic units for the elastic tensor and intenal strain
coupling parameters.  Acheiving this agreement does require forcing
a considerably more dense fft grid than the default which is
accepted here. (ngfft= 50 50 50 was used.)  While only the PBE GGA
is tested here, all forms present in Abinit should give comparably
accurate results." data-toggle="popover" href="/tests/v4/Input/t58.abi" target="_blank">t58.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-toggle="popover" href="/tests/v5/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
PAW: test of some variables reserved to experienced users or developers
BCC iron, ferromagnetic, GGA, PAW. Also tests maxnsym keyword." data-toggle="popover" href="/tests/v5/Input/t06.abi" target="_blank">t06.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Ni atom in a big box.
Test the use of the atvshift variable :
change the potential of the d orbitals." data-toggle="popover" href="/tests/v5/Input/t14.abi" target="_blank">t14.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Chain of Silicon diatomic molecules (4 Si2 molecules in the cell)
Freeze oscillatory perturbations with different wavelengths and intensities,
thanks to the qprtrb and vprtrb input variables.
This should be linked with the computation of the dielectric constant,
test v2#05, that uses directly the RF capabilities of ABINIT,
for one diatomic molecule.
For dataset 1, one reproduces the results obtained in Tv2#05, multiplied by 4.
The total energy is consistent up to more than 10 digits :
-6.6499924738006 Ha for Tv2#05, -26.599969895203 Ha for the present calculation.
For dataset 2, the perturbation qprtrb 0 0 1 is frozen in, with vprtrb 100.
The total energy is -26.600317638775 Ha. The difference wrt the unperturbed situation is
0.000348743572 Ha.
For dataset 3, a much smaller perturbation (10 times smaller) is taken,
giving total energy -26.599973367786 Ha. The difference wrt the unperturbed situation is
0.3472583 microHa.
For dataset 4, an even smaller perturbation (100 times smaller) is taken,
giving total energy -26.599969929928 Ha. The difference wrt the unperturbed situation is
0.000034725 microHa. With datasets 3 and 4, we are in the linear regime.
The previous amplitude is better for such studies.
Dataset 5 is the same as 3, with reversed amplitude. Results are similar to dataset 3.
I had no sufficient time to analyze these data correctly and make the connection with the
results of Tv2#05, unfortunately. The following (also test 02 below) gives some more data, and raise questions.
There might be some problem with the use of qprtrb and vprtrb.
For dataset 2, the group of the four lowest eigenenergies (each corresponding to a different molecule) is :
-0.47198  -0.46381  -0.46091  -0.45266 , whose spread is 0.01932 Ha.
One might think that the maximum and minimum of the potential are separated roughly by 0.02 Ha.
The value vprtrb 100 corresponds to a cosine wave whose amplitude is 100, divided by the volume
of the cell, that is 5000 Bohr^3 : 0.02 Ha. The maximum
and minimum of the potential should thus be separated by 0.04 Ha. There seems to be a factor of 2 off." data-toggle="popover" href="/tests/v6/Input/t01.abi" target="_blank">t01.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Chain of Silicon diatomic molecules (4 Si2 molecules in the cell)
Freeze oscillatory perturbations with different wavelengths and intensities,
thanks to the qprtrb and vprtrb input variables.
Compute the dielectric constant. Similar to test v6#01,
but uses a more symmetric geometry, to examine invariance
of the response with respect to shifts of potential, and also
a shorter wavelength. I do not understand why vprtrb 0 10.0 leads to no response.
Such a sine wave should cause similar response as for the cosine wave.
No time presently to investigate this problem (XG090909)" data-toggle="popover" href="/tests/v6/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
MgB2, space group P6/m m m (#191); Bravais hP (primitive hexag.)
GS calculation followed by a band structure calculation in which the
k-path is automatically defined via the ndivsm input variable.
The third dataset tests the calculation and the output of the Fermi surface (prtfsurf=1)
in the Xcrysden format." data-toggle="popover" href="/tests/v6/Input/t14.abi" target="_blank">t14.abi</a> &hellip;</li>
<li>v67mbpt:  <a class="wikilink abifile-wikilink" data-tippy-content="
Isolated hydrogen atom
Calculation of the HF levels with different ways to integrate the Coulomb divergence" data-toggle="popover" href="/tests/v67mbpt/Input/t09.abi" target="_blank">t09.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Silicon: Solution of the Bethe-Salpeter equation (BSE) with norm-conserving pseudopotentials.
W is calculated at the RPA level while the scissors operator is used to open the gap by 0.8 eV.
First the BSE is solved with the direct diagonalization of the two-particle Hamiltonian, then
the Haydock iterative method is employed to calculate the macroscopic dielectric function.
The last dataset solves the BSE problem including the coupling between resonant and
anti-resonant transition via brute force diagonalization." data-toggle="popover" href="/tests/v67mbpt/Input/t11.abi" target="_blank">t11.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Molecular SiH4 in GWLS (G0W0 with Lanczos basis and Sternheimer equations)" data-toggle="popover" href="/tests/v67mbpt/Input/t15.abi" target="_blank">t15.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="Random stopping power within RPA and TD-LDA" data-toggle="popover" href="/tests/v7/Input/t16.abi" target="_blank">t16.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
EuO rock salt structure.
20-electron PAW pseudo (4f in the core, but 4d10 as semi-core : 4d10 5s2 5p6 (5d0) 6s2 )
Test of computation of dielectric tensor, Born effective charges, dynamical matrix.
To keep the CPU time to a reasonable level, the responses are computed
with minimal parameters (nstep and the kpoint grid are too small,
ecut might be increased, and also pawecutdg).
Note: Charge neutrality is not achieved with the present dataset for Born Effective Charges,
but can be easily reached by increasing some parameters; for instance:
(ngkpt 6 6 6,  ecut 40. pawecutdg 40.) gives Z(Eu)=2.5085, Z(O)=-2.4982" data-toggle="popover" href="/tests/v7/Input/t43.abi" target="_blank">t43.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Diamond. Test the getddb variable that cancels the divergence in el-ph for small q-points
Such divergence is induced by a residual electric field.
Note: The Fan terms in DATASET 6 should be different (renormalized) from the one of DATASET 5." data-toggle="popover" href="/tests/v7/Input/t58.abi" target="_blank">t58.abi</a> &hellip;</li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="Test generation of k-point mesh with kptrlatt and shiftk.
version 8.0.8b was stopping at the level of the parser." data-toggle="popover" href="/tests/v8/Input/t03.abi" target="_blank">t03.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Simple test of velocity verlet integrator" data-toggle="popover" href="/tests/v8/Input/t12.abi" target="_blank">t12.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of molecular dynamics of B atoms, in the isokinetic
ensemble, based on the Thomas-Fermi functional (very fast, even with 32 atoms).
Fix the position of one atom along one direction." data-toggle="popover" href="/tests/v8/Input/t22.abi" target="_blank">t22.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
Calculation of the imaginary part of the Fan-Migdal self-energy matrix elements.
Test of the calculation of the Fermi level under different conditions" data-toggle="popover" href="/tests/v9/Input/t56.abi" target="_blank">t56.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Preliminary step for test 108:
bcc Fe ferromagnetic structure. Relatively low cut-offs and few k-points.
This test is done within GGA and magnetism in PAW formalism.
Wavefunction printed to be used in v9#106 and v9#107 tests.
Density is printed out to prepare v9#108 test." data-toggle="popover" href="/tests/v9/Input/t105.abi" target="_blank">t105.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="Antisymmetric vector potential response and quadrupoles of HF molecule" data-toggle="popover" href="/tests/v9/Input/t146.abi" target="_blank">t146.abi</a> &hellip;</li>
</ul>
</details>
<p>Gives the starting velocities of atoms, in cartesian coordinates, in
Bohr/atomic time units (atomic time units given where <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a> is described).
For <a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> = 8 (Nose thermostat), if  <strong>vel</strong>  is not initialized, a random
initial velocity giving the right kinetic energy will be generated.
If the atom manipulator is used,  <strong>vel</strong>  will be related to the preprocessed
set of atoms, generated by the atom manipulator. The user must thus foresee
the effect of this atom manipulator (see <a class="wikilink codevar-wikilink" href="/variables/geo#objarf">objarf</a>).
Velocities evolve is <a class="wikilink codevar-wikilink" href="/variables/rlx#geoopt">geoopt</a> == &ldquo;viscous&rdquo; (<a class="wikilink codevar-wikilink" href="/variables/dev#ionmov">ionmov</a> == 1).</p>
<h2 id="vel_cell"><strong>vel_cell</strong><a class="headerlink" href="#vel_cell" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> VELocity of the CELL parameters<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">EVOLVING</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a PIMD calculation" data-toggle="popover" href="/topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3,3)<br />
<em>Commentdims:</em> It is represented internally as <a class="wikilink codevar-wikilink" href="/variables/rlx#vel_cell">vel_cell</a>(3,3,<a class="wikilink codevar-wikilink" href="/variables/rlx#nimage">nimage</a>)<br />
<em>Default value:</em> <em>3<br />
*Only relevant if:</em> <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> in [9,13] and <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a> &gt; 0
(Path-Integral Molecular Dynamics with NPT algorithm)<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [0/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
</details>
<p>Irrelevant unless <a class="wikilink codevar-wikilink" href="/variables/rlx#imgmov">imgmov</a> = 9 or 13 and <a class="wikilink codevar-wikilink" href="/variables/rlx#optcell">optcell</a>&gt;0 (Path-Integral
Molecular Dynamics with NPT algorithm).
Gives the starting velocities of the dimensional cell parameters in
Bohr/atomic time units (atomic time units given where <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a> is described).</p>
<h2 id="vis"><strong>vis</strong><a class="headerlink" href="#vis" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> VIScosity<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to perform a molecular dynamics calculation" data-toggle="popover" href="/topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Moderately used, [29/1314] in all abinit tests, [10/178] in abinit tutorials</summary>
<ul>
<li>etsf_io:  <a class="wikilink abifile-wikilink" data-tippy-content="
Bulk water ice 1h, made-up with 8 water molecules.
Density is exported using the ETSF I/O file format.
Should be a nice system to look at with visualisation tools." data-toggle="popover" href="/tests/etsf_io/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>fast:  <a class="wikilink abifile-wikilink" data-tippy-content="
21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-toggle="popover" href="/tests/fast/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
move with natfix.
Output also GEO files" data-toggle="popover" href="/tests/fast/Input/t29.abi" target="_blank">t29.abi</a> &hellip;</li>
<li>libxc:  <a class="wikilink abifile-wikilink" data-tippy-content="
Isolated Helium atom
Compare the results using the native XC library and the
corresponding ones of LibXC, in the non-spin-polarized case :
5 LDAs (Teter93, Perdew-Zunger, Wigner, Hedin-Ludqvist, PW92)
7 GGAs (PBE, RPBE, HCTH93, HCTH120, HCTH147, HCTH407, WuCohen)
Results are excellent (better than 10 microHa)." data-toggle="popover" href="/tests/libxc/Input/t01.abi" target="_blank">t01.abi</a> &hellip;</li>
<li>paral:  <a class="wikilink abifile-wikilink" data-tippy-content="
SiC in zinc-blende structure
Calculation of the GW corrections with GWR code.
Test symmetrization with time-reversal and umklapp g0 vectors
as well as parallelization over k/g/tau." data-toggle="popover" href="/tests/paral/Input/t79.abi" target="_blank">t79.abi</a> &hellip;</li>
<li>tutoplugs:  <a class="wikilink abifile-wikilink" data-tippy-content="Test interface with Wannier90 (PAW pseudopotentials)" data-toggle="popover" href="/tests/tutoplugs/Input/tw90_5.abi" target="_blank">tw90_5.abi</a> &hellip;</li>
<li>tutorespfn:  <a class="wikilink abifile-wikilink" data-tippy-content="
LiF, Fm-3m space group.
Preparatory run for VarPEq calculations (hole polaron).
Computation of electron-phonon matrix elements on 3x3x3, 4x4x4, 5x5x5
k/q-grids using GSTORE." data-toggle="popover" href="/tests/tutorespfn/Input/teph4vpq_3.abi" target="_blank">teph4vpq_3.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
LiF, Fm-3m space group.
Computation of a small hole polaron on 3x3x3, 4x4x4, 5x5x5 k/q-grids
via Variational Polaron Equations with long-range Frohlich correction." data-toggle="popover" href="/tests/tutorespfn/Input/teph4vpq_4.abi" target="_blank">teph4vpq_4.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
LiF, Fm-3m space group.
Computation of a small hole polaron on 3x3x3, 4x4x4, 5x5x5 k/q-grids
via Variational Polaron Equations without long-range Frohlich correction." data-toggle="popover" href="/tests/tutorespfn/Input/teph4vpq_5.abi" target="_blank">teph4vpq_5.abi</a> &hellip;</li>
<li>tutorial:  <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline silicon

Computation of the band structure.
First, a SCF density computation, then a non-SCF band structure calculation." data-toggle="popover" href="/tests/tutorial/Input/tbase3_5.abi" target="_blank">tbase3_5.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
H6 Supercell
Generates a 6 atom Hydrogen supercell with the multiplicity of 1:2:3 (x:y:z)
Produces a WFK file to be analyzed with fold2bloch" data-toggle="popover" href="/tests/tutorial/Input/tfold2bloch_1.abi" target="_blank">tfold2bloch_1.abi</a> &hellip;</li>
<li>v1:  <a class="wikilink abifile-wikilink" data-tippy-content="
Isolated Helium atom (see above). Treated with LDA, Hedin-Lundqvist functional (ixc=5).
The total energy is -2.8335 Ha. For speed-up, it begins with the wavefunctions from test 11." data-toggle="popover" href="/tests/v1/Input/t15.abi" target="_blank">t15.abi</a> &hellip;</li>
<li>v2:  <a class="wikilink abifile-wikilink" data-tippy-content="
Basic test of geometry optimization for water
First computing the ground state of a close configuration
Uses 5 different methods to find the relaxed positions" data-toggle="popover" href="/tests/v2/Input/t88.abi" target="_blank">t88.abi</a> &hellip;</li>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a> &hellip;</li>
<li>v5:  <a class="wikilink abifile-wikilink" data-tippy-content="
Test of geometry optimization using using ionmov=1,4,5 and 7
Hydrogen molecule inside a box of 12 Bohr
This test was created to test ionmov=5" data-toggle="popover" href="/tests/v5/Input/t02.abi" target="_blank">t02.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Generate both DEN and PAWDEN files for silicon. Note that in the PAW case, DEN files
include the valence density and compensation charge, while PAWDEN files include the
valence and on-site corrections, but not the compensation charge. Therefore, DEN files are
needed for further computations, while PAWDEN files are useful for visualization, AIM, and other
post-processing of the density.
PAW AE DEN not compatible with parallelization over atoms." data-toggle="popover" href="/tests/v5/Input/t11.abi" target="_blank">t11.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline Silicon to test the CP2K pseudopotential implementation with
the new pseudo type: pspcod 10 (same formalism as HGH pseudopotentials, pspcod 3).
Same input file as t35.in of tests/tutorial, the only difference is that
in the test the used pseudo is taken from the CP2K library.
(http://cvs.berlios.de/cgi-bin/viewcvs.cgi/cp2k/potentials/Goedecker/abinit/)
The bandstructure is in agreement with that calculated in the tutorial
example with the pspcod=1 pseudopotential." data-toggle="popover" href="/tests/v5/Input/t54.abi" target="_blank">t54.abi</a> &hellip;</li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
MgB2, space group P6/m m m (#191); Bravais hP (primitive hexag.)
GS calculation followed by a band structure calculation in which the
k-path is automatically defined via the ndivsm input variable.
The third dataset tests the calculation and the output of the Fermi surface (prtfsurf=1)
in the Xcrysden format." data-toggle="popover" href="/tests/v6/Input/t14.abi" target="_blank">t14.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Bismuth, rhombohedral, two atoms per unit cell
Scalar relativistic.
Relax four different images, using the steepest descent algorithm." data-toggle="popover" href="/tests/v6/Input/t21.abi" target="_blank">t21.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline silicon. Diamond structure.
Quick test of the use of datasets with a numbering beyond 1000 (the last one is 1022).
(Tests 35-40 are for response-function)" data-toggle="popover" href="/tests/v6/Input/t34.abi" target="_blank">t34.abi</a> &hellip;</li>
<li>v7:  <a class="wikilink abifile-wikilink" data-tippy-content="Test magnetic constraint code. Variables magconon and magcon_lambda. BCC iron with low cutoff
and kpts, non collinear spin case with nspden=4 imposed. First normal case then constrain direction only
along z (DS2), along y (DS3) and impose value of vector (DS4). Then work in the collinear case, unconstrained (DS5),
with constrained direction (DS6), and constrained direction and value (DS7).
Total energy in DS1, DS2, DS3 and DS5 are identical (good). Should be also identical in DS6 , but not the case at present.
Also, the results for DS3 are NOT really portable, hence the very large tolerances.
Revised by XG" data-toggle="popover" href="/tests/v7/Input/t05.abi" target="_blank">t05.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the extension of the pspcod=8 input format to include
spin-orbit coupling with multiple non-local projectors.  The
band structure of fcc Pt is computed at Gamma, X, and L.
The pseudopotential used in the test is generated by the open-source
ONCVPSP code, which is available at www.mat-simresearch.com, and
described in D. R. Hamann, Phys. Rev. B 88, 085177 (2013).
The relativistic extension is documented in the code package.
The Pt 5s and 5p semi-cores are treated as valence, and the band
structure is well-converged at the ecut used in the test (20 Ha).
A complete description of the input format is avaiable in
doc/psp_infos/psp8_info.txt." data-toggle="popover" href="/tests/v7/Input/t07.abi" target="_blank">t07.abi</a>, <a class="wikilink abifile-wikilink" data-tippy-content="
Test of the extension of the pspcod=8 input format to include
spin-orbit coupling with multiple non-local projectors and pseudized valence charge density
The relativistic extension is documented in the code package.
The Pt 5s and 5p semi-cores are treated as valence.
A complete description of the input format is avaiable in  doc/psp_infos/psp8_info.txt." data-toggle="popover" href="/tests/v7/Input/t17.abi" target="_blank">t17.abi</a> &hellip;</li>
<li>v9:  <a class="wikilink abifile-wikilink" data-tippy-content="
Crystalline alpha-quartz: GS calculations with UPF2 pseudos.
Test scalar-relativistic pseudos with useylm 0 and 1." data-toggle="popover" href="/tests/v9/Input/t130.abi" target="_blank">t130.abi</a> &hellip;</li>
<li>wannier90:  <a class="wikilink abifile-wikilink" data-tippy-content="
Cannot be executed in parallel (mlwfovl_qp)
Si fcc, in primitive cell (2 atoms/cell)
Test of self-consistent model GW (2 iterations) following Faleev et al.,
[PRL 93, 126406 (2004)] followed by construction of quasiparticle
maximally-localized wannier functions [Hamann &amp; Vanderbilt,
arXiv:0810.3616v1 (cond-mat.mtrl-sci)].  Cutoffs are set for test-
acceptable speed, and the results are poorly converged.  The input
file is sufficiently annotated to serve as a model.  Note that well-
converged GW calculations are extremely time consuming, and in general
it is advisable to run the SCGW part separately on a parallel system,
and then run a separate serial job modeled on the last dataset,
substituting &quot;irdwfk&quot; and &quot;irdqps&quot; for &quot;getwfk&quot; and &quot;getqps,&quot; with
appropriate links to the files produced in the serial run.  Note that
the _DEN file from the first dataset is also needed as input, although
the discontinued &quot;irdden&quot; input variable is not needed or supported.
Note that acceptable names for the secondary input file needed by the
wannier90 library are wannier90.win, w90.win, abo_DSn_w90.win (ndtset
&gt;0) and abo_w90.win (ndtset=0), where abo is the 4th line of the .files
file and n is the wannier dataset." data-toggle="popover" href="/tests/wannier90/Input/t03.abi" target="_blank">t03.abi</a> &hellip;</li>
</ul>
</details>
<p>The equations of motion are:</p>
<p>M  I  d  2  R  I  /dt  2  = F  I  -  <strong>vis</strong>  dR  I  /dt</p>
<p>The atomic unit of viscosity is hartree * (atomic time units)/Bohr 2. Units
are not critical as this is a fictitious damping used to relax structures. A
typical value for silicon is 400 with <a class="wikilink codevar-wikilink" href="/variables/rlx#dtion">dtion</a> of 350 and atomic mass 28
<a class="wikilink codevar-wikilink" href="/variables/rlx#amu">amu</a>. Critical damping is most desirable and is found only by optimizing
 <strong>vis</strong>  for a given situation.</p>
<h2 id="wtatcon"><strong>wtatcon</strong><a class="headerlink" href="#wtatcon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> WeighTs for AToms in CONstraint equations<br />
<em>Characteristics:</em> <a class="wikilink" href="/guide/abinit#parameters">NO_MULTI</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-tippy-content="How to constrain the geometry of the system in geometry optimization, molecular dynamics or searches" data-toggle="popover" href="/topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3,<a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>,<a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>)<br />
<em>Default value:</em> 0<br />
<em>Added in version:</em> before_v9  </p>
<details class="note">
<summary>Test list (click to open). Rarely used, [3/1314] in all abinit tests, [0/178] in abinit tutorials</summary>
<ul>
<li>v3:  <a class="wikilink abifile-wikilink" data-tippy-content="
8 atoms of Si in an elongated box.
4 of them are fixed, while atoms 5 and 6
and 7 and 8 are grouped by pair, such as
the sum of their relative separations along y is constrained to 8.0 .
This is a crazy constraint, just to show that it can be done !" data-toggle="popover" href="/tests/v3/Input/t40.abi" target="_blank">t40.abi</a></li>
<li>v6:  <a class="wikilink abifile-wikilink" data-tippy-content="
Hydrogen diatomic molecule in a cell, close to BCC.
One hydrogen atom is at the origin. The other lies initially
along the 1 1 1 direction, at a distance close to the one of the free dimer distance.
The final position is the mirror with respect to the plane perpendicular to x, passing at 1/2 0 0 .
The cell parameter is less than twice the H2 interatomic distance, so that the transition
path come close to the x axis, although it never reaches it (the closest being in the
mirror plane).
Test the images : 6 images, exploring the transition path by a simple algorithm.
MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-toggle="popover" href="/tests/v6/Input/t22.abi" target="_blank">t22.abi</a></li>
<li>v8:  <a class="wikilink abifile-wikilink" data-tippy-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-toggle="popover" href="/tests/v8/Input/t05.abi" target="_blank">t05.abi</a></li>
</ul>
</details>
<p>Gives the weights determining how the motion of atoms is constrained during
structural optimization or molecular dynamics (see <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a>, <a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a>,
and <a class="wikilink codevar-wikilink" href="/variables/rlx#iatcon">iatcon</a>). For each of the <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a> independent constraint equations,
wtatcon is a 3*<a class="wikilink codevar-wikilink" href="/variables/rlx#natcon">natcon</a> array giving weights, W  I, for the x, y, and z
components of each of the atoms (labeled by I) in the list of indices
<a class="wikilink codevar-wikilink" href="/variables/rlx#iatcon">iatcon</a>. Prior to taking an atomic step, the calculated forces, F  I, are
replaced by projected forces, F&rsquo;  I, which satisfy the set of constraint equations</p>
<p>Sum  mu=x,y,z; I=1,natcon: W  mu,I  * F&rsquo;  mu,I  = 0 for each of the <a class="wikilink codevar-wikilink" href="/variables/rlx#nconeq">nconeq</a> arrays W  I.</p>
<p>Different types of motion constraints can be implemented this way. For example,</p>
<div class="codehilite"><pre><span></span><code>nconeq 1 natcon 2 iatcon 1 2 wtatcon 0 0 +1 0 0 -1
</code></pre></div>

<p>could be used to constrain the relative height difference of two adsorbate
atoms on a surface (assuming their masses are equal), since F&rsquo;  z,1  - F&rsquo;
z,2  = 0 implies z  1  - z  2  = constant.</p>







  
  






                

<!--
Add dropdown button with toolbar (based on https://bootsnipp.com/snippets/featured/inbox-by-gmail)
and return to Top button (based on https://codepen.io/rdallaire/pen/apoyx)
-->
<div class="md-container">
  <div class="row">
    <div id="inbox">
      <div class="fab btn-group show-on-hover dropup">
        <div data-toggle="tooltip" data-placement="left" title="Menu" style="margin-left: 42px;">
          <button id="return-to-top" type="button" class="btn btn-primary btn-io dropdown-toggle" data-toggle="dropdown">
            <span class="fa-stack fa-2x">
               <i class="fa fa-circle fa-stack-2x fab-backdrop"></i>
               <i class="fa fa-chevron-up fa-stack-1x fa-inverse fab-primary"></i>
               <i class="fa fa-map-signs fa-stack-1x fa-inverse fab-secondary"></i>
            </span>
          </button>
        </div>
        <ul class="dropdown-menu dropdown-menu-right" role="menu">
        
        
          <li>
            <a href="https://gmatteo.github.io/abinit/about/release-notes/"
               data-toggle="popover" data-tippy-content="About">
               <i class="fa fa-info-circle"></i>
            </a>
          </li>
        
          <li>
            <a href="https://gmatteo.github.io/abinit/developers/git_and_gitlab/"
               data-toggle="popover" data-tippy-content="Developers">
               <i class="fa fa-terminal"></i>
            </a>
          </li>
        
          <li>
            <a href="https://gmatteo.github.io/abinit/theory/geometry/"
               data-toggle="popover" data-tippy-content="Theory">
               <i class="fa fa-book"></i>
            </a>
          </li>
        
          <li>
            <a href="https://gmatteo.github.io/abinit/faq/configuration_and_compilation/"
               data-toggle="popover" data-tippy-content="FAQ">
               <i class="fa fa-graduation-cap"></i>
            </a>
          </li>
        
          <li>
            <a href="https://gmatteo.github.io/abinit/tutorial/"
               data-toggle="popover" data-tippy-content="Tutorials">
               <i class="fa fa-file-text-o"></i>
            </a>
          </li>
        
          <li>
            <a href="https://gmatteo.github.io/abinit/variables/"
               data-toggle="popover" data-tippy-content="Variables">
               <i class="fa fa-question"></i>
            </a>
          </li>
        
          <li>
            <a href="https://gmatteo.github.io/abinit/topics/features/"
               data-toggle="popover" data-tippy-content="Topics">
               <i class="fa fa-user-circle-o"></i>
            </a>
          </li>
        
          <li>
            <a href=""
               data-toggle="popover" data-tippy-content="User Guides">
               <i class="fa "></i>
            </a>
          </li>
        
        </ul>
      </div>
    </div>
  </div>
</div>

              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="footer.title">
      
        <a href="../dfpt/" class="md-footer__link md-footer__link--prev" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              DFPT
            </div>
          </div>
        </a>
      
      
        <a href="../rttddft/" class="md-footer__link md-footer__link--next" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              RTTDDFT
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
            <center>Abinit version 10.4.7</center>
        
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2020-2024 The Abinit Group
            
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/abinit/abinit" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.youtube.com/watch?v=gcbfb_Mteo4" target="_blank" rel="noopener" title="www.youtube.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305m-317.51 213.508V175.185l142.739 81.205z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="http://www.facebook.com/abinit.org" target="_blank" rel="noopener" title="www.facebook.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M512 256C512 114.6 397.4 0 256 0S0 114.6 0 256c0 120 82.7 220.8 194.2 248.5V334.2h-52.8V256h52.8v-33.7c0-87.1 39.4-127.5 125-127.5 16.2 0 44.2 3.2 55.7 6.4V172c-6-.6-16.5-1-29.6-1-42 0-58.2 15.9-58.2 57.2V256h83.6l-14.4 78.2H287v175.9C413.8 494.8 512 386.9 512 256"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/hashtag/abinit" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253"/></svg>
    </a>
  
</div>
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "header.autohide"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": "10.4.7"}</script>
    
    
      <script src="../../assets/javascripts/bundle.83f73b43.min.js"></script>
      
        <script src="../../extra_javascript/abidocs.js"></script>
      
    
  </body>
</html>